---
title: "Laboratory Notebook"
output: html_notebook
---

July 17, 2020

Parkinson's disease (PD) is the second most common neurodegenerative disease in the world [1]. It is characterized by the loss of dopaminergic neurons in the substantia nigra pars compacta (SNpc), as well as the caudal-to-rostral development of α-synclein-containing Lewy bodies (LB) [1]. Neurons containing neuromelanin (NM), including the dopaminergic neurons of the SNpc, have been observed to be selectively vulnerable to neurodegeneration and death in PD [2]. However, not much is understood about the biological mechanisms behind this selective vulnerability, nor is it fully understood how NM impacts dopaminergic cell death. Interestingly, age is the greatest risk factor for PD [1], and NM is has been confirmed to accumulate in the SNpc with age in healthy individuals [2-6]. Furthermore, NM levels appear to be generally higher in PD patients [2,7], and numerous studies have suggested using brain MRIs to diagnose presymptomatic PD based on brain NM levels [2, 7-8]. Therefore, developing a better understanding of the biological mechanisms that contribute to age-dependent NM accumulation could shine some light onto the link between neuromelanin and dopaminergic cell death in Parkinson's disease.

References:

[1] A. Kouli, K. M. Torsney, and W. Kuan, “Parkinson’s Disease: Etiology, Neuropathology, and Pathogenesis,” pp. 3–26, 2018, doi: https://doi.org/10.15586/codonpublications.parkinsonsdisease.2018.ch1.

[2] M. Vila, “Neuromelanin, aging, and neuronal vulnerability in Parkinson’s disease,” Mov. Disord., vol. 34, no. 10, pp. 1440–1451, 2019, doi: 10.1002/mds.27776.

[3] R. L. Haining and C. Achat-Mendes, “Neuromelanin, one of the most overlooked molecules in modern medicine, is not a spectator,” Neural Regen. Res., vol. 12, no. 3, pp. 372–375, 2017, doi: 10.4103/1673-5374.202928.

[4] J. Segura-Aguilar and I. Paris, “Handbook of neurotoxicity,” Handb. Neurotox., vol. 1–3, pp. 1–2371, 2014, doi: 10.1007/978-1-4614-5836-4.

[5] P. Muñoz, S. Huenchuguala, I. Paris, and J. Segura-Aguilar, “Dopamine oxidation and autophagy,” Parkinsons. Dis., vol. 2012, 2012, doi: 10.1155/2012/920953.

[6] A. Herrera, P. Muñoz, H. W. M. Steinbusch, and J. Segura-Aguilar, “Are Dopamine Oxidation Metabolites Involved in the Loss of Dopaminergic Neurons in the Nigrostriatal System in Parkinson’s Disease?,” ACS Chem. Neurosci., vol. 8, no. 4, pp. 702–711, 2017, doi: 10.1021/acschemneuro.7b00034.

[7] I. Carballo-Carbajal et al., “Brain tyrosinase overexpression implicates age-dependent neuromelanin production in Parkinson’s disease pathogenesis,” Nat. Commun., vol. 10, no. 1, 2019, doi: 10.1038/s41467-019-08858-y.

[8] D. Sulzer et al., “Neuromelanin detection by magnetic resonance imaging (MRI) and its promise as a biomarker for Parkinson’s disease,” Nature, vol. 4, no. 1. Nature Publishing Group, Dec. 01, 2018, doi: 10.1038/s41531-018-0047-3.

____________________________________________________________________________________________

July 18, 2020

Upon further literature review, it was discovered that there is one known pathway commonly noted in the literature that is connected to both dopaminergic cell death and neuromelanin: dopamine oxidation. Dopamine oxidation is assumed to be a spontaneous process that culminates in neuromelanin formation [9]. Dopamine auto-oxdizes in the cytosol of dopaminergic neurons to form dopamine-o-quinone, and continues to form aminochrome [9-11]. Aminochrome is the most stable quinone formed in this pathway, and has been linked to numerous neurodegenerative mechanisms [9-11], and a mouse model of aminochrome has been able to induce dopaminergic neuron degeneration but not death [12]. Aminochrome then forms 5,6-indolequinone, which has been deteced in neuromelanin granules along with a number of other dopamine oxidation products [9-11]. Dopamine can be prevented from oxidizing in the cytosol by packing it into low pH synaptic vesicles by the action of the VMAT2 enzyme [13]. Presumably, if more dopamine is produced, the oxidation and the resulting neurotoxicity might increase as well. However, no study has been able to induce neurodegeneration or dopaminergic neuron death in rodent models by solely increasing dopamine levels [2], so there may be more involved in this pathway than previously speculated.

References:

[9] A. Herrera, P. Muñoz, H. W. M. Steinbusch, and J. Segura-Aguilar, “Are Dopamine Oxidation Metabolites Involved in the Loss of Dopaminergic Neurons in the Nigrostriatal System in Parkinson’s Disease?,” ACS Chem. Neurosci., vol. 8, no. 4, pp. 702–711, 2017, doi: 10.1021/acschemneuro.7b00034.

[10] J. Segura-Aguilar and I. Paris, “Handbook of neurotoxicity,” Handb. Neurotox., vol. 1–3, pp. 1–2371, 2014, doi: 10.1007/978-1-4614-5836-4.

[11] P. Muñoz, S. Huenchuguala, I. Paris, and J. Segura-Aguilar, “Dopamine oxidation and autophagy,” Parkinsons. Dis., vol. 2012, 2012, doi: 10.1155/2012/920953.

[12] A. Herrera et al., “Aminochrome induces dopaminergic neuronal dysfunction: a new animal model for Parkinson’s disease,” Cell. Mol. Life Sci., 2016, doi: 10.1007/s00018-016-2182-5.

[13] K. M. Lohr and G. W. Miller, “VMAT2 and Parkinson’s disease: harnessing the dopamine vesicle,” NCBI, vol. 14, no. 10, pp. 1115–1117, 2014, doi: 10.1586/14737175.2014.960399.VMAT2.

____________________________________________________________________________________________

July 19, 2020

RESEARCH RATIONALE

Recently, the dopamine oxidation pathway and neuromelanin gave been suggested to contribute to the neurotoxic mechanisms that cause Parkinson’s disease (PD), the second most common neurodegenerative disorder in the world. The goal of this research is to determine the extent of involvement of numerous genes regulating this pathway and to identify new pathways contributing to neuromelanin formation using gene expression data.

____________________________________________________________________________________________

July 20, 2020

Based on a review of literature, the target genes for this study were determined. Although dopamine oxidation itself is assumed to be spontaneous, there are a number of antioxidant pathways that are catalyzed by enzymes and therefore linked to genes and gene expression. These are the genes that have been shown to be involved in the dopamine oxidation pathway in some way.

  - Tyrosine Hydroxylase (TH)
  - Aromatic L-amino acid decarboxylase (DDC)
  - Vesicular Monoamine Transporter 2 (SLC18A2)
  - Dopamine Active Transporter (SLC6A3)
  - Glutathione S-Transferase Mu 2 (GSTM2)
  - DT-diaphorase (NQO1)
  - Monoamine Oxidase A (MAOA)
  - Monoamine Oxidase B (MAOB)
  - Catechol-O-methyltransferase (COMT)
  
TH and DDC are both involved in dopamine synthesis [2, 9-11]. SLC18A2 (the gene coding for the VMAT2 enzyme), as explained previously, works to pack dopamine into synaptic vesicles to prevent its oxidation [13]. It is important to note that VMAT2 levels have been inversely correlated with neuromelanin levels, and the VMAT2 enzyme has been observed ot be dysfunctional in PD [13]. SLC6A3 codes for the dopamine active transporter, an enzyme which transfers dopamine from extracellular matrix to dopaminergic neuron cytosol [9]. GSTM2 and NQO1 both code for enzymes that are involved in metabolizing aminochrome into non-neurotoxic products, and therefore have been suggested to be neuroprotective against aminochrome [9-11]. MAOA, MAOB, and COMT are involved in two antioxidant pathways that prevent dopamine oxidation by metabolizing dopamine into homovanillic acid [9-11].

It has not been commonly noted in the literature that any of these genes are differentially expressed in the brains of Parkinson's patients and normal people, although one study noted that TH and SLC6A3 are slightly downregulated in the dopaminergic neurons of Parkinson's patient, according to a qRT-PCR analysis [14]. The evidence of this downregulation, however, was weak and contradicted microarray results from the same study which noted no downregulation of those genes, so no clear conclusion could be established.

References: 

[14] F. Simunovic et al., “Gene expression profiling of substantia nigra dopamine neurons: further insights into Parkinson’s disease pathology,” Brain, vol. 132, no. 7, pp. 1795–1809, 2009, doi: 10.1093/brain/awn323.

____________________________________________________________________________________________

July 21, 2020

The objectives were determined. Overall goals of this study include: 
  
  1) Age in SN: Analyzing gene expression in substantia nigra of non-neurological people of different ages and Parkinson's patients of different ages
  2) mDBR (melanized dopaminergic brain region expression analysis): Analyzing gene expression in melanized and non-melanized dopaminergic brain regions of non-neurological people and Parkinson's patients
  
This study can be characterized as an observational study, as the data was obtained from available online sources and no treatment was applied to the subjects from which the samples were obtained.
  
Preliminary datasets to use for analysis were also collected:
  
  1) Age Correlations
      - Non-neurological individuals: UK Brain Expression Consortium (GSE46706)
      - Parkinson's patients: GSE8397, GSE20292
  2) mDBR
      - Non-neurological individuals (only): Allen Human Brain Atlas

____________________________________________________________________________________________
____________________________________________________________________________________________

AGE CORRELATIONS - NON-NEUROLOGICAL INDIVIDUALS
____________________________________________________________________________________________
____________________________________________________________________________________________

July 22, 2020 

  Initially, the SuperSeries GSE46706 was loaded into RStudio using the getGEO() function from the GEOquery package. GSE46706 was generated by UK Brain Expression Consortium, and contains gene expression data in the brain from a large cohort of neurologically and neuropathologically normal individuals. This SuperSeries consists of two platforms, GPL5175 and GPL5188, both consisting of Affymetrix Human Exon 1.0 ST Arrays, and corresponding to SubSeries GSE60862 and GSE60863 respectively. GPL5175 is the transcript (gene) version, and thus GSE60862 contains a gene-level analysis of expression, and GPL5188 is the probe (exon) version, and thus GSE60863 is contains an exon-level analysis. Each SubSeries consists of 1231 samples from 134 Caucasian individuals. It was initially submitted to GEO May 7, 2013, and updated Aug 12, 2019.
  
  The arrays from both SubSeries had been pre-processed using RMA quantile normalization with background correction. Therefore, normalization after loading the data would most likely not be necessary.
  
  However, the GSE46706 file was too large and uploading it resulted in memory exhaustion. Therefore, it had to be selectively downloaded for only the desired samples. GSE60862 and GSE60863 were loaded in separately, selecting samples GSM1491421 to GSM1492619. These two samples correspond to the first and last substantia nigra samples in the dataset. Although only substantia nigra samples were desired, the dataset did not list the samples in order of brain region. Since the GSElimits argument in the getGEO() function only allows for a vector of length two specifying the sample to start at and the sample to end at, numerous samples were also downloaded that were not from the substantia nigra. Therefore, the data would need to be filtered again for only substantia nigra samples.
  
```{r, echo=FALSE}
library(GEOquery)

# Load in GSE46706, store as large list. This did not work because GSE46706 is too large of a file.
# gset <- getGEO('GSE46706', GSEMatrix = TRUE)

# Load in select samples from GSE60862, store as large list
gset_gene <- getGEO('GSE60862', destdir = '/Users/rezaadibnia/Desktop/PD NM Project/GSE46706 data/GSE60862', GSElimits = c('GSM1491421', 'GSM1492619'), GSEMatrix = TRUE)

# Could not load in GSE60863, as it led to memory exhaustion.
# gset_exon <- getGEO('GSE60863', destdir = '/Users/rezaadibnia/Desktop/PD NM Project/GSE46706 data/GSE60863', GSElimits = c('GSM1491421', 'GSM1492619'), GSEMatrix = TRUE)

```

  Gene expression data from GSE60862 (GPL5157) was read as a numeric matrix (gene), with columns as samples and rows as gene probe sets. Samples were lined with the age of the donor from which the sample was obtained and the brain region of the sample (gene_filt).
  
  gene_filt was then filtered for expression data in the substantia nigra (gene_filt_SN), which yielded 101 samples. Since gene_filt was transposed before filtering, gene_filt_SN contained rows as samples (observations) and columns as gene probe sets (variables). Age, as aligned with the substantia nigra samples, was extracted as its own numeric vector called age_gene_SN to use for computing correlations using the cor() and cor.test() functions in the future. While gene_filt_SN, after cleaning up the data frame, only contained the expression data, a copy of gene_filt_SN, gene_filt_SN_with_age, was made that included the age column as well.

```{r, echo=FALSE}
library(GEOquery)
library(tidyverse)
library(stringr)

# to avoid scientific notation
options(scipen = 999)

########################## PREPARE DATA ##########################

# Gene expression microarray data. Genes are listed using numeric IDs.
gene <- gset_gene[["GSE60862_series_matrix.txt.gz"]]@assayData[["exprs"]]

# Gives age of donor and brain region for each sample
gene_sample_annotations <- gset_gene[["GSE60862_series_matrix.txt.gz"]]@phenoData@data

# Names of genes that correspond to each numeric ID
gene_names <- gset_gene[["GSE60862_series_matrix.txt.gz"]]@featureData@data

# Observe features of data frames
# head(gene_sample_annotations)
# head(gene_names)


# add age of donor (column 48) and brain region (column 11) for each sample
gene_filt <- t(gene)
gene_filt <- gene_filt %>% 
  cbind(gene_sample_annotations[,11], gene_sample_annotations[,48]) %>% 
  data.frame()
rm(gene) # save space


# filter for only substantia nigra (SN) gene expression data
gene_filt_SN <- gene_filt %>% 
  filter(V26494 == 'brain region: substantia nigra')


# extract age as its own numeric vector
extract_age_gene <- function(gene_filt_brain_region) {
  
  # All data frames filtered for brain region have same genes (columns), therefore $V26495 works for all filtered data frames
  age_gene <- as.numeric(gene_filt_brain_region$V26495)
  
  return(age_gene)
  
}

age_gene_SN <- extract_age_gene(gene_filt_SN)

# Creates a numeric data frame version of the inputted data frame
# All expression values must be converted to numeric, because using cbind() on gene_filt converts them all to character values
convert_to_numeric <- function(gene_expression_dataframe) {
  
  df <- matrix(data = NA, nrow = dim(gene_expression_dataframe)[1], ncol =
                 dim(gene_expression_dataframe)[2])
  
  for (i in 1:dim(gene_expression_dataframe)[2]) {
    df[,i] <- c(as.numeric(gene_expression_dataframe[,i]))
  }
  
  rownames(df) <- row.names(gene_expression_dataframe)
  colnames(df) <- colnames(gene_expression_dataframe)
  
  return(df)
}

# Call convert_to_numeric function, with parameter being gene_filt_SN without the brain region and age columns
gene_filt_SN <- convert_to_numeric(gene_filt_SN[,-c(26494,26495)])
gene_filt_SN_with_age <- cbind.data.frame(gene_filt_SN, age_gene_SN)


```

  After exploring the gene_names data frame, it appeared that there were two different probe sets corresponding to the DDC gene (3001557 and 3050388), and two different probe sets both corresponding to the MAOA gene (3975227 and 4055661). For each gene, the probe set with the greatest variance across all SN samples was chosen [15]. For DDC the ID chosen was 3050388, and for MAOA the ID chosen was 4055661.
  
  Target Gene IDs:
  
  - TH - 3359180
  - DDC - 3050388
  - SLC18A2 - 3266279
  - SLC6A3 - 2845921
  - GSTM2 - 2350952
  - NQO1 - 3696666
  - MAOA - 4055661
  - MAOB - 4006210
  - COMT - 3937092
  
  After determining the appropriate probe set IDs to use for each target gene, the expression data of the target genes in the substantia nigra was gathered into one data frame called target_gene_expression_SN, in case such a data frame would be useful in the future. However, it is important to note that this data frame was formed using the gather() function from dplyr, meaning it was formatted such that each row corresponded to the expression value for a specific gene in a specific sample.

```{r}
library(hash)

# select probe sets for DDC and MAOA based on variance

# DDC
var(gene_filt_SN[,"X3001557"])
var(gene_filt_SN[,"X3050388"]) # greater variance

# MAOA
var(gene_filt_SN[,"X3975227"])
var(gene_filt_SN[,"X4055661"]) # greater variance


########################## GATHER TARGET GENE EXPRESSION DATA ########################## 

# keys as gene IDs and values as gene names
target_gene_IDs_to_names <- hash(c('X3359180', 'X3050388', 
                                   'X3266279', 'X2845921', 
                                   'X2350952', 'X3696666', 
                                   'X4055661', 'X4006210', 'X3937092'), 
                                 c('TH', 'DDC', 
                                   'SLC18A2', 'SLC6A3', 
                                   'GSTM2', 'NQO1',
                                   'MAOA',
                                   'MAOB', 'COMT'))

# gathering expression data for all target genes in a certain brain region into 1 data frame
target_genes_only <- function(gene_filt_brain_region) {
  
  gene_filt_brain_region <- data.frame(gene_filt_brain_region)
  target_gene_expression <- gene_filt_brain_region %>% 
    select('X3359180', 'X3050388', 'X3266279', 'X2845921', 'X2350952', 'X3696666', 'X4055661', 'X4006210', 'X3937092') %>%
    gather(key = 'Gene', value = 'Expression', na.rm = TRUE)

  # convert all gene IDs in target_gene_expression to gene names
  for (i in 1:dim(target_gene_expression)[1]) {
    target_gene_expression[i,1] <- target_gene_IDs_to_names[[target_gene_expression[i,1]]]
  }
  
  return(target_gene_expression)
}

target_gene_expression_SN <- target_genes_only(gene_filt_SN)


```

References: 

[15] A. Keo et al., “Transcriptomic signatures of brain regional vulnerability to Parkinson’s disease,” Commun. Biol., vol. 3, no. 1, pp. 1–12, 2020, doi: 10.1038/s42003-020-0804-9.

____________________________________________________________________________________________

July 23, 2020

  Exploratory analysis of gene_filt_SN was performed. A boxplot of all 101 samples (columns) revealed that the data was relatively median-centered. This is reasonable, since it was indicated on the page of each sample that the arrays were pre-processed using RMA quantile normalization with background correction. If any of the boxplots deviated significantly than the others, this could have indicated that the fluorescence for that sample was either systematically higher or lower than for others, causing the reads for it to be erroneously higher or lower than for others. In such cases, quantile normalization would be necessary. However, as the data was relatively median-centered, the data did not need to be normalized. This is reasonable since the study from which the data was obtained did note that RMA normalization with background correction had been performed on the data before it was submitted to GEO. Additionally, the histogram showed that the data was roughly symmetric, indicating that Pearson's correlations could be applied to the data. 

```{r}
library(dplyr)
library(preprocessCore)

# SN gene expression data does appear relatively median-centered, therefore normalization is not necessary. t(gene_filt_SN) makes the boxplot display the distributions of gene expression values for all the genes across all samples.
gene_filt_SN %>% t %>% boxplot

# Quantile normalized
gene_filt_SN %>% 
  t %>% 
  normalize.quantiles %>% 
  boxplot

gene_filt_SN %>% 
  t %>% 
  hist(main = 'Histogram of Substantia Nigra Gene Expression Data', xlab = 'Expression')
# gene_filt_SN %>% t %>% as.matrix %>% normalize.quantiles %>% hist(main = 'Histogram of normalized Substantia Nigra Gene Expression Data', xlab = 'Expression')

# # log2 transform of gene_filt_SN
# gene_filt_SN_log2 <- log2(gene_filt_SN)
# 
# boxplot(t(gene_filt_SN_log2))
# gene_filt_SN_log2 %>% t %>% as.matrix %>% normalize.quantiles %>% boxplot
# 
# hist(gene_filt_SN_log2, main = 'Histogram of log2-transformed Substantia Nigra Gene Expression Data', xlab = 'Expression')
# gene_filt_SN_log2 %>% t %>% as.matrix %>% normalize.quantiles %>% hist(main = 'Histogram of normalized, log2-transformed Substantia Nigra Gene Expression Data', xlab = 'Expression')
# 
# rm(gene_filt_SN_log2)

```

  It is apparent that neither quantile normalization nor logarithmic transformations are necessary, as the raw expression data is median-centered and roughly symmetric. The log2 transformation only makes the data more left-skewed. Therefore, the raw data frame gene_filt_SN will be used for further analyses.
  
  From the first boxplot, we can also see that there are many more high outliers in the data than low outliers. This means there are many more gene with very high expression in the substantia nigra than genes with very low expression in the substantia nigra.

  Additionally, the age data (age_gene_SN) needed to be checked to ensure that it matched a normal distribution. Using a significance level of α = 0.05, a Shapiro-Wilk normality test gave a p-value of 0.1511, indicating that the data was not significantly different from a normal distribution. To further confirm that age_gene_SN matched a normal distribution, a histogram of age_gene_SN was computed, and it showed that the data was roughly symmetric. A normality plot (or quantile-quantile plot) also showed negligible deviation from a normal distribution, as the correlation between the age data and a normal distribution appeared strong with few outliers from the trend.

```{r}
library(ggpubr)

shapiro.test(age_gene_SN)

hist(age_gene_SN, main = 'Histogram of Patient Ages', xlab = 'Age of Patient')

ggqqplot(age_gene_SN, ylab = 'Patient Ages')

```
____________________________________________________________________________________________

July 24, 2020

  Downstream analysis of gene_filt_SN was performed. A correlation matrix of the target genes and the age of the samples was computed using the rcorr() function from the Hmisc package. Unfortunately, the original target_gene_expression_SN data frame, which was created using the gather() function, would not be useful in this case. The target_gene_expression_SN data frame was redefined to contain the columns has samples at nine rows, each row corresponding to a different target gene. The correlation matrix showed both correlations between target gene expression in the substantia nigra and patient age, and correlations between the expression of different target genes with each other to determine their interconnectedness. The rcorr() function also outputted a matrix of the p-values of correlation. The output of this function was stored in the list cor_matrix_target_SN. The correlation matrix was also visualized using the corrplot() function from the corrplot package, with insignificant correlations (p > 0.05) left blank.

```{r}
library(Hmisc)
library(corrplot)

# list of gene IDs of target genes
target_genes = c('X3359180', 'X3050388', 'X3266279', 'X2845921', 'X2350952', 'X3696666', 'X4055661', 'X4006210', 'X3937092')

target_gene_expression_SN <- matrix(data = NA, nrow = length(target_genes), ncol = dim(gene_filt_SN)[1])
target_gene_expression_SN <- data.frame(target_gene_expression_SN)

for (i in 1:length(target_genes)) {
  df <- gene_filt_SN %>% 
    t %>% 
    data.frame %>% 
    filter(rownames(data.frame(t(gene_filt_SN))) == as.character(target_genes[i]))
  target_gene_expression_SN[i,] <- df
}

colnames(target_gene_expression_SN) <- rownames(gene_filt_SN)
rownames(target_gene_expression_SN) <- target_genes

for (i in 1:length(rownames(target_gene_expression_SN))) {
  rownames(target_gene_expression_SN)[i] <- target_gene_IDs_to_names[[rownames(target_gene_expression_SN)[i]]]
}

target_gene_expression_SN <- target_gene_expression_SN %>% add_row()
target_gene_expression_SN[10,] <- age_gene_SN
rownames(target_gene_expression_SN)[10] <- "age"

# Correlation matrix of target genes with raw numbers
cor_matrix_target_SN <- target_gene_expression_SN %>% t %>% as.matrix %>% rcorr
cor_matrix_target_SN

# Visualization of correlation matrix, with insignificant correlations left blank
corrplot(cor_matrix_target_SN$r, type="upper", order="hclust", 
         p.mat = cor_matrix_target_SN$P, sig.level = 0.05, insig = "blank")

```

  From these correlation matrix, it can be observed that most target genes, except SLC18A2, SLC6A3, MAOA, and MAOB, are significantly correlated with age using a significance level α = 0.05. NQO1 has the greatest |r| of all the target genes, with r = -0.29 and p = 0.0029. The genes TH, DDC, SLC18A2, and SLC6A3 were highly correlated with each other (r > 0.90), which is reasonable since they are all directly involved in dopamine synthesis. The other genes did have some significant correlations with each other, though none had correlation coefficients as large as those between TH, DDC, SLC18A2, and SLC6A3.

____________________________________________________________________________________________

July 25, 2020

  All the probe sets were aligned to individual genes in the data frame gene_filt_SN according to the procedure below, which matches procedures previously used in the literature [15]. This would allow for performing age correlations for all the genes. Probe sets that were not assigned to a gene were removed.
  
  - If there is only one probe set for a gene, that one is chosen
  - If there are two probe sets for one gene, the one with the max variance across all samples is chosen (method = maxRowVariance)
  - If there more than two probe sets for one gene, the probe set with the highest connectivity (summed adjacency) is chosen (connectivityBasedCollapsing = TRUE)

```{r}
library(dplyr)
library(WGCNA)

gene_filt_SN <- gene_filt_SN %>% t %>% data.frame
gene_filt_SN$probe_set_ID <- gene_names[, "ID"]
gene_filt_SN$gene_assignment <- gene_names[, "gene_assignment"]

# Remove probe sets not assigned to a gene first
probesets_missing_gene <- gene_filt_SN$gene_assignment == "---"
print(paste(sum(probesets_missing_gene), "probe sets without gene assignments"))
gene_filt_SN <- gene_filt_SN %>% filter(gene_filt_SN$gene_assignment != "---")


# Align probe sets to genes
for (i in 1:dim(gene_filt_SN)[1]) {
  gene_filt_SN[i,  "gene_assignment"] <- strsplit(gene_filt_SN[i, "gene_assignment"], " ")[[1]][3]
}
rownames(gene_filt_SN) <- gene_filt_SN$probe_set_ID
probe2gene_UKBEC <- collapseRows(gene_filt_SN[, -103],
                           rowGroup = gene_filt_SN$gene_assignment,
                           rowID = gene_filt_SN$probe_set_ID,
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
gene_filt_SN <- probe2gene_UKBEC[["datETcollapsed"]] %>% data.frame
print(paste(dim(gene_filt_SN)[1], "genes were aligned to 24882 probe sets in the UKBEC data"))


```

____________________________________________________________________________________________

July 26, 2020

  It is possible that the target genes may not be only genes involved in neuromelanin formation and/or dopamine oxidation noted in literature; there may be genes involved in these processes that are not noted in literature. Therefore, correlations were computed for all the genes, and the p-values of correlation were adjusted using the same method used for the target genes. The goal of this was to identify which genes did change in expression with age, and then later determine whether they have some relationship to neuromelanin formation or the dopamine oxidation pathway, either based on further literature review or melanized dopaminergic brain region (mDBR) differential expression analysis.
  
```{r}
library(dplyr)

# Create a data frame that will store correlation coefficients and p values for each gene 
all_age_correlations <- matrix(nrow = dim(gene_filt_SN)[1], ncol = 5) %>% data.frame
rownames(all_age_correlations) <- gene_filt_SN$probe_set_ID
all_age_correlations[,1] <- rownames(gene_filt_SN)
all_age_correlations[,2] <- gene_filt_SN$probe_set_ID
colnames(all_age_correlations) <- c("Gene", "ID", "Correlation Coefficient", "Raw p-value", "BH-corrected p-value")

# Compute age correlations for all genes and store raw p-values and correlation coefficients
for (i in 1:dim(gene_filt_SN)[1]) {
  
  cor <- cor.test(as.numeric(gene_filt_SN[i, 1:101]), age_gene_SN, method = "pearson")
  all_age_correlations[i, "Correlation Coefficient"] <- cor[["estimate"]][["cor"]]
  all_age_correlations[i, "Raw p-value"] <- cor[["p.value"]]
  
}

```

____________________________________________________________________________________________

July 27, 2020

  To minimize the false discovery rate of significant correlations, the Benjamini-Hochberg (BH) procedure was applied to all the p-values using the p.adjust() function.
  
```{r}
library(dplyr)

# Adjust p-values using Benjamini-Hochberg (BH) procedure
all_age_correlations[, "BH-corrected p-value"] <- p.adjust(all_age_correlations[, "Raw p-value"], 
                                                           method = "BH")

# Filter for target genes
target_genes <- c(3359180, 3050388, 3266279, 2845921, 2350952, 3696666, 4055661, 4006210, 3937092)
age_cor_target <- all_age_correlations %>% 
  filter(all_age_correlations$ID %in% target_genes)

```

____________________________________________________________________________________________

July 29, 2020

  After adjusting p-values, it was evident that none of the target genes were actually significantly correlated with age. Even for NQO1, the most significantly correlated target gene, had p = 0.06190312 > 0.05. However, filtering for genes that are significantly correlated with age may reveal some information about biological pathways that are altered with age upon future enrichment analysis. Therefore, as stated in the procedure, the age correlations of all genes were filtered for the top 10% of genes with the highest correlation coefficients and the bottom 10% of genes with the lowest correlation coefficients, given that their BH-corrected p-values < 0.05. These genes are called ACGs (age-correlated genes).
  
```{r}
library(dplyr)

# Filter for genes significantly correlated with age
sig_age_correlations <- all_age_correlations %>% 
  filter(all_age_correlations[, "BH-corrected p-value"] < 0.05)

# Filter for genes with positive correlation coefficient
pos_sig_age_correlations <- sig_age_correlations %>% 
  filter(sig_age_correlations[, "Correlation Coefficient"] > 0)
write.csv(x=pos_sig_age_correlations, file="pos_sig_age_correlations.csv") # Writes csv file, saves to working directory to allow for simpler gene set enrichment analysis
print(paste(dim(pos_sig_age_correlations)[1], "genes were upregulated with age"))

# Filter for genes with negative correlation coefficient
neg_sig_age_correlations <- sig_age_correlations %>% 
  filter(sig_age_correlations[, "Correlation Coefficient"] < 0)
write.csv(x=neg_sig_age_correlations, file="neg_sig_age_correlations.csv") # Writes csv file, saves to working directory to allow for simpler gene set enrichment analysis
print(paste(dim(neg_sig_age_correlations)[1], "genes were downregulated with age"))

# Filter top 10% of positively correlated genes
top_pos_sig_age_correlations <- pos_sig_age_correlations %>% 
  filter(pos_sig_age_correlations[, "Correlation Coefficient"] > quantile(pos_sig_age_correlations[, "Correlation Coefficient"], 0.9))

# Filter bottom 10% of negatively correlated genes
top_neg_sig_age_correlations <- neg_sig_age_correlations %>% 
  filter(neg_sig_age_correlations[, "Correlation Coefficient"] < quantile(neg_sig_age_correlations[, "Correlation Coefficient"], 0.1))

# Recombine data frame
ACGs <- bind_rows(top_pos_sig_age_correlations, top_neg_sig_age_correlations)
write.csv(x=ACGs, file="ACGs.csv") # Writes csv file of ACGs, saves to working directory to allow for simpler gene set enrichment analysis
print(paste(dim(ACGs)[1], "age-correlated genes (ACGs)"))

```

____________________________________________________________________________________________

July 31, 2020

  A volcano plot of age correlations was constructed to better visualize the results. This included the -log(p-value of correlation) on the y-axis and the correlation coefficient on the x-axis. Positively correlated ACGs are blue, negatively correlated ACGs are red, and genes not significantly correlated with age are in grey.
```{r}
library(ggplot2)
library(dplyr)
library(ggrepel)

volcano_plot_df <- data.frame(all_age_correlations)

# Threshold values for fold change
# horz_min_age = (sig_age_correlations %>% filter(sig_age_correlations[, "Correlation Coefficient"] < 0) %>% data.frame)[, "Correlation.Coefficient"] %>% quantile(0.10)
# horz_max_age = (sig_age_correlations %>% filter(sig_age_correlations[, "Correlation Coefficient"] > 0) %>% data.frame)[, "Correlation.Coefficient"] %>% quantile(0.90)

# Add a column indicating whether probe is upregulated, downregulated, or not significant
volcano_plot_df$Correlation <- "Not Significant"

# If log2Foldchange > horz_max and pvalue < 0.05, set as "Upregulated" 
volcano_plot_df$Correlation[volcano_plot_df$Correlation.Coefficient > 0 & volcano_plot_df$BH.corrected.p.value < 0.05] <- "Upregulated"

# If log2Foldchange < horz_min and pvalue < 0.05, set as "Downregulated"
volcano_plot_df$Correlation[volcano_plot_df$Correlation.Coefficient < 0 & volcano_plot_df$BH.corrected.p.value < 0.05] <- "Downregulated"

# Add labels for top 0.01% of upregulated and downregulated genes
volcano_plot_df$label <- NA

volcano_plot_df$label[volcano_plot_df$BH.corrected.p.value < 0.05 & (volcano_plot_df$Correlation.Coefficient > quantile(volcano_plot_df$Correlation.Coefficient, 0.9999) | volcano_plot_df$Correlation.Coefficient < quantile(volcano_plot_df$Correlation.Coefficient, 0.0001))] <- volcano_plot_df$Gene[volcano_plot_df$BH.corrected.p.value < 0.05 & (volcano_plot_df$Correlation.Coefficient > quantile(volcano_plot_df$Correlation.Coefficient, 0.9999) | volcano_plot_df$Correlation.Coefficient < quantile(volcano_plot_df$Correlation.Coefficient, 0.0001))]

volcano_plot_df$label[volcano_plot_df$Gene == "PYGL"] <- volcano_plot_df$Gene[volcano_plot_df$Gene == "PYGL"]

# Set the theme
theme <- theme(panel.background = element_blank(),
               axis.line = element_line(colour = "black"),
               axis.title =  element_text(size = 15),
               plot.title = element_text(size = 16, face = "bold"),
               legend.title = element_text(size = 14))

# Create the plot
volcano_plot_age <- ggplot(data=volcano_plot_df, 
                            aes(Correlation.Coefficient, 
                                -log10(BH.corrected.p.value), 
                                col=Correlation)) +
                    geom_point(size = 1) +
                    geom_text_repel(aes(label=label), size = 5) + 
                    scale_colour_manual(values = c("Not Significant"="grey", 
                                                   "Downregulated"="red", 
                                                   "Upregulated"="blue"), 
                                        guide = guide_legend(override.aes = list(linetype = c(rep("blank", 3)), shape = c(rep(NA, 3))))) +
                    labs(x = bquote("Correlation with Age ("*italic(r)*")"), 
                         y = bquote("-"*log[10]*" "*italic(P)*"-value")) +
                    ggtitle("Age Correlations") +
                    theme

volcano_plot_age

```
  
  The volcano plot appears oddly parabolic, suggesting that there is a strong relationship between the p-value of correlation for a certain gene and its correlation coefficient.

____________________________________________________________________________________________

August 1, 2020

  Gene set enrichment analysis was performed for the 77 ACGs, the 123 genes significantly upregulated with age and the 634 genes significantly dowregulated with age separately. These gene sets were inputted into the PANTHER classification system for gene ontology (GO) enrichment analysis.  The goal of this was to identify the GO terms associated with biological processes for which each gene set is enriched using Fischer's exact test with false discovery rate calculation (search for "biological process"). A search was also done for enrichment of cellular components (search for "cellular components" in PANTHER database). Results in PANTHER classification system were deemed significant if the false discovery rate (FDR) p-value < 0.05. These gene sets were also inputted into the STRING database to determine how their proteins interact. While STRING DB does also provide GO enrichment analysis, PANTHER DB allows for the input of more genes, therefore GO results from PANTHER DB were preferred. The results of this analysis are listed below:
  
PANTHER DB:

  ACGs: 
    - No statistically significant results for biological process search 
    - Cellular component search:
      - Very high fold enrichment for astrocyte end-foot (fold enrichment > 100, FDR p = 0.0141) and astrocyte projection (fold enrichment = 43.91, FDR p = 0.0315)
      - Enriched for mitochondrial membrane component GO terms as well (mitochondrial membrane, fold enrichment = 4.18, FDR p = 0.0457)
      
  Positively correlated genes: 
    - No statistically significant results for both biological process search and cellular component search
    
  Negatively correlated genes:
     - Biological process search:
      - Relaxation of vascular associated smooth muscle (fold enrichment = 19.91, FDR p = 0.0373)
        - Unrelated to Parkinson's disease
      - Mitochondrial electron transport, NADH to ubiquinone (fold enrichment = 7.82, FDR p = 0.000672)
        - Includes oxidative phosphorylation (fold enrichment = 4.94, FDR p = 0.000345)
        - Biological process also noted to be dysfunctional in PD
      - Epithelial cilium movement involved in extracellular fluid movement	(fold enrichment = 7.62, FDR p = 0.0241) 
      - Mitochondrial respiratory chain complex I assembly	(fold enrichment = 6.34, FDR p = 0.00111)
        - NADH dehydrogenase complex assembly	(fold enrichment = 6.34, FDR p = 0.00116)
      - Protein targeting to peroxisome	(fold enrichment = 5.66, FDR p = 0.00121)
      - Cellular aldehyde metabolic process	(fold enrichment = 5.50, FDR p = 0.0236)
      - Fatty acid beta-oxidation	(fold enrichment = 5.32, FDR p = 0.0285)
      - Acyl-CoA metabolic process	(fold enrichment = 3.99, FDR p = 0.0432)
      - Extrinsic apoptotic signaling pathway	(fold enrichment = 3.91, FDR p = 0.0461)
      - Cellular amino acid catabolic process	(fold enrichment = 3.65, FDR p = 0.0315)
      - Translational elongation	(fold enrichment = 3.62, FDR p = 0.0324)
      - Nucleus organization	(fold enrichment = 3.46, FDR p = 0.0413)
      - Organic hydroxy compound metabolic process	(fold enrichment = 2.24, FDR p = 0.0183)
      - Cellular response to stress	(fold enrichment = 1.63, FDR p = 0.00809)
      - Detection of chemical stimulus involved in sensory perception of smell	(fold enrichment < 0.01, FDR p = 0.00273)
    - Cellular component search:
      - Astrocyte projection	(fold enrichment = 9.96, FDR p = 0.00435)
        - Glial cell projection	(fold enrichment = 6.97, FDR p = 0.00700)
      - Mitochondrial respiratory chain complex I	(fold enrichment = 7.67, FDR p = 0.0000592)
      - Peroxisomal matrix	(fold enrichment = 5.07, FDR p = 0.0160)
      - Peroxisomal membrane	(fold enrichment = 4.22, FDR p = 0.0456)
      - Mitochondrial matrix	(fold enrichment = 3.05, FDR p = 0.000000178)
      - Extracellular exosome	(fold enrichment = 1.74, FDR p = 0.00000467)
      - Cytosol	(fold enrichment = 1.38, FDR p = 0.0000277)
        - The fact that these genes were also enriched for presence in the cytosol is further link to both neuromelanin formation and dopamine oxidation
      - By far the most interesting result was the enrichment of the melansome (fold enrichment = 3.95, FDR p = 0.00593)
        - Pigment granule	(fold enrichment = 3.95, FDR p = 0.00578)
          - Vesicle	(fold enrichment = 1.38, FDR p = 0.00164)
        - This is directly connected to neuromelanin formation
        - 12 genes were found to be part of the melanosome and pigment granule GO terms: OCA2, CD63, HSP90AA1, GNA13, PRDX1, CCT4, RAB9A, DTNBP1, HSP90B1, SLC1A4, YWHAZ, NMB
          - The results indicate that the downregulation of these 12 genes contribute to the age-dependent accumulation of neuromelanin
          
STRING DB:

  ACGs: 69 proteins could be mapped to the inputted 77 ACGs
    - Set to medium confidence (0.400)
    - RPS2 and RPS20 appeared to be the center of protein hubs, having connections to five other genes, both of which are highly associated with each other (interaction score = 0.999)
    - Overall, the vast majority of proteins appeared to have no connections to each other, save for a few such as COX7C, COX4I1, UQCR11, NDUFS3, and PTPMT1, which interact with each other and are all relate to mitochondrial respiratory or electron transport chains, as COX7C, COX4I1, are subunits of cytochrome c
    - No notable enrichments of GO terms different from those already determined from PANTHER DB analysis
    
  Positively correlated genes: 97 proteins could be mapped to the inputted 123 genes
    - Set to medium confidence (0.400) 
    - SNAI2 and MECOM have a very strong interaction score
      - MECOM interacts with MYH11, which is another hub center connecting to MYOCD, TAGLN, MYL9, TBX18, and PDGFRB
      - SNAI2 interacts with NOTCH4 and NOTCH3, which have a high interaction score with each other, as well as some other weaker interactions with DAXX and CAV1
    - Overall, the vast majority of proteins appeared to have no connections to each other
    - No notable enrichments of GO terms different from those already determined from PANTHER DB analysis
    
  Negatively correlated genes: 606 proteins could be mapped to the inputted 634 genes
    - This network was incredibly large, so only edges with the highest confidence (0.900) were shown to better observe the clusters
      - A number of clusters of proteins, all with highest confidence interactions with each other, were observed:
        - One cluster around the NDUF proteins (NDUFA1, NDUFA13, NDUFA6, NDUFA8, NDUFB10, NDUFB2, NDUFB6, NDUFB7, NDUFB8, NDUFS3, NDUFS6), that includes connections to COX7C and COX4I1
          - Cluster revolves around dehydrogenases; NDUF proteins are NADH dehydrogenase (ubiquinone) subunits, connected to subunits of cytochrome c oxidase (COX7C, COX4I1, UQCR11, COXA6) as well as subunits of succinate dehydrogenase cytochrome b (SDHD, SDHC)
        - One cluster around the MRPL proteins (mitochondrial ribosomal proteins)
        - One cluster around RPS2, RPS20, RPS3, and RPLP2, that also has connections to the MRPL cluster (ribosomal proteins, connected to many transcription factors)
        - One very closely knit cluster including PRPF, CD2B, SNRPD3, PUF60, GPKOW, SNRNP27, LSM4, LSM8, LSM5 and HNRNPH2 (transcription factors or parts of the spliceosome)
        - TRIM28 had interactions with all the zinc finger proteins in this set, none of which had any interactions with each other
        - EGLN3, UBB, PSMC3, PSMA7, and PSMB7 were closely knit, and appeared to be the center of another protein hub
        - One relatively small protein cluster around the ALDH proteins (ALDH2, ALDH3A2, ALDH61, ALDH9A1, ALDH1A3, subunits of aldehyde dehydrogenase)
        - One cluster around HIST (HIST1H3D, HIST1H3J, HIST1H4B, HIST1H4D, HIST2H2AB, and HIST4H4) and FBX proteins
        - CCT4 appeared to be a hub center, connecting to the HIST/FBX cluster, the RPS cluster, and numerous other proteins
          - CCT4 is T-complex protein 1 (called CCT) subunit delta
          - Associated with sensory peripheral neuropathy and hereditary sensory neuropathy
          - Substrates of CCT include the cytoskeletal proteins actin and tubulin
      - A large number of the proteins (at least half) appeared to have no connections to each other
      - Using highest confidence, none of the melanosome- or pigment granule-related proteins showed any level of interactions with each other
      - No notable enrichments of GO terms different from those already determined from PANTHER DB analysis, other than enrichment for glial cell proliferation (SOX11, NF2, EEF2, IL33)
    
  Considering the significance of the melanosome/pigment granule GO term and the 12 genes enriched for it, these 12 genes were inputted into STRING DB as well:
    - RAB9A, GNA13, SLC1A4, and NMB showed no connections with any other proteins
    - OCA2 and DTNBP1 had an edge with medium confidence (0.400)
    - HSP90AA1 appeared to be a network hub, having edges of medium confidence with PRDX1, CD63, and YWHAZ, and edges of highest confidence (0.900) with CCT4 and HSP90B1
      - HSP90AA1 may be a reliable protein/gene target for preventing high levels of neuromelanin accumulation, and therefore preventing the development of Parkinson's disease in older individuals
    - GO term enrichment anlaysis provided by STRING DB verified that these 12 genes are highly involved in the melanosome
      - Enriched for pigment cell differentation (OCA2, CD63) and cellular pigmentation (CD63, DTNBP1) biological processes, and the melanosome (all except NMB) and melanosome membrane (OCA2, DTNBP1) cellular components
    
  The clusters of transcription factors, particularly for negatively correlated genes such as TRIM28, as well as histone modifying proteins like HIST1H3D, HIST1H3J, HIST1H4B, HIST1H4D, HIST2H2AB, and HIST4H4, suggests that an epigenomic analysis the changes to the genome with age could reveal interesting changes to genes regulated by those transcription factors. Additionally, HSP90AA1 may be a reliable protein target for preventing high levels of neuromelanin accumulation, and therefore preventing the development of Parkinson's disease in older individuals, because of its interconnectedness to other melanosome/pigment granule-related proteins that are downregulated with age.

____________________________________________________________________________________________
____________________________________________________________________________________________

AGE CORRELATIONS - NON-NEUROLOGICAL INDIVIDUALS EXON-LEVEL ANALYSIS
____________________________________________________________________________________________
____________________________________________________________________________________________

The following code was written to allow for an exon-level correlation analysis of gene expression with age. It can only be run if GSE46706 can be loaded into RStudio as the list called gset without memory exhaustion as written in the first code chunk of this notebook.

August 8, 2020

  Exon microarray data from GSE46706 (GPL517) was read into a numeric matrix (exon), with columns as samples and rows as exon probes. Samples were lined with the age of the donor from which the sample was obtained and the brain region of the sample, to form the data frame exon_filt. Each exon was then aligned to a specific transcript cluster ID within the exon_filt data frame. This would allow for filtering the exons based on their corresponding gene, as each transcript cluster ID corresponded to one gene (this means multiple exons, or columns, in exon_filt could have the same transcript cluster ID).
  
  exon_filt was then filtered for expression data in the substantia nigra, the output of which was stored in the data frame exon_filt_SN. As rbind() and cbind() had been used to add brain regions, age of patient, transcript cluster IDs, and the type of exon to exon_filt, the expression values of exon_filt were coerced into the character data. Because exon_filt was filtered to make exon_filt_SN, the expression values of exon_filt_SN were also character values. The expression values in exon_filt_SN were converted to numeric and stored in a new data frame called exon_filt_SN_numeric. In the end, both exon_filt_SN and exon_filt_SN_numeric contained the rows as different samples and columns as different exons. They both also contained a row at the bottom that contained the transcript cluster IDs for each column (i.e., exon), and exon_filt_SN had two columns at the end, one indicating the brain region of each sample (which was all substantia nigra), and another indicating the age of the patient sample.
  
  Age, as aligned with the substantia nigra samples in exon_filt_SN, was extracted as its own numeric vector age_exon to use for computing correlations using the cor() and cor.test() functions.

```{r, echo=FALSE}
# library(GEOquery)
# library(dplyr)
# library(stringr)
# 
# # to avoid scientific notation
# options(scipen = 999)
# 
# # Exon expression microarray data. Exon are listed using numeric transcript cluster IDs.
# exon <- gset[["GSE46706-GPL5188_series_matrix.txt.gz"]]@assayData[["exprs"]]
# # Gives age of donor and brain region for each sample
# exon_sample_annotations <- gset[["GSE46706-GPL5188_series_matrix.txt.gz"]]@phenoData@data
# # Shows names of genes that correspond to each numeric ID
# exon_names <- gset[["GSE46706-GPL5188_series_matrix.txt.gz"]]@featureData@data
# 
# # Add transcript cluster IDs, sample ages, sample brain regions
# exon_filt <- exon
# # Add sample ages and brain regions as two extra rows at bottom of exon_filt
# exon_filt <- exon_filt %>% rbind(exon_sample_annotations[,11], exon_sample_annotations[,46])
# # Add exon transcript cluster IDs as extra column at end of exon_filt
# exon_filt <- exon_filt %>% cbind(exon_names[,10], exon_names[,15]) #provides a warning that doesn't matter. exon_names[,15] adds label of core/extended/full exons 
# colnames(exon_filt)[1232] <- 'Transcript Cluster ID'
# # Put 'brain region: substantia nigra' in Transcript Cluster ID column so that it includes the transcript cluster ID row after it is filtered
# exon_filt[291705,1232] <- 'brain region: substantia nigra'
# exon_filt[291706,1232] <- NA
# 
# 
# # Filter exon_filt for substantia nigra samples only
# exon_filt <- data.frame(t(exon_filt)) #rows as samples, columns as exons, to filter for samples in the substantia nigra
# exon_filt_SN <- exon_filt %>% filter(exon_filt[,291705] == 'brain region: substantia nigra')
# 
# # Change column labels for exon_filt to avoid confusion in future reference
# colnames(exon_filt)[291706] = 'age'
# colnames(exon_filt)[291705] = 'brain_region'
# row.names(exon_filt)[1233] = 'exon_type'
# 
# # At these two positions, the exon type repeats
# exon_filt[1233,291705] <- NA
# exon_filt[1233,291706] <- NA
# 
# # Extract age as its own numeric vector
# age_exon <- exon_filt_SN[,291706]
# age_exon <- as.numeric(age_exon[-102])
# 
# # Convert all applicable values in exon_filt_SN to numeric using convert_to_numeric function that was previously defined
# exon_filt_SN_numeric <- convert_to_numeric(exon_filt_SN[,c(-291705,-291706)])
# 
# # Print first 100 exons
# head(exon_filt_SN_numeric[,1:100])

```

  Before downstream analysis could be performed, the exploratory analysis needed to be performed to determine whether normalization would be necessary or not. A boxplot of the exon expression values in each of the 101 samples revealed that the data was relatively median-centered, indicating that quantile normalization was not necessary. A comparison of this boxplot and a boxplot with the quantile normalized exon expression values also supports this claim.
```{r, echo=FALSE}
# library(preprocessCore)
# 
# # Data does appear median centered
# boxplot(t(exon_filt_SN_numeric[-102,]))
# 
# # Quantile normalizing the exon data and making a boxplot for comparison with original data
# boxplot(normalize.quantiles(as.matrix(t(exon_filt_SN_numeric[-102,]))))

```
  Additionally, looking at a histogram of exon expression, it can be observed that exon expression across all samples is roughly symmetric and therefore approximately normal, indicating that Pearson's correlations can be applied.
  
```{r}
# library(dplyr)
# 
# exon_filt_SN_numeric[-102,] %>% 
#   hist(main = 'Exon Expression', xlab = 'Exon Expression')

```
  
  The exploratory analysis did reveal that there were many more high outliers than low outliers, which means that exons are generally more likely to be overexpressed than underexpressed in the substantia nigra. This trend was even more prevalent than in the gene expression data, as the exon expression data had almost no low outliers.

  At this point, the data was ready for downstream analysis. This would consist of computing correlations between exon expression and age of the sample donor.

____________________________________________________________________________________________

August 9, 2020

  Correlations analysis tests were performed for all target genes, computing the correlation between exon expression and the age of the sample donor. The format was as follows:

  - Correlation Coefficient: cor(normalized exon expression data, age, method = 'pearson')
  - P-value of correlation: cor.test(normalized exon expression data, age, method = 'pearson')[['p.value']]

  For this experiment, the significance level was set to α = 0.05. Pearson's correlations were performed for all exons.
  
  For each target gene, analysis results were organized into a matrix, with correlation coefficients and raw p-values as columns and the different exons as rows.
  
  Finally, the matrices for each gene were all compiled into a list called all_exon_correlations.

```{r, echo=FALSE}
# library(dplyr)
# 
# # Returns matrix with raw p-value of correlation and correlation coefficient (columns) for each exon (rows) for each target gene.
# compute_correlation_exon <- function(Cluster_ID) {
#   
#   # Filter exon_filt_SN_numeric for all exons having the belonging to the same gene
#   Exon <- exon_filt_SN_numeric %>% t() %>% data.frame()
#   Exon <- Exon %>% filter(Exon[,102] == Cluster_ID)
#   Exon <- Exon[,-102]
#   
#   # Computing correlations
#   Exon <- data.frame(t(Exon))
#   cor_Exon <- matrix(data = NA, nrow = dim(Exon)[2], ncol = 2)
#   colnames(cor_Exon) <- c('Correlation Coefficient', 'Raw p-value')
#   rownames(cor_Exon) <- colnames(Exon)
#   for (i in 1:dim(Exon)[2]) {
#     cor_Exon[i,'Correlation Coefficient'] <- cor(as.numeric(Exon[,i]), 
#                                                  age_exon, 
#                                                  method='pearson')
#     cor_Exon[i,'Raw p-value'] <- cor.test(as.numeric(Exon[,i]), 
#                                           age_exon, 
#                                           method='pearson')[["p.value"]]
#   }
#   return(cor_Exon)
#   
# }
# 
# target_exon_correlations <- lapply(c(3359180, 3050388, 3266279, 2845921, 2350952, 3696666, 4055661, 4006210, 3937092), compute_correlation_exon)

```

____________________________________________________________________________________________

August 12, 2020

  To minimize the false discovery rate of significant correlations, the Benjamini-Hochberg procedure was applied to all the p-values using the p.adjust() function. First, p-values of correlation between age and all the exons, not just the exons of the target genes, had to be computed. Then the adjusted p-values were filtered again for those of the exons of the target genes. 
  
  This would also calculate correlations of all exons with age. These correlations were both filtered for the target genes and for ACGs (age-correlated genes) that were previously determined from the gene-level analysis.

```{r, echo=FALSE}
# library(dplyr)
# 
# # Computing all raw p values of correlation, along with correlation coefficients for each exon (row)
# all_exon_cor <- matrix(data = NA, nrow = dim(exon_filt_SN_numeric[-102,])[2], ncol = 5)
# colnames(all_exon_cor) <- c('Correlation Coefficient', 'Raw p-value', 'BH-corrected p-value', 'Transcript Cluster ID', 'Gene Assignment')
# for (i in 1:dim(exon_filt_SN_numeric[-102,])[2]) {
#   all_exon_cor[i, 'Correlation Coefficient'] <- cor(as.numeric(exon_filt_SN_numeric[-102,i]), 
#                                                     age_exon, 
#                                                     method='pearson')
#   all_exon_cor[i, 'Raw p-value'] <- cor.test(as.numeric(exon_filt_SN_numeric[-102,i]), 
#                                              age_exon, 
#                                              method='pearson')[["p.value"]]
# }
# rownames(all_exon_cor) <- colnames(exon_filt_SN_numeric[-102,])
# 
# # Adding adjusted p-values
# all_exon_cor[, 'BH-corrected p-value'] <- p.adjust(all_exon_cor[, 'Raw p-value'], 
#                                                    method='BH')
# 
# # Adding transcript cluster IDs and gene assignments to allow for filtering
# all_exon_cor[, 'Transcript Cluster ID'] <- exon_filt_SN_numeric[102,]
# all_exon_cor[, 'Gene Assignment'] <- exon_names[,"gene_assignment"]
# 
# # Filtering for exons of target genes
# target_cluster_IDs <- c(3359180, 3050388, 3266279, 2845921, 2350952, 3696666, 4055661, 4006210, 3937092)
# target_exon_corr <- all_exon_cor %>% 
#   data.frame %>% 
#   filter(all_exon_cor[, 'Transcript Cluster ID'] %in% target_cluster_IDs)
# print(paste(sum(target_exon_corr$BH.corrected.p.value < 0.05), 
#             "target gene exons out of the 9 target genes that were significantly correlated with age"))

```

  Out of all the target genes, NQO1 had five exons that were significantly correlated with age (BH-corrected p-value < 0.05), all of which had negative correlation coefficients. This was quite an interesting result considering that the gene-level analysis revealed that NQO1 was not significantly correlated with age. This supports the concept that performing an exon-level analysis may be advantageous as it may provide results that would be masked by simply doing gene-level analysis.

```{r}
# library(dplyr)
# 
# # Filtering for ACGs determined by gene-level analysis
# ACG_exon_corr <- all_exon_cor %>% 
#   data.frame %>% 
#   filter(all_exon_cor[, 'Transcript Cluster ID'] %in% ACGs[, "ID"])
# print(paste(sum(ACG_exon_corr$BH.corrected.p.value < 0.05), 
#             "ACG exons out of the", 
#             length(ACG_exon_corr$BH.corrected.p.value), 
#             "exons from ACGs that were significantly correlated with age"))
# 
# # Filtering for all exons with BH-corrected p-value < 0.05
# sig_corr_exon <- all_exon_cor %>% 
#   data.frame %>% 
#   filter(all_exon_cor[, 'BH-corrected p-value'] < 0.05)
# print(paste("Out of the", 
#             dim(sig_corr_exon)[1], 
#             "significantly correlated exons,", 
#             sum(sig_corr_exon$Transcript.Cluster.ID %in% ACGs[, "ID"]), 
#             "were from ACGs"))

```

  Interestingly, less than half of the exons from ACGs were significantly correlated with age. Additionally, only 4.09% of all exons that were significantly correlated were from ACGs, which indicates that the gene-level analysis left out almost 96% of all exons that were significantly correlated with age. Although this exon-level analysis did reveal interesting information about the age-correlated genes and the target genes that the gene-level analysis did not, this data cannot be used for identifying exons that contribute to neuromelanin formation as the Allen Human Brain Atlas and the other GEO datasets that are to be used in this study only provide a gene-level analysis.

____________________________________________________________________________________________
____________________________________________________________________________________________

AGE CORRELATIONS - VALIDATION IN PD PATIENTS
____________________________________________________________________________________________
____________________________________________________________________________________________

August 20, 2020

  GSE8397 and GSE20292 were loaded into RStudio using getGEO() function from GEOquery package.
  
```{r, echo=FALSE}
library(GEOquery)

GSE8397 <- getGEO("GSE8397", GSEMatrix = TRUE)
GSE20292 <- getGEO("GSE20292", GSEMatrix = TRUE)

```

  The datasets were explored. The samples in each dataset were filtered for only Parkinson's patients and expression in the substantia nigra (sample annotations), and the probes in each dataset were labeled with their probe IDs. Additionally, since there were multiple probes in this data set, each probe needed to be aligned to a gene. The following procedure was used for aligning probes to genes using the collapseRows function from the WGCNA package [15]:
  
  - If there is only one probe for a gene, that one is chosen
  - If there are two probes for one gene, the one with the max variance across all samples (both control and PD) is chosen (method = maxRowVariance)
  - If there more than two probes for one gene, the probe with the highest connectivity (summed adjacency) is chosen (connectivityBasedCollapsing = TRUE)


```{r, echo=FALSE}
library(tidyverse)
library(GEOquery)
library(stringr)
library(WGCNA)

########### EXPLORING DATA ###########

# GSE8397 info
GSE8397_exp_gene <- GSE8397[["GSE8397-GPL96_series_matrix.txt.gz"]]@assayData[["exprs"]]
GSE8397_sample_annotations <- GSE8397[["GSE8397-GPL96_series_matrix.txt.gz"]]@phenoData@data
GSE8397_gene_IDs <- GSE8397[["GSE8397-GPL96_series_matrix.txt.gz"]]@featureData@data
# head(GSE8397_sample_annotations)
# head(GSE8397_gene_IDs)


# GSE20292 info
GSE20292_exp_gene <- GSE20292[["GSE20292_series_matrix.txt.gz"]]@assayData[["exprs"]]
GSE20292_sample_annotations <- GSE20292[["GSE20292_series_matrix.txt.gz"]]@phenoData@data
GSE20292_gene_IDs <- GSE20292[["GSE20292_series_matrix.txt.gz"]]@featureData@data
# head(GSE20292_sample_annotations)
# head(GSE20292_gene_IDs)


########### PREPARING DATA ###########

# GSE8397

# Convert gene expression table to data frame
GSE8397_exp_gene <- data.frame(GSE8397_exp_gene)

# Add column containing gene symbols to gene expression data frame
for (i in 1:dim(GSE8397_exp_gene)[1]) {
  GSE8397_exp_gene[i, "Gene"] <- strsplit(GSE8397_gene_IDs[i,"Gene Symbol"], " ")[[1]][1]
}

# Remove rows (probes) without a gene assignment
print(paste(sum(is.na(GSE8397_exp_gene$Gene)), "probes without gene assignments in GSE8397"))
GSE8397_exp_gene <- GSE8397_exp_gene %>% filter(is.na(GSE8397_exp_gene$Gene) == FALSE)

# Align probes to genes
probe2gene_GSE8397 <- collapseRows(GSE8397_exp_gene[, -48], 
                           rowGroup = GSE8397_exp_gene$Gene, 
                           rowID = rownames(GSE8397_exp_gene),
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
GSE8397_exp_gene <- probe2gene_GSE8397[["datETcollapsed"]]
print(paste(dim(GSE8397_exp_gene)[1], "genes were aligned to 22283 probes in GSE8397"))

# Add columns that for type of sample and age to gene expression data frame
GSE8397_exp_gene <- data.frame(GSE8397_exp_gene)
GSE8397_exp_gene <- add_row(GSE8397_exp_gene, .before = 1)
GSE8397_exp_gene <- add_row(GSE8397_exp_gene, .before = 1)
GSE8397_exp_gene[1,] <- GSE8397_sample_annotations[,"title"]
GSE8397_exp_gene[2,] <- as.numeric(str_sub(GSE8397_sample_annotations[,"age:ch1"], 1, 2))
rownames(GSE8397_exp_gene)[1:2] <- c("sample_type", "age")

# Extract age as its own vector
GSE8397_exp_gene <- GSE8397_exp_gene %>% t %>% data.frame
GSE8397_age_gene <- as.numeric(GSE8397_exp_gene$age)


# GSE20292
 
# Convert gene expression table to data frame
GSE20292_exp_gene <- data.frame(GSE20292_exp_gene)

# Add column containing gene symbols to gene expression data frame
for (i in 1:dim(GSE20292_exp_gene)[1]) {
  GSE20292_exp_gene[i, "Gene"] <- strsplit(GSE20292_gene_IDs[i,"Gene Symbol"], " ")[[1]][1]
}

# Remove rows (probes) without a gene assignment
print(paste(sum(is.na(GSE20292_exp_gene$Gene)), "probes without gene assignments in GSE202972"))
GSE20292_exp_gene <- GSE20292_exp_gene %>% filter(is.na(GSE20292_exp_gene$Gene) == FALSE)

# Align probes to genes
probe2gene_GSE20292 <- collapseRows(GSE20292_exp_gene[,-30], 
                           rowGroup = GSE20292_exp_gene$Gene, 
                           rowID = rownames(GSE20292_exp_gene),
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
GSE20292_exp_gene <- probe2gene_GSE20292[["datETcollapsed"]]
print(paste(dim(GSE20292_exp_gene)[1], "genes were aligned to 22283 probes in GSE20292"))

# Add columns that for type of sample and age to gene expression data frame
GSE20292_exp_gene <- data.frame(GSE20292_exp_gene)
GSE20292_exp_gene <- add_row(GSE20292_exp_gene, .before = 1)
GSE20292_exp_gene <- add_row(GSE20292_exp_gene, .before = 1)
GSE20292_exp_gene[1,] <- GSE20292_sample_annotations[,"disease state:ch1"]
GSE20292_exp_gene[2,] <- as.numeric(GSE20292_sample_annotations[,"age:ch1"])
rownames(GSE20292_exp_gene)[1:2] <- c("sample_type", "age")

# Extract age as its own vector
GSE20292_exp_gene <- GSE20292_exp_gene %>% t %>% data.frame
GSE20292_age_gene <- as.numeric(GSE20292_exp_gene$age)


########### FILTERING DATA ###########

# GSE8397

# Filter for expression in substantia nigra, for Parkinson's patients only
GSE8397_exp_gene_SN_PD <- GSE8397_exp_gene[c(15:23, 32:46), ]

# Extract age for PD patients in substantia nigra as its own numeric vector
GSE8397_age_gene_SN_PD <- as.numeric(GSE8397_exp_gene_SN_PD$age)

# Filter for expression only in 9 target genes
GSE8397_exp_gene_SN_PD <- add_row(data.frame(GSE8397_exp_gene_SN_PD), .before = 1)
GSE8397_exp_gene_SN_PD[1,] <- colnames(GSE8397_exp_gene_SN_PD)
# GSE8397_exp_gene_SN_PD_target <- GSE8397_exp_gene_SN_PD %>% 
#   t %>% 
#   data.frame %>% 
#   filter(GSE8397_exp_gene_SN_PD[1,] =='GSTM2' |
#          GSE8397_exp_gene_SN_PD[1,]=='NQO1' |
#          GSE8397_exp_gene_SN_PD[1,] =='TH' |
#          GSE8397_exp_gene_SN_PD[1,] =='SLC18A2' |
#          GSE8397_exp_gene_SN_PD[1,] =='SLC6A3' |
#          GSE8397_exp_gene_SN_PD[1,] =='DDC'|
#          GSE8397_exp_gene_SN_PD[1,] =='COMT' |
#          GSE8397_exp_gene_SN_PD[1,] =="MAOB" |
#          GSE8397_exp_gene_SN_PD[1,] =="MAOA" |
#          GSE8397_exp_gene_SN_PD[1,] =="sample_type" |
#          GSE8397_exp_gene_SN_PD[1,] =="age")
GSE8397_exp_gene_SN_PD <- GSE8397_exp_gene_SN_PD[-1,]
# GSE8397_exp_gene_SN_PD_target <- GSE8397_exp_gene_SN_PD_target[,-1] %>% t


# GSE20292

# Filter for expression in substantia nigra, for Parkinson's patients only
GSE20292_exp_gene_SN_PD <- GSE20292_exp_gene %>% filter(sample_type == "Parkinsons disease")

# Extract age for PD patients in substantia nigra as its own numeric vector
GSE20292_age_gene_SN_PD <- as.numeric(GSE20292_exp_gene_SN_PD$age)

# Filter for expression only in 9 target genes
GSE20292_exp_gene_SN_PD <- add_row(data.frame(GSE20292_exp_gene_SN_PD), .before = 1)
GSE20292_exp_gene_SN_PD[1,] <- colnames(GSE20292_exp_gene_SN_PD)
# GSE20292_exp_gene_SN_PD_target <- GSE20292_exp_gene_SN_PD %>% 
#   t %>% 
#   data.frame %>% 
#   filter_all(any_vars(.=='GSTM2'|
#                       .=='NQO1'|
#                       .=='TH'|
#                       .=='SLC18A2'|
#                       .=='SLC6A3'|
#                       .=='DDC'|
#                       .=='COMT'|
#                       .=='MAOB'|
#                       .=='MAOA'|
#                       .=="sample_type"|
#                       .=="age"))
GSE20292_exp_gene_SN_PD <- GSE20292_exp_gene_SN_PD[-1,]
# GSE20292_exp_gene_SN_PD_target <- GSE20292_exp_gene_SN_PD_target[,-1] %>% t

```

____________________________________________________________________________________________

August 22, 2020

  During the filtering of the data frames, the expression values had been converted to the character data type. In order to perform analyses, they needed to be converted back to numeric. In order to do this, the convert_to_numeric function was applied to the GSE8397_exp_gene_SN_PD_target, GSE8397_exp_gene_SN_PD, GSE20292_exp_gene_SN_PD_target, and GSE20292_exp_gene_SN_PD data frames. The age and sample_type columns had to be removed first.
  
  The new function returned a data frame containing only the expression values for each gene without age or sample type labels. This was not a reason for concern since the age feature of each sample had been extracted previously.

```{r}

GSE8397_exp_gene_SN_PD <- convert_to_numeric(GSE8397_exp_gene_SN_PD[,-c(1:2)])
# GSE8397_exp_gene_SN_PD_target <- convert_to_numeric(GSE8397_exp_gene_SN_PD_target[,-c(1:2)])

GSE20292_exp_gene_SN_PD <- convert_to_numeric(GSE20292_exp_gene_SN_PD[,-c(1:2)])
# GSE20292_exp_gene_SN_PD_target <- convert_to_numeric(GSE20292_exp_gene_SN_PD_target[,-c(1:2)])


```

____________________________________________________________________________________________

August 25, 2020

  Both expression data frames had 22283 rows, or 22283 different genes. In order to ensure that all gene expression data will be included when these data frames are joined, it needed to be confirmed that all genes were the same. The intersect() function was used to determine this. If the two data frames had expression on the exact same genes, the conditional expression will return "TRUE".

```{r}
library(dplyr)

# Confirmation that the two data frames (for GSE8397 and GSE20292) have expression data on all the same genes
dim(GSE8397_exp_gene_SN_PD)[2] == length(intersect(colnames(GSE8397_exp_gene_SN_PD), colnames(GSE20292_exp_gene_SN_PD)))

# Confirmation that the two data frames (for GSE8397 and GSE20292) have expression data on all the same target genes
# dim(GSE8397_exp_gene_SN_PD_target)[2] == length(intersect(colnames(GSE8397_exp_gene_SN_PD_target), colnames(GSE20292_exp_gene_SN_PD_target)))

```

  After it was confirmed that the GSE20292_exp_gene_SN_PD and GSE8397_exp_gene_SN_PD contained expression data on all the same genes, they could be joined into one data frame.

```{r}

# Joining expression data from both data frames
exp_gene_SN_PD <- rbind.data.frame(GSE8397_exp_gene_SN_PD, GSE20292_exp_gene_SN_PD)
# exp_gene_SN_PD_target <-  rbind.data.frame(GSE8397_exp_gene_SN_PD_target, GSE20292_exp_gene_SN_PD_target)

# Combining age numeric vectors
age_gene_SN_PD <- c(GSE8397_age_gene_SN_PD, GSE20292_age_gene_SN_PD)

```

____________________________________________________________________________________________

August 29, 2020

  Now that the gene expression data frames have been joined, exploratory analysis needs to be performed on the data to ensure that the joined datasets can be analyzed together.
  
```{r}

boxplot(t(exp_gene_SN_PD))

```
  
  The boxplot revealed that the samples from GSE8397 gave systemically lower expression values than the samples from GSE20292. Therefore, to allow for joint analysis, the joint data frame was quantile normalized.
  
```{r}
library(preprocessCore)

exp_gene_SN_PD <- normalize.quantiles(as.matrix(t(exp_gene_SN_PD)))

# exp_gene_SN_PD_target <- normalize.quantiles(as.matrix(t(exp_gene_SN_PD_target)))

boxplot(exp_gene_SN_PD)
# boxplot(exp_gene_SN_PD_target)


```
  
____________________________________________________________________________________________

September 3, 2020

  Upon reading further literature, it was determined that it would be best not to analyze the two data sets together, as quantile normalization across two completely different data sets could affect some of the information. Therefore, the joint data frames were removed and all analyses will proceed for each data sets individually.
  
```{r}

rm(exp_gene_SN_PD)
# rm(exp_gene_SN_PD_target)
rm(age_gene_SN_PD)

```

____________________________________________________________________________________________

September 4, 2020

```{r}

# Transpose data frames containing gene expression for each sample to have genes as rows and samples as columns
GSE8397_exp_gene_SN_PD <- t(GSE8397_exp_gene_SN_PD)
# GSE8397_exp_gene_SN_PD_target <- t(GSE8397_exp_gene_SN_PD_target)
GSE20292_exp_gene_SN_PD <- t(GSE20292_exp_gene_SN_PD)
# GSE20292_exp_gene_SN_PD_target <- t(GSE20292_exp_gene_SN_PD_target)

# Compute age correlations with BH-correction of p-values

compute_all_age_correlations <- function(exp, age_vector) {
  
  all_cor <- matrix(nrow = dim(exp)[1], ncol = 4) %>% data.frame
  colnames(all_cor) <- c("Gene", "Correlation Coefficient", "Raw p-value", "BH-corrected p-value")
  rownames(all_cor) <- rownames(exp)
  all_cor[,"Gene"] <- rownames(exp)

  for (i in 1:dim(exp)[1]) {
    cor <- cor.test(as.numeric(exp[i, ]), 
                    age_vector, 
                    method = "pearson")
    all_cor[i, "Correlation Coefficient"] <- cor[["estimate"]][["cor"]]
    all_cor[i, "Raw p-value"] <- cor[["p.value"]]
  }
  
  all_cor[, "BH-corrected p-value"] <- p.adjust(all_cor[, "Raw p-value"], method = "BH")
  return(all_cor)
  
}

GSE8397_all_cor <- compute_all_age_correlations(GSE8397_exp_gene_SN_PD, GSE8397_age_gene_SN_PD)
# GSE8397_target_cor <- compute_all_age_correlations(GSE8397_exp_gene_SN_PD_target, GSE8397_age_gene_SN_PD)
GSE20292_all_cor <- compute_all_age_correlations(GSE20292_exp_gene_SN_PD, GSE20292_age_gene_SN_PD)
# GSE20292_target_cor <- compute_all_age_correlations(GSE20292_exp_gene_SN_PD_target, GSE20292_age_gene_SN_PD)

print(paste(sum(GSE8397_all_cor[, "BH-corrected p-value"] < 0.05), "genes that were significantly correlated with age in PD patients from GSE8397"))
print(paste(sum(GSE20292_all_cor[, "BH-corrected p-value"] < 0.05), "genes that were significantly correlated with age in PD patients from GSE20292"))


```
  
  Evidently, there were no genes that were observed to be significantly correlated with age (BH-corrected p-value > 0.05). This was most likely due to the sample sizes. GSE8397 contained only 24 samples from PD patients, and GSE20292 contained just 11 samples. Additionally, PD  samples come from generally older patients anyway. This can be demonstrated by looking at boxplots of the ages of patients from UKBEC, ages of PD patients from GSE8397, and ages of PD patients from GSE20292.
  
```{r}

boxplot(age_gene_SN, GSE8397_age_gene_SN_PD, GSE20292_age_gene_SN_PD, ylab = "Age")

# t-test of UKBEC age data and GSE8397 PD age data
t.test(age_gene_SN, GSE8397_age_gene_SN_PD)

# t-test of UKBEC age data and GSE20292 PD age data
t.test(age_gene_SN, GSE20292_age_gene_SN_PD)

```
  Boxplot 1 corresponds to UKBEC age data, boxplot 2 corresponds to GSE8397 PD age data, and boxplot 3 corresponds to GSE20292 PD age data. Evidently, the PD age data is systematically larger than the data from UKBEC, which contains non-neurological individuals. Both the GSE8397 PD age data and the GSE20292 PD age data were very significantly different from the UKBEC age data, as noted from the p-values of the t-tests being far less than 0.05 (1.615e-15 and 3.617e-7 respectively). This skewedness of PD sample ages makes it very difficult to perform any reasonable age correlations. Using other data sets with age data distributions that match the UKBEC distribution more closely may be a viable option for performing age correlation validation analysis in PD patients. Otherwise, the results from performing age correlations with GSE8397 and GSE20292 are not valid, and therefore the current list of ACGs obtained from UKBEC correlation analysis is not altered. 

____________________________________________________________________________________________
____________________________________________________________________________________________

mDBR
____________________________________________________________________________________________
____________________________________________________________________________________________

September 23, 2020

  The formation of neuromelanin in the brain is a phenomenon that is known to be unique to humans [2], as animals do not accumulate the pigment. It is observed to form only in certain catecholaminergic neurons, namely the dopaminergic neurons of the substantia nigra pars compacta (SNpc) [1]. Interestingly, neuromelanin is closely linked to Parkinson's disease, as it is these dopaminergic neuromelanin-containing neurons that are exclusively lost in Parkinson's [1].

  Currently, it is assumed that neuromelanin formed in dopaminergic neurons is a product of dopamine oxidation, a process not known to be catalyzed by any particular enzymes [4-6]. However, studies in animal models have failed to produce neuromelanin by solely increasing dopamine levels and dopamine oxidation [2]. Additionally, neuromelanin does not form in all catecholaminergic neurons. In fact, the dopaminergic neurons of the ventral tegmental area (VTA) generally lack neuromelanin [1], suggesting that some other molecular mechanism may be involved in neuromelanin formation.

  The goal of this section, called mDBR (melanized dopaminergic brain region analysis), is to use gene expression data to give some insight into the formation of neuromelanin. Learning more about these molecular pathways may even provide some insight into the mechanisms that cause the selective death of neuromelanin-containing neurons in Parkinson's disease. The primary method by which this is achieved is by analyzing differences in gene expression between the SNpc, a brain region containing a large population of melanized dopaminergic neurons, and the VTA, a brain region containing a large population of non-melanized dopaminergic neurons.

  The Allen Human Brain Atlas (https://human.brain-map.org/microarray/search) provides microarray gene expression data of numerous different brain regions in 6 neurologically and neuropathologically normal individuals, including the SNpc and VTA. It also provides a function on its website that does differential expression analysis of different brain regions, outputting p-values and fold change values for each probe. This differential expression analysis uses a 2-sample t-test followed by Benjamini-Hochberg (BH) false discorvery rate correction.However, it only outputs the fold change for probes that are upregulated in the target structure with respect to the contrast structure. Therefore, this search had to be done twice: once with the SNpc as the target structure and the VTA as the contrast structure, and another time with the VTA as the target structure and the SNpc as the contrast structure. This data could be directly downloaded from the site. However, it could only be downloaded 2000 probes at a time. This was rather tedious, and knowing that many probes are not significantly differentially expressed anyway, only the first 10000 probes with the highest fold change were downloaded, giving a total of 20000 probes (10000 that are upregulated in the SNpc with respect to the VTA, and 10000 that are upregulated in the VTA with respect to the SNpc).

  First, the differential expression data of the top 20000 probes was loaded into RStudio. This data included fold change values and p-values for each individual probe.
```{r, echo=FALSE}
library(readr)
library(dplyr)


# SNpc target, VTA contrast
files_1 <- list.files(path = "/Users/rezaadibnia/Desktop/PD NM Project/mDBR - VTA vs SNpc/SNpc target, VTA contrast/", pattern = "Probes_.*csv", full.names = TRUE, ignore.case = FALSE)
SNt_VTAc_probe_list <- lapply(files_1, read_csv)
SNt_VTAc_probes <- data.frame(bind_rows(SNt_VTAc_probe_list))

# VTA target, SNpc contrast
files_2 <- list.files(path = "/Users/rezaadibnia/Desktop/PD NM Project/mDBR - VTA vs SNpc/VTA target, SNpc contrast/", pattern = "Probes_.*csv", full.names = TRUE, ignore.case = FALSE)
VTAt_SNc_probe_list <- lapply(files_2, read_csv)
VTAt_SNc_probes <- data.frame(bind_rows(VTAt_SNc_probe_list))

```


  After loading in the data, it needed to be transformed to allow for simpler analysis. This included transforming fold change values for VTA target, SNpc contrast. P-values did not need to be BH-corrected, because the differential expression search already performed this correction. The definition of the fold change of B with respect to A is B/A. Therefore, the fold change of A with respect to B is A/B. In other words, the fold change of gene expression in the SNpc with respect to the VTA is equal to 1/(the fold change of gene expression in the VTA with respect to the SNpc). By taking the reciprocal of the fold change values with VTA target and SNpc contrast, we get the equivalent fold change for SNpc target and VTA contrast, allowing us to combine the differential expression analysis of all 20000 probes into one data frame.
```{r}

# Converting VTA target, SNpc contrast fold change to SNpc target, VTA contrast fold change
VTAt_SNc_probes$fold.change <- 1/(VTAt_SNc_probes$fold.change)

# Combining the data frames containing p values and fold change
mDBR_p_and_FC <- data.frame(bind_rows(SNt_VTAc_probes, VTAt_SNc_probes))
mDBR_p_and_FC <- mDBR_p_and_FC[,c(4,10:11)]


```

____________________________________________________________________________________________

September 24, 2020

FILTERING FOR P < 0.05 and FC > 90th PERCENTILE (TOP 10%)
  Filters for top 10% of genes with highest FC compared to all probes, not just significantly differentially expressed prboes VTA target, SNpc contrast is filtered for fold changes less than the 10th percentile because fold change are all less than 1. The lowest 10% of fold change values indicate the top 10% of probes that are the most downregulated in the SNpc compared to the VTA.
```{r}
library(dplyr)

# SNpc target, VTA contrast
sig_SNt_VTAc_probes <- SNt_VTAc_probes %>% 
  filter(SNt_VTAc_probes$p < 0.05 & 
         SNt_VTAc_probes$fold.change > quantile(SNt_VTAc_probes$fold.change, 0.90))

# VTA target, SNpc contrast
sig_VTAt_SNc_probes <- VTAt_SNc_probes %>% 
  filter(VTAt_SNc_probes$p < 0.05 & 
         VTAt_SNc_probes$fold.change < quantile(VTAt_SNc_probes$fold.change, 0.10))

```

  In order to visualize the probes that are differentially expressed, a volcano plot of all the probes was constructed. The vertical lines indicate the log2(fold change) thresholds, the horizontal line indicates the p-value threshold (0.05). The top 10% of upregulated probes and bottom 10% of downregulated probes were chosen to be significant (based on fold change values). The significantly differentially expressed probes are found in the upper-left and upper-right corners of the volcano plot. Since there were many probes that were significantly upregulated or downregulated, the individual points were not labeled on the volcano plot.
```{r, echo=FALSE}
library(ggplot2)
library(dplyr)

# Vertical lines for log2(fold change) thresholds, horizontal line for the p-value threshold
# Top 10% of upregulated probes and bottom 10% of downregulated probes are chosen
horz_min_mDBR = quantile(log2(VTAt_SNc_probes$fold.change), 0.10)
horz_max_mDBR = quantile(log2(SNt_VTAc_probes$fold.change), 0.90)

# Add a column to mDBR_p_and_FC data frame indicating whether probe is upregulated, downregulated, or not significant
mDBR_p_and_FC$differentially_expressed <- "Not Significant"
# If log2Foldchange > horz_max and pvalue < 0.05, set as "Upregulated" 
mDBR_p_and_FC$differentially_expressed[log2(mDBR_p_and_FC$fold.change) > horz_max_mDBR & 
                                         mDBR_p_and_FC$p < 0.05] <- "Upregulated"
# If log2Foldchange < horz_min and pvalue < 0.05, set as "Downregulated"
mDBR_p_and_FC$differentially_expressed[log2(mDBR_p_and_FC$fold.change) < horz_min_mDBR & 
                                         mDBR_p_and_FC$p < 0.05] <- "Downregulated"

# Construct volcano plot                      
volcano_plot_mDBR <- ggplot(data=mDBR_p_and_FC, 
                       aes(x=log2(fold.change), 
                           y=-log10(p), 
                           col=differentially_expressed)) + 
                      geom_point(size = 1) + 
                      scale_colour_manual(values = c("Not Significant"="grey", 
                                                     "Downregulated"="red", 
                                                     "Upregulated"="blue")) +
                      geom_vline(xintercept=c(horz_min_mDBR, horz_max_mDBR), 
                                 col="black") + 
                      geom_hline(yintercept=-log10(0.05), 
                                 col="black") +
                      labs(x = bquote(""*log[2]*"FC"), 
                           y = bquote("-"*log[10]*" "*italic(P)*"-value")) +
                      ggtitle("Probes Differentially Expressed in SNpc") + 
                      theme

```


```{r}

volcano_plot_mDBR


```

  This volcano plot showed only probes, and not individual genes. For it to show genes only, the genes needed to be aligned to their probes first.

____________________________________________________________________________________________

September 25, 2020

  Expression data for the top 20000 differentially expressed probes was read into RStudio. This data is actually not necessary to overall analysis, but is good to have for reference.
```{r, echo=FALSE}
library(readr)
library(dplyr)

read_csv_labeled_header <- function(file) {
  labels = c("Probe_IDs", "SN_1", "SN_2", "SN_3", "SN_4", "SN_5", "SN_6", "VTA_1", "VTA_2", "VTA_3", "VTA_4", "VTA_5", "VTA_6")
  read_csv(file, col_names = labels)
}

# SNpc target, VTA contrast
files_expr_1 <- list.files(path = "/Users/rezaadibnia/Desktop/PD NM Project/mDBR - VTA vs SNpc/SNpc target, VTA contrast/", pattern = "Expression_.*csv", full.names = TRUE, ignore.case = FALSE)
SNt_VTAc_expr_list <- lapply(files_expr_1, read_csv_labeled_header)
SNt_VTAc_expr <- data.frame(bind_rows(SNt_VTAc_expr_list))

# VTA target, SNpc contrast
files_expr_2 <- list.files(path = "/Users/rezaadibnia/Desktop/PD NM Project/mDBR - VTA vs SNpc/VTA target, SNpc contrast/", pattern = "Expression_.*csv", full.names = TRUE, ignore.case = FALSE)
VTAt_SNc_expr_list <- lapply(files_expr_2, read_csv_labeled_header)
VTAt_SNc_expr <- data.frame(bind_rows(VTAt_SNc_expr_list))

```

  Expression and probe data frames were joining into one, giving a data frame containing all both the expression of each probe in each sample, and the p-value and fold change for that gene.
```{r}
library(dplyr)

SNt_VTAc_expr <- SNt_VTAc_expr %>% inner_join(SNt_VTAc_probes, by = c("Probe_IDs" = "id"))
VTAt_SNc_expr <- VTAt_SNc_expr %>% inner_join(VTAt_SNc_probes, by = c("Probe_IDs" = "id"))

sig_SNt_VTAc_expr <- SNt_VTAc_expr %>% inner_join(sig_SNt_VTAc_probes, by = c("Probe_IDs" = "id"))
sig_VTAt_SNc_expr <- VTAt_SNc_expr %>% inner_join(sig_VTAt_SNc_probes, by = c("Probe_IDs" = "id"))

mDBR_expr <- data.frame(bind_rows(SNt_VTAc_expr, VTAt_SNc_expr))
sig_mDBR_expr <- data.frame(bind_rows(sig_SNt_VTAc_expr, sig_VTAt_SNc_expr))

```

____________________________________________________________________________________________

September 26, 2020

  After filtering for probes that had p-values < 0.05 and significant fold changes, the probes needed to be aligned to individual genes. To do this, the expression data for all probes in all brain regions in all 6 donors had to be downloaded from Allen Human Brain Atlas (AHBA). The probes lacking Entrez IDs were first filtered out. Conveniently, AHBA provides a presence/absence flag that determines whether a certain probe's expression is well above the background levels or not. This flag was used to filter out any probes that are not expressed above the background in less than 1% of all samples.
  There may be multiple probes that align to the same gene, but only one probe per gene can be selected. Therefore, a systematic method was used to align probes to genes using the collapseRows function from the WGCNA package, similar to methods previously used in the literature [15]:

  - If there is only one probe for a gene, that one is chosen
  - If there are two probes for one gene, the one with the max variance across all samples is chosen (method = maxRowVariance)
  - If there more than two probes for one gene, the probe with the highest connectivity (summed adjacency) is chosen (connectivityBasedCollapsing = TRUE)

```{r}
library(WGCNA)

########## Load data ##########

# Set donor names
donorNames <- c("donor9861", "donor10021", "donor12876", "donor14380", "donor15496", "donor15697")
names(donorNames) <- donorNames

# Location of AHBA directories
ahba_download <- "/Users/rezaadibnia/Desktop/AHBA Full Data/Microarray" # Where downloaded files are stored

# Read in probe info (files are the same for each donor)
probeInfo <- read.csv(paste0(ahba_download, "/normalized_microarray_donor9861/Probes.csv")) # Same for each donor

# Read expression data for each donor (probes x samples)
brainExpr <- lapply(donorNames, function(d){
  file1 <- paste0(ahba_download, "/normalized_microarray_", d, "/MicroarrayExpression.csv")
  e <- read.csv(file1, header = FALSE)
  rownames(e) <- e[,1]
  e <- e[,-1]
  file2 <- paste0(ahba_download, "/normalized_microarray_", d, "/SampleAnnot.csv")
  sample_annotation <- read.csv(file2)
  colnames(e) <- sample_annotation$structure_id
  e
})

########## Filter probes based on concatenated expression data from all donors ##########

# Probes with missing Entrez IDs
probes_missing_entrezID <- is.na(probeInfo$entrez_id) # same in all donors
print(paste(sum(probes_missing_entrezID), "probes with missing entrez IDs"))

# Probes with expression well above background in at least 1% of samples in all donors
pa_call <- lapply(donorNames, function(d){
  file <- paste0(ahba_download, "/normalized_microarray_", d, "/PACall.csv")  
  pa <- read.csv(file, header = FALSE)
  rownames(pa) <- pa[,1]
  pa[,-1]
})
pa_concat <- Reduce(cbind, pa_call)
sum <- rowSums(pa_concat)
low_presence <- sum < 0.01*ncol(pa_concat)
print(paste(sum(low_presence), "probes present in <1% of samples"))

# Concatenate data across all donors and filter probes by ID
expr_concat <- Reduce(cbind, brainExpr)
expr_concat <- expr_concat[!(probes_missing_entrezID | low_presence),]
AHBA_expr_concat <- expr_concat
probeInfo <- probeInfo[!(probes_missing_entrezID | low_presence), ]

probe2gene <- collapseRows(AHBA_expr_concat, 
                           rowGroup = probeInfo$entrez_id, 
                           rowID = probeInfo$probe_id,
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
selected_probes <- probe2gene$selectedRow
entrez_id <- probeInfo$entrez_id[selected_probes]
entrez_order <- order(entrez_id)

probeInfo <- probeInfo[selected_probes, ]
probeInfo <- probeInfo[entrez_order,]

# Save probeInfo and AHBA_expr_concat with probes aligned to genes to csv file for later use
write.csv(x=probeInfo, file="/Users/rezaadibnia/Desktop/AHBA Full Data/updated_probeInfo.csv")
write.csv(x=AHBA_expr_concat, file="/Users/rezaadibnia/Desktop/AHBA Full Data/updated_AHBA_expr.csv")

```

  To align significantly differentially expressed probes to genes, the data frame containing the significantly differentially expressed probes and the data frame containing all the probes selected for each genes were joined by probe ID.
```{r}
library(dplyr)

# Combine data frames by probe ID to filter out all probes not corresponding to a gene
final_expr <- sig_mDBR_expr %>% inner_join(probeInfo, by = c("Probe_IDs" = "probe_id"))

# Remove unnecessary and repeated columns
final_expr <- final_expr[,-c(14, 18:21, 24:29)] %>% data.frame
final_expr <- final_expr[order(final_expr$fold.change.x, decreasing = TRUE), ]

mDBR_genes <- final_expr[,c(15, 17:18)]
colnames(mDBR_genes) <- c("Gene", "p", "fold.change")

# Top 10 upregulated genes
mDBR_genes[1:10,]

# Top 10 downregulated genes
mDBR_genes[1984:1993,]

```
  
  Among the 739 genes found to be within the top 10% of genes significantly upregulated or downregulated in the SNpc relative to the VTA were two of the target genes:
  - SLC18A2 (BH-corrected p = 0.00145, FC = 4.656)
  - SLC6A3 (BH-corrected p = 0.000828, FC = 4.784)
  
  The SLC18A2 gene codes for the VMAT2 enzyme, which transports dopamine into monoaminergic vesicles to prevent it from oxidizing [2]. VMAT2 activity has been shown to be inversely proportional to neuromelanin levels, so one would hypothesize that higher SLC18A2 expression would contribute to less neuromelanin [2], and therefore it should be downregulated in melanized dopaminergic brain regions. However, the opposite case is evident here, indicating that the hypothesis was rejected. With fold change = 4.656, SLC18A2 is actually upregulated in the melanized SNpc. This suggests that there may be more at play in neuromelanin formation than just the action of VMAT2, and that VMAT2 activity is only inversely related to neuromelanin in specific in vitro settings.
  
  COMTD1 was also significantly upregulated in the SNpc (BH-corrected p = 0.004663278, FC = 2.149), a gene that is closely related to the target gene COMT.
  
  Additionally, there were numerous other genes linked to dopamine, including dopamine receptor 2 (DRD2, BH-corrected p = 1.72e-06, FC = 3.341) and dopamine receptor 5 (DRD5, BH-corrected p = 1.47e-09, FC = 0.2442002). This, as well as the upregulation of SLC18A2 and SLC6A3, may be due to a difference in dopaminergic neuron population size between the two brain regions. Therefore, it would be wise to perform gene expression analysis while correcting for cell type. This would 

  Further analysis should be done on these 696 genes to determine which molecular pathways they are involved in.
  
____________________________________________________________________________________________

September 27, 2020

  Differentially expressed genes were visualized in a volcano plot, similar to how probes visualized previously.

```{r}
library(ggplot2)
library(dplyr)

# Align all probes to genes
mDBR_expr_genes <- mDBR_expr %>% inner_join(probeInfo, by = c("Probe_IDs" = "probe_id"))
print(paste(dim(mDBR_expr_genes)[1], "genes were aligned to 58692 probes"))

# Filter for upregulated and downregulated genes, separating into different data frames
up_mDBR_genes_plot <- mDBR_expr_genes %>% filter(mDBR_expr_genes$fold.change > 1)
down_mDBR_genes_plot <- mDBR_expr_genes %>% filter(mDBR_expr_genes$fold.change < 1)

# Vertical lines for log2(fold change) thresholds, horizontal line for the p-value threshold
# Top 10% of upregulated probes and bottom 10% of downregulated probes are chosen
horz_min_mDBR = quantile(log2(down_mDBR_genes_plot$fold.change), 0.10)
horz_max_mDBR = quantile(log2(up_mDBR_genes_plot$fold.change), 0.90)

# Add a column to mDBR_genes data frame indicating whether probe is upregulated, downregulated, or not significant
mDBR_expr_genes$differentially_expressed <- "Not Significant"
# If log2Foldchange > horz_max and pvalue < 0.05, set as "Upregulated" 
mDBR_expr_genes$differentially_expressed[log2(mDBR_expr_genes$fold.change) > horz_max_mDBR & 
                                         mDBR_expr_genes$p < 0.05] <- "Upregulated"
# If mDBR_expr_genes < horz_min and pvalue < 0.05, set as "Downregulated"
mDBR_expr_genes$differentially_expressed[log2(mDBR_expr_genes$fold.change) < horz_min_mDBR & 
                                         mDBR_expr_genes$p < 0.05] <- "Downregulated"

# Construct volcano plot                      
volcano_plot_mDBR <- ggplot(data=mDBR_expr_genes, 
                       aes(x=log2(fold.change), 
                           y=-log10(p), 
                           col=differentially_expressed)) + 
                      geom_point(size = 1) + 
                      scale_colour_manual(values = c("Not Significant"="grey", 
                                                     "Downregulated"="red", 
                                                     "Upregulated"="blue")) +
                      geom_vline(xintercept=c(horz_min_mDBR, horz_max_mDBR), 
                                 col="black") + 
                      geom_hline(yintercept=-log10(0.05), 
                                 col="black") +
                      labs(x = bquote(""*log[2]*"FC"), 
                           y = bquote("-"*log[10]*" "*italic(P)*"-value")) +
                      ggtitle("Genes Differentially Expressed in Substantia Nigra Pars Compacta (SNpc)") + 
                      theme

volcano_plot_mDBR


```

____________________________________________________________________________________________
  
September 28, 2020

  Genes found to be significantly part of the top 10% of genes differentially expressed in the SNpc were separated into those that were upregulated and those that were downregulated. They were written to csv files to make it easier to copy and paste the gene lists into enrichment analysis programs.
```{r}
library(dplyr)

up_mDBR_genes <- mDBR_genes %>% filter(mDBR_genes$fold.change > 1)
write.csv(x=up_mDBR_genes, file="up_mDBR_genes.csv")

down_mDBR_genes <- mDBR_genes %>% filter(mDBR_genes$fold.change < 1)
write.csv(x=down_mDBR_genes, file="down_mDBR_genes.csv")

```

  Gene set enrichment analysis was performed for these two gene sets (upregulated in SNpc and downregulated in SNpc). These gene sets were inputted into the PANTHER classification system for gene ontology (GO) enrichment analysis. The goal of this was to identify the GO terms associated with biological processes for which each gene set is enriched using Fischer's exact test with false discovery rate calculation (search for "biological process"). A search was also done for enrichment of cellular components (search for "cellular component" in PANTHER database). Results in PANTHER classification system were deemed significant if the false discovery rate (FDR) p-value < 0.05. These gene sets were also inputted into the STRING database to determine how their proteins interact. While STRING DB does also provide GO enrichment analysis, PANTHER DB allows for the input of more genes, therefore GO results from PANTHER DB were preferred. The results of this analysis are listed below:
  
PANTHER DB:
  
  Upregulated in SNpc:
    - Biological process search:
      - Biological processes upregulated in the SNpc were generally associated with dopaminergic neurons and synapses
      - Aminergic neurotransmitter loading into synaptic vesicle (fold enrichment = 43.03, FDR p = 0.0478)
        - Includes SLC18A2, DDC, VAT1
      - Catecholamine biosynthetic process (fold enrichment = 14.34, FDR p = 0.0217)
        - Includes SLC18A2, DDC, VAT1 
      - Dopaminergic neuron differentiation (fold enrichment = 11.10, FDR p = 0.0163)
        - Includes RSPO2, LMX1B, EN1, WNT3, PITX3, FOXA1
      - Regulation of dopamine secretion (fold enrichment = 9.56, FDR p = 0.0261)
        - Includes RAB3A, SYT17, SNCA, SYT1, DRD2, CHRNA6
      - Calcium-ion regulated exocytosis (fold enrichment = 9.56, FDR p = 0.010)
      - Synaptic vesicle exocytosis (fold enrichment = 9.56, FDR p = 0.00137)
      - Positive regulation of synapse assembly (fold enrichment = 6.95, FDR p = 0.0170)
      - Regulation of exocytosis (fold enrichment = 3.67, FDR p = 0.0311)
      - Regulation of transmembrane transporter activity (fold enrichment = 3.27, FDR p = 0.0217)
      - Axonogenesis (fold enrichment = 2.80, FDR p = 0.0296)
      - Modulation of chemical synaptic transmission (fold enrichment = 2.76, FDR p = 0.0247)
      - Regulation of membrane potential (fold enrichment = 2.61, FDR p = 0.0444)
      - Ion transport (fold enrichment = 1.89, FDR p = 0.0295)
      - Gene expression (fold enrichment = 0.30, FDR p = 0.00101)
      - RNA metabolic process (fold enrichment = 0.25, FDR p = 0.00252)
    - Cellular component search:
      - Clathrin-sculpted monoamine transport vesicle membrane (fold enrichment = 45.89, FDR p = 0.00182)
      - Cytoskeleton of presynaptic active zone
      - Platelet dense tubular network membrane
      - Terminal bouton
      - Synaptic vesicle membrane
      - Sacroplasm
      - Glutamatergic synapse (fold enrichment = 3.31, FDR p = 0.00918)
      - Postsynaptic membrane (fold enrichment = 3.13, FDR p = 0.0169)
      - Ion channel complex
      - Postsynaptic density
      - Integral component of plasma membrane	
      - Negatively enriched for nucleoplasm (fold enrichment = 0.65, FDR p = 0.0390)

  Downregulated in SNpc:
    - Biological process search:
      - Regulation of epinephrine secretion	(fold enrichment = 39.57, FDR p = 0.00283)
        - Includes VIP, CRH, CARTPT, ADRA2A
      - Neuron-glial cell signaling	(fold enrichment = 35.61, FDR p = 0.0197)
        - Includes ADRA1A, ADRA1D, ADRA1B
      - Positive regulation of glucocorticoid secretion	(fold enrichment = 29.68, FDR p = 0.0256)
        - Includes GALR1, CRH, TAC1
      - Positive regulation of heart rate by epinephrine-norepinephrine	(fold enrichment = 29.68, FDR p = 0.0255)
        - Includes ADRA1A, ADRA1D, ADRA1B
        - Positive regulation of blood pressure by epinephrine-norepinephrine	(fold enrichment = 33.92, FDR p = 0.00325)
          - Includes ADRA1A, ADRA1D, ADRA1B, DRD5
      - Regulation of serotonin secretion	(fold enrichment = 29.68, FDR p = 0.00440)
        - Incluldes SLC6A4, CNR1, HTR1A, CRH
      - Adenylate cyclase-activating adrenergic receptor signaling pathway (fold enrichment = 14.13, FDR p = 0.00684)
        - Adrenergic receptor signaling pathway	(fold enrichment = 12.90, FDR p = 0.00898)
          - Includes ADRA1A, ADRA1D, ADRA1B, DRD5, ADRA2A
      - Positive regulation of cAMP-mediated signaling (fold enrichment = 13.49, FDR p = 0.00794)
      - Neurotransmitter biosynthetic process	(fold enrichment = 13.19, FDR p = 0.0296)
        - Includes SLC6A4, GAD2, NOS1, GAD1
      - Regulation of synaptic transmission, GABAergic (fold enrichment = 13.19, FDR p = 0.0296)
        - Includes ADRA1A, BAIAP3, CNR1, CNTNAP4, TAC1
      - Oligodendrocyte differentiation	(fold enrichment = 5.77, FDR p = 0.0253)
        - Includes FA2H, PLP1, BOK, NKX6-1, NKX2-2, ZNF488, NKX6-2
      - Membrane repolarization during cardiac muscle cell action potential
      - Neuron fate specification
      - Positive regulation of calcium ion import
      - Behavioral fear response
      - Vasoconstriction
      - Negative regulation of potassium ion transport
      - Extracellular matrix organization
      - Semaphorin-plexin signaling pathway
      - Cardiac muscle cell action potential involved in contraction
      - Neuropeptide signaling pathway
      - Negative regulation of blood pressure
      - Diencephalon development
      - Adenylate cyclase-inhibiting G protein-coupled receptor signaling pathway
      - Pancreas development
      - Ventricular cardiac muscle tissue development
      - Sensory perception of pain
      - Phospholipase C-activating G protein-coupled receptor signaling pathway	
      - Cardiac ventricle morphogenesis
      - Endocrine system development
      - Regulation of cardiac muscle contraction
      - Epithelial cell proliferation
      - Nerve development
      - Epithelial cell migration
      - Regulation of neurotransmitter receptor activity
      - Regulation of potassium ion transmembrane transport
      - Response to catecholamine
      - Response to ethanol
      - Regulation of postsynaptic membrane potential
      - Regulation of cation channel activity
      - Positive regulation of neurogenesis
      - Positive regulation of cation transmembrane transport
      - Response to alkaloid
      - Locomotory behavior
      - Regulation of G protein-coupled receptor signaling pathway
      - Chemical synaptic transmission
      - Adult behavior
      - Regulation of axonogenesis
      - Regulation of insulin secretion
      - Regulation of synaptic plasticity
      - Axonogenesis
    - Cellular component search:
      - Dopaminergic synapse (fold enrichment = 19.78, FDR p = 0.0342)
        - Includes RAB3B, CHRNA5, ADRA2A
      - Main axon
      - Presynaptic membrane
      - Synaptic vesicle membrane
      - Perikaryon
      - Postsynaptic membrane
      - Dendrite
      - Negatively enriched for cytosol	and nucleoplasm GO terms (genes downregulated in SNpc are not those that are present in cytosol or nucleoplasm)


STRING DB:

  Upregulated in SNpc: 354 proteins aligned to 374 genes
    - Medium confidence (0.400) and higher edges shown only
    - Large cluster of nodes and edges in center of network includes DDC, SLC18A2, SLC6A3, DRD2, SNCA, GAL, PRL, PROK2, AGTR1, DLGAP3, GNG3, KCNJ6, RAB3A
    - GO term enrichment analysis by STRING found similar results to PANTHER DB
    
  Downregulated in SNpc: 346 proteins aligned to 365 genes
    - Medium confidence (0.400) and higher edges shown only
    - Large cluster of highly connected proteins (most nodes with highest confidence edges connecting to all other nodes)
      - Includes GAD1, GAD2, SLC6A4, ADRA2A, GALR1, PNOC, GRM7, CNR1, SLC17A6, HTR2C, DRD5, CRH, PENK, ADCYAP1, GNG4, TAC1, NMBR, CHRNA5, OPRM1, APLNR, SSTR2, HTR1A, TAS2R5, CALB1, PMCH, SST, ADCYAP1
    - GO term enrichment analysis by STRING found similar results to PANTHER DB, though also noted enrichment for melanocyte migration (biological process term, ZEB2 and KIT genes, log10 fold enrichment = 1.75, FDR = 0.0274), positive regulation of melanocyte differentiation (biological process term, ZEB2 and ADAMTS9 genes, log10 fold enrichment = 1.45, FDR = 0.0495) biological process term, and regulation of developmental pigmentation (biological process term, ZEB2, KIT, IHH and ADAMTS9 genes, log10 fold enrichment = 1.15, FDR = 0.0079)

  Interestingly, SNCA, the gene coding for alpha synuclein, which was part of the regulation of dopamine secretion GO term, was found to be significantly upregulated in the SNpc. This suggests that SNpc neurons may be more vulnerable to neurodegeneration due to a difference in alpha synuclein levels and not neuromelanin, though there may be an as-of-yet undetermined link between alpha synuclein and neuromelanin, a concept that has been explored in numerous papers. 
  Additionally, the enrichment of upregulated genes for the gene expression GO term urges an differential epigenomic analysis of the SNpc and VTA to better understand the transcription mechanisms that are different between the two brain regions.
  Overall, all of the GO terms associated with downregulated genes seem to be associated with different cell types, including adrenergic, serotonergic, and GABAergic neurons, as well as oligodendrocytes. Considering this, it may be necessary to perform a cell-type specific analysis so that the differential expression analysis is not masked by the differences in cell type composition of the SNpc and VTA.


____________________________________________________________________________________________
____________________________________________________________________________________________

COMBINING AGE CORRELATIONS AND mDBR
____________________________________________________________________________________________
____________________________________________________________________________________________
  
October 1, 2020

  A joint data frame of ACGs and mDBR genes was created to determine if there was any overlap between ACGs and genes differentially expressed in the SNpc relative to the VTA.
```{r}
library(dplyr)

ACG_mDBR_joined <- ACGs %>% inner_join(mDBR_genes, by = c("Gene" = "Gene"))
ACG_mDBR_joined


```

  Only two genes were observed to ACGs and differentially expressed in SNpc with respect to the VTA:
    - MYH11 (r = 0.4609939, FC = 0.446628)
    - DPYS (r = -0.4766697, FC = 2.517000)
  
  One would hypothesize that a gene that increases neuromelanin formation is upregulated in the SNpc and upregulated with age, and a gene that hinders neuromelanin formation is downregulated in the SNpc and downregulated with age. Interestingly, MYH11 and DPYS show the opposite trend: while MYH11 is positively correlated with age, meaning it is upregulated in the elderly, it is downregulated in the SNpc, and while DPYS is negatively correlated with age, meaning it is downregulated in the elderly, it is upregulated in the SNpc. Therefore, both of these genes reject the hypothesis.
____________________________________________________________________________________________

October 2, 2020

  The small number of genes observed to be both differentially expressed in SNpc and significantly correlated with age may be due to prior filtering. Selecting the top 10% of age-correlated genes and the top 10% of differentially expressed genes may erase some genes that are both differentially expressed and significantly correlated with age. Therefore, rather than selecting just the top 10% based on correlation coefficients and fold changes, all genes were selected whose BH-corrected p-value < 0.05 for both age-correlated genes and mDBR genes. These two data frames were aligned to determine their overlap.
```{r}
library(dplyr)

# All significant age correlations
print(paste(dim(sig_age_correlations)[1], "genes whose expression in the substantia nigra of non-neurological individuals is significantly correlated with age"))

# Redefine an "age-correlated gene" as any gene whose BH-corrected p-value of correlation with age < 0.05
ACGs <- sig_age_correlations
ACGs <- ACGs[,-2]

# All genes significantly differentially expressed in SNpc relative to VTA (mDBR genes)
all_sig_SNt_VTAc_probes <- SNt_VTAc_probes %>% 
  filter(SNt_VTAc_probes$p < 0.05)

all_sig_VTAt_SNc_probes <- VTAt_SNc_probes %>% 
  filter(VTAt_SNc_probes$p < 0.05)

all_sig_VTAt_SNc_probes <- all_sig_VTAt_SNc_probes[order(all_sig_VTAt_SNc_probes$fold.change, decreasing = TRUE), ]

all_sig_mDBR <- bind_rows(all_sig_SNt_VTAc_probes, all_sig_VTAt_SNc_probes)

# Align mDBR probes to genes (as done previously)
all_sig_mDBR <- all_sig_mDBR %>% inner_join(probeInfo, by = c("id" = "probe_id"))

print(paste(dim(all_sig_mDBR)[1], "genes that are differentially expressed in the substantia nigra pars compacta relative to the ventral tegmental area in non-neurological individuals"))

# Redefine "mDBR_genes" as any gene whose BH-corrected p-value of differential expression is less than 0.05 
mDBR_genes <- all_sig_mDBR
mDBR_genes <- mDBR_genes[, c("gene.symbol", "p", "fold.change")]

# Aligning all genes significantly correlated with age and significantly differentially expressed in SNpc
all_ACG_mDBR_joined <- ACGs %>% inner_join(mDBR_genes, by = c("Gene" = "gene.symbol"))
print(paste(dim(all_ACG_mDBR_joined)[1], "genes that are both significantly correlated with age and differentially expressed in the substantia nigra pars compacta relative to the ventral tegmental area in non-neurological individuals"))


```

```{r}

# Overlap of all genes significantly correlated with age and significantly differentially expressed in SNpc
all_ACG_mDBR_joined[,c("Gene", "Correlation Coefficient", "BH-corrected p-value", "p", "fold.change")]

```
  As stated previously, one would hypothesize that a gene that increases neuromelanin formation is upregulated in the SNpc and upregulated with age, and a gene that hinders neuromelanin formation is downregulated in the SNpc and upregulated with age. To observe this trend, the fold change from SNpc to VTA of each gene was graphed against its correlation coefficient with age.
  
```{r}

plot(all_ACG_mDBR_joined[,"Correlation Coefficient"], log2(all_ACG_mDBR_joined[,"fold.change"]), xlab = "Correlation Coefficient", ylab = "log2(fold change)")

```
  Evidently, no trend can be established between correlation coefficient and fold change, as there appear to be an equal number of genes with positive and negative log2 fold changes for both positive and negative correlation coefficients. Additionally, the skewedness of correlation cofficients to the two extremes makes it difficult to establish an connection between the two. This can be traced back to the fact that genes with lower p-values of correlation generally have greater absolute values of correlation coefficients, as observed from the volcano plot. Therefore, the act of filtering for p-value of correlation < 0.05 automatically skews the correlation coefficients to highly positive or highly negative. Most likely, performing gene set enrichment analysis of these 118 genes will reveal more information about the pathways involved in neuromelanin formation.
  
____________________________________________________________________________________________

October 4, 2020

  The genes were separated into four categories: genes that are upregulated with age and upregulated in the SNpc, genes that are downregulated with age and downregulated in the SNpc, genes that are upregulated with age and downregulated in the SNpc, and genes that are downregulated with age and upregulated in the SNpc. The first two categories correspond to genes that align with the hypothesis, and therefore are the genes that either contribute to or hinder age-dependent neuromelanin accumulation in the SNpc. The last two categories correspond to genes who do not align with the hypothesis. Performing gene set enrichment analysis of these four categories separately may reveal in better detail which biological mechanisms are involved in age-dependent neuromelanin accumulation.
  
```{r}
library(dplyr)

write.csv(x=all_ACG_mDBR_joined[,c("Gene", "Correlation Coefficient", "fold.change")], file="all_ACG_mDBR_joined.csv") # Writes csv file, saves to working directory to allow for simpler gene set enrichment analysis

age_up_SNpc_up <- all_ACG_mDBR_joined %>% filter(all_ACG_mDBR_joined[,"Correlation Coefficient"] > 0 & all_ACG_mDBR_joined[,"fold.change"] > 1)
age_down_SNpc_down <- all_ACG_mDBR_joined %>% filter(all_ACG_mDBR_joined[,"Correlation Coefficient"] < 0 & all_ACG_mDBR_joined[,"fold.change"] < 1)
age_up_SNpc_down <- all_ACG_mDBR_joined %>% filter(all_ACG_mDBR_joined[,"Correlation Coefficient"] > 0 & all_ACG_mDBR_joined[,"fold.change"] < 1)
age_down_SNpc_up <- all_ACG_mDBR_joined %>% filter(all_ACG_mDBR_joined[,"Correlation Coefficient"] < 0 & all_ACG_mDBR_joined[,"fold.change"] > 1)

# Genes that are upregulated with age and upregulated in the SNpc,
age_up_SNpc_up[,c("Gene", "Correlation Coefficient", "fold.change")]


# Genes that are downregulated with age and downregulated in the SNpc
age_down_SNpc_down[,c("Gene", "Correlation Coefficient", "fold.change")]
write.csv(x=age_down_SNpc_down[,c("Gene", "Correlation Coefficient", "fold.change")], file="age_down_SNpc_down.csv") # Writes csv file, saves to working directory to allow for simpler gene set enrichment analysis


# Genes that are upregulated with age and downregulated in the SNpc
age_up_SNpc_down[,c("Gene", "Correlation Coefficient", "fold.change")]
write.csv(x=age_up_SNpc_down[,c("Gene", "Correlation Coefficient", "fold.change")], file="age_up_SNpc_down.csv") # Writes csv file, saves to working directory to allow for simpler gene set enrichment analysis


# Genes that are downregulated with age and upregulated in the SNpc
age_down_SNpc_up[,c("Gene", "Correlation Coefficient", "fold.change")]
write.csv(x=age_down_SNpc_up[,c("Gene", "Correlation Coefficient", "fold.change")], file="age_down_SNpc_up.csv") # Writes csv file, saves to working directory to allow for simpler gene set enrichment analysis


```
  
  Gene set enrichment analysis was performed for the last three categories (the first category, both upregulated with age and in the SNpc, only contained three genes), each as one gene set, as well as all 118 genes that are both significantly correlated with age and differentially expressed in the SNpc as another gene set. These gene sets were inputted into the PANTHER classification system for gene ontology (GO) enrichment analysis. The goal of this was to identify the GO terms associated with biological processes for which each gene set is enriched using Fischer's exact test with false discovery rate calculation (search for "biological process"). A search was also done for enrichment of cellular components (search for "cellular components" in PANTHER database). Results in PANTHER classification system were deemed significant if the false discovery rate (FDR) p-value < 0.05. These gene sets were also inputted into the STRING database to determine how their proteins interact. While STRING DB does also provide GO enrichment analysis, PANTHER DB allows for the input of more genes, therefore GO results from PANTHER DB were preferred. The results of this analysis are listed below:
  
PANTHER DB:
  Age down, SNpc down:
    - No statistically significant results for both biological process search and cellular component search
  Age up, SNpc down:
    - No statistically significant results for both biological process search and cellular component search
  Age down, SNpc up:
    - No statistically significant results for both biological process search and cellular component search
  All 118 genes:
    - No statistically significant results for both biological process search and cellular component search
    
STRING DB:
  Age down, SNpc down: 30 proteins mapped to 30 genes
    - No interactions between any proteins, no significant GO term enrichment results
  Age up, SNpc down: 16 proteins mapped to 16 genes
    - High confidence (0.700) interaction between MYH11 and TAGLN
    - Enrichment for MAP kinase tyrosine/serine/threonine phosphatase activity (strength = 2.27, FDR = 0.0065)
  Age down, SNpc up: 69 proteins mapped to 69 genes
    - SOD1 appeared to be a hub center, with highest confidence (0.900) interactions with TXND1, GPX4
    - COX7C, UQCR11, NDUFA6, COA6 all have highest confidence interactions with each other
    - Enrichment for glyceraldehyde-3-phosphate dehydrogenase (NAD+) (non-phosphorylating) activity, nucleobase binding
  All 118 genes: 117 proteins mapped to 118 genes
    - ALDH2, CAT, ALDH3A2 showed highest confidence interactions with each other
    - No other new connections between proteins
    - No significant enrichment detected
  
  While no statistically significant results were found, there were still a number of genes in these sets known to be associated with the melanosome/pigment granule GO term from enrichment analysis on ACGs. These included OCA2 (r = -0.3195436, FC = 0.3787879), RAB9A (r = -0.3879255, FC = 0.5938242), and YWHAZ (r = -0.3164595, FC = 1.415). The reason for the observation that no significantly enriched pathways is likely due to the fact that there are a relatively small number of genes in each gene set.
  
____________________________________________________________________________________________
____________________________________________________________________________________________

RE-ASSESSING METHODOLOGY
____________________________________________________________________________________________
____________________________________________________________________________________________

November 1, 2020

Reassessing target genes

  None of the nine target genes specified at the beginning of this study were found to be both correlated with age and differentially expressed in the SNpc. Gene ontology enrichment analysis of all genes provided much more significant results. Overall, analyzing only a set of nine genes can shift the focus away from other potential biological mechanisms that may be involved in age-dependent neuromelanin accumulation and neurotoxicity. Therefore, analyses will not longer be focused solely on the nine target genes specified at the beginning of this study, and will instead be focused on identifying any genes that may be involved in age-dependent neuromelanin accumulation in Parkinson's disease, as was the original goal of this study.
  
____________________________________________________________________________________________

November 5, 2020

Correcting for cell-type bias
  
  Particularly after performing melanized dopaminergic brain region differential expression analysis (mDBR), it was determined that cell-type bias may be confounding the results of differential expression analysis. Although the substantia nigra pars compacta and ventral tegmental area are similar, they contain some key differences in cell type populations that most likely gave rise to the observed results. Genes downregulated in the SNpc (and thus upregulated in the VTA) were enriched for epineprhine, norepineprhine, and serotonin-related biologicial processes and cellular components, while genes upregulated in the SNpc were generally enriched for dopaminergic neuron-related biological processes and cellular components.
  An analysis method called populations-specific expression analysis (PSEA) present in the literature that could help eliminate cell-type bias, and even determine whether is it only cell type that affects the differential expression of certain genes. PSEA methods include both analyzing individual genes and performing deconvolution of expression profiles based on cell type in whole data sets. Adding PSEA to differential expression analysis could shine some light on misunderstood mechanisms in PD. For example, hypothetically, it could explain the observed changes in DAT (dopamine active transporter, coded by SLC6A3) and TH levels observed in neuromelanin rat models of PD; they may be due entirely to dopaminergic neuron death. In fact, we might observe an upregulation of dopaminergic neuron-specific expression of TH and SLC6A3 in Parkinson's disease, but the reason we observe an overall downregulation is the death of dopaminergic neurons neurons.
  PSEA could also be performed on the UKBEC data (used for age correlations), and it could shine some light on the different cell populations in the substantia nigra (i.e., pars compacta versus pars reticulata). However, it is not particularly necessary for the purpose of this study, as there is little cell-type bias across samples in that data set since they are all from the same brain region.

____________________________________________________________________________________________

November 10, 2020

Reassessing GEO data set analyses for age correlations validation in PD

  Initially, it was thought that age correlations could not be validated using PD data sets due to the large difference between the ages of the non-neurological individuals and the ages of PD patients. However, upon re-analysis of the observed results, there were likely other reasons for insignificant results from analysis of these data sets.

```{r}

df <- GSE20292_all_cor[order(GSE20292_all_cor$`Correlation Coefficient`), c("Gene", "Correlation Coefficient", "Raw p-value", "BH-corrected p-value")]
df[1:10,] # GSE20292

df <- GSE8397_all_cor[order(GSE8397_all_cor$`Correlation Coefficient`), c("Gene", "Correlation Coefficient", "Raw p-value", "BH-corrected p-value")]
df[1:10,] # GSE8397
rm(df)

```

  First of all, looking at the GEO correlation analysis results, it seems that the probes were not aligned to genes. Therefore, when re-analyzing, probes should be aligned to genes first using collapseRows as previously done for all data sets.
  Second of all, it appears that the BH-correction of p-values on the GEO datasets erased all of the significant results. Looking at GSE20292, it appears that all of the p-values were corrected to 0.9999895, even those that were originally on the scale of 1e-5. The same phenomenon likely occurred for GSE8397. In fact, a lot of the correlation coefficients from the GEO datasets are much greater in magnitude than the correlation coefficients obtained in UKBEC data analysis, further indicated that there were a number of genes significantly correlated with age in the GEO datasets. Therefore, instead of BH-correcting p-values, it may be wise to simply decrease the significance threshold to 0.01 and use raw p-values.
  Of course, the fact that the distributions of samples ages of GSE8397 and GSE20292 are vastly different from the distribution of UKBEC sample ages still remains. However, this does not nullify the significant correlation results observed by looking at raw p-values; rather, it provides an explanation for the lack of overlap of significantly correlated genes (based on raw p-values) between the two data sets and with the correlated genes obtained from UKBEC analysis. Performing similar analysis on more data sets may address this problem.
  The previous analysis done on the GEO datasets only involved looking at PD samples. However, it may be wise to consider control samples from those data sets as well, to perform multiple linear regressions for the expression of each gene rather than correlation analyses in just PD patients. The multiple linear regressions would be relatively simple, with only two regressors: age and disease state (which is binary). By incorporating control samples from GEO data sets into the analyses, the distribution of ages would also increase and more closely match that of the UKBEC data. Additionally, it would allow us to do age-specific differential expression analysis: determining which genes change their age-specific expression significantly in Parkinson's disease. The fact that the change is age-specific is critical, because it allows us to draw connections between those genes and age-dependent neuromelanin accumulation.

____________________________________________________________________________________________

November 12, 2020

Reassessing selection criteria

  Finding the overlap between mDBR and age correlations had provided some significant results. However, it is possible to go into more detail with selection criteria, especially by adding age-specific expression changes from control to PD and population-specific expression analyses. This has the potential to provide even more significant and impactful results. Essentially, new selection criteria should include both age-dependent and population-specific expression signatures.
  
  Finally, one other validation analysis method that is necessary is to perform age correlations in other brain regions provided by in the UKBEC data sets. This could verify whether the observed enrichment for mitochondria-related and melanosome-related GO terms in genes negatively correlated with age is specific to the substantia nigra or not.

____________________________________________________________________________________________

November 15, 2020

Overview of improved/added methodology

  From this point on, further methods will include:
    - Population-specific expression analysis (PSEA) on mDBR (AHBA)
      - Can do PSEA deconvolution on UKBEC age correlations as well, if time, although it is not necessary because they are all substantia nigra samples and cell-type bias across samples will be minimal
    - Validation of results
      - Age correlations for other brain regions to ensure observed enrichment for GO-terms is specific to the substantia nigra
      - Multiple linear regression analysis of Parkinson's disease gene expression in the substantia nigra
        - To determine whether genes both correlated with age in the substantia nigra and differentially expressed in SNpc are also implicated in Parkinson's disease
      - Searching the literature for genes and biological pathways found to be significant and/or interesting in the results of this research
    - Selecting genes based on new criteria:
      - |r| of age correlation in non-neurological patients (given BH-corrected p-value of age correlation < 0.05)
      - Fold change of age-specific expression from control to PD (given BH-corrected p-value of Type < 0.05)
      - Fold change of dopaminergic neuron-specific expression from VTA to SNpc (given BH-corrected p-value of differential expression < 0.05)

  Overview of multiple linear regression on GEO datasets:
    - Datasets: GSE8397, GSE20292, GSE49036
    - Perform for each dataset individually
    - Regressors: Age and disease state (0 = non-neurological control, 1 = PD in model matrix)
    - Disease state refers to PD offset; that is, the offset of PD expression values from control expression values
    - Regression for all genes
      - lm(Expression~Type + Age)
    - Fold change in PD with respect to control (i.e., from control to PD) = (Coefficient of Type + Coefficient of Age) / Coefficient of Age
      - Age-specific expression in controls is given by reference signal coefficient, age-specific expression in PD is given by sum of both coefficients
    - Can perform PSEA deconvolution as well, although it is not necessary because they are all substantia nigra samples and cell-type bias across samples will be minimal

____________________________________________________________________________________________
____________________________________________________________________________________________

POPULATION-SPECIFIC EXPRESSION ANALYSIS CONCEPT
____________________________________________________________________________________________
____________________________________________________________________________________________

November 22, 2020

  The concept of population-specific expression analysis (PSEA), developed by Kuhn et al., [16], is relatively simple. It involves using a couple of marker genes to create a "reference" expression signal describe each cell population and constructing mutliple linear regression models for each gene using all of those reference signals as regressors. The reference signal is simply the average expression of all the marker genes. The main assumption is that the expression of the marker genes for a given cell population is directly proportional to the size of that cell population, and therefore the reference signal is also directly proportional to the size of that cell population. Therefore, by constructing regression models for each gene with reference signals as regressors, one is essentially describing the expression of a given gene using the size of the different cell populations. The p-value for each regressor describes whether the added regressor, that is, the added population type, significantly improves the model. In other words, it answers the question, "does the expression of this gene X significantly depend on cell type Y?". The slope for each regressor describes the magnitude of the population-specific expression. In theory, for a good model, the slope should never be negative, as that means the presence of that cell type decreases the expression of a certain gene. Although that is theoretically possible via some epigenomic mechanisms, it does not make sense in the scope of PSEA models. Because the slope describes the magnitude of population-specific expression, dividing slopes for different regressors, or even for the same regressor from different genes (different models), gives the fold change.
  
  In theory, for n different cell types (regressors), there are nCn + nC(n-1) + nC(n-2) + ... + nC1 + nC0 different models that can be created for each individual gene ("C" stands for "choose"; "nC1" means "n choose 1"). To determine which model is the best for each gene, Akaike information criterion (AIC) is used assess the models for each gene relative to each other [16]. This method of selecting models for each gene is, in context, the same as choosing which cell populations contribute the most to the observed expression of the gene within the brain region, and is called "deconvolution of expression profiles".
  
  Kuhn et al. developed an R-package called PSEA that provides a lot of the functions to perform this analysis. This study uses the package to deconvolute expression profiles of the UKBEC data (used for age correlations) and the AHBA SNpc and VTA expression data (used for mDBR).

References

[16] Kuhn, A., Thu, D., Waldvogel, H. J., Faull, R. L. M. & Luthi-Carter, R. Population-specific expression analysis (PSEA) reveals molecular changes in diseased brain. Nat. Methods 8, 945–947 (2011).

____________________________________________________________________________________________

November 23-24, 2020

Selecting marker genes for each cell population

  Marker genes are typically selecting using data from previous studies and scientific literature. A new database has recently been developed providing information on marker genes for different cell types. PanglaoDB is a database for the scientific community interested in exploration of single cell RNA sequencing experiments from mice and humans. It collects and integrates data from multiple studies and present them through a unified framework. Although it is not currently being developed, it was last updated in May 2020, which is relatively recent in terms of the development of single cell RNA sequencing research. Additionally, the different marker genes identified using this databases were cross-verified using other databases, including CellMarker, and studies, including the neural marker guide provided by the biotechnology company Abcam and GSE19380, a data set containing rat neurons, astrocytes, oligodendrocytes, and microglial cells.
  
  Based on the GO term enrichment analysis previously done, the following cell types are to be used as regressors:
    - Neurons (general)
    - Astrocytes
    - Oligodendrocytes
    - Microglia
    - Dopaminergic neurons
    - Glutamatergic neurons
    - GABAergic neurons
    - Serotonergic neurons
    - Noradrenergic neurons

  Serotonergic and noradrenergic neurons were selected mainly because GO enrichment analysis found that genes downregulated in the SNpc with respect to the VTA (and therefore upregulated in the VTA) included regulation of serotonin secretion and the adrenergic receptor signaling pathway. Glutamatergic and GABAergic neurons were selected as well because they were known to make up a substantial percentage of the neuron population in the VTA. Adrenergic neurons were not selected as a regressor because they are a more general cell type that includes both dopaminergic neurons and noradrenergic neurons, and there would likely be a lot of overlap between the three cell types.
  
  Cell-type markers were selected from PanglaoDB based on the following criteria before cross verification:
    - Each cell type must have unique marker genes
    - Human specificity (how frequently the gene is expressed in cells not of the particular cell type) is less than 0.01
    
  The marker genes selected for each cell type are listed below:
    - Neurons (general): NeuN (RBFOX3), NEFL, ENO2, SLC12A5
    - Astrocytes: GFAP, AQP4, GJA1
    - Oligodendrocytes: MBP, MOBP, MOG, MAG
    - Microglia: AIF1
    - Dopaminergic neurons: TH, SLC6A3
    - Glutamatergic neurons: SLC17A7, SLC17A6
    - GABAergic neurons: GAD1, GAD2, SLC32A1
    - Serotonergic neurons: SLC22A3, FEV, TPH2, PET1
    - Noradrenergic neurons: SLC6A2, SLC39A11, SLC18A3, DBH
    
____________________________________________________________________________________________
____________________________________________________________________________________________

mDBR PSEA
____________________________________________________________________________________________
____________________________________________________________________________________________

November 30, 2020

  Performing mDBR PSEA requires probe expression data for the VTA and SNpc to be downloaded from AHBA. This data was previously downloaded in order to align probes to genes and stored in the brainExpr, which contains all probes, not all genes aligned to probes. This data frames includes the concatenated data from each donor. The data also need to be filtered for SNpc and VTA samples only. Also, the data downloaded from AHBA for differential expression analysis did not contain all genes; it only contained the top 20000 differentially expression probes. All probes would be necessary to perform PSEA, which is another reason why AHBA_expr_concat had to be used.
  
  Due to the small number of samples in the AHBA data set for the desired brain regions (8 for SNpc and 8 for VTA), PSEA could not be used to perform deconvolution of the expression profiles. This is because the large number of reference signals applied as potential regressor in the linear models would be very prone to overfitting, resulting in models for each gene that include every reference signal. This would not provide any useful information about which cell types the expression of certain genes depends on. Therefore, instead of constructing a large number of models for each gene based on every possible combination of regressors and selecting the best one, each relevant reference signal was regressed on each gene independently. This would avoid overfitting, but also provide significant information about the cell types which express certain genes the most.
  
  With AHBA data, we were mainly interesting in observing which genes exhibited population-specific expression changes in the SNpc relative to the VTA. This would require us to define another regressor for each cell type: the difference in expression in the SNpc relative to the VTA. Therefore, one model would represent both the population-specific expression of a gene and the difference in population-specific expression of that gene in the SNpc relative to the VTA. Each model would contain two regressors: one that representing the reference signal for a certain cell type in the VTA, and another representing the difference in the reference signal for the cell type in the SNpc relative to the VTA. Therefore, by adding the coefficients of each regressor, we would obtain the population-specific expression in the SNpc for a certain gene. If we then divide that by the first coefficient, we obtain the fold change in population-specific from VTA to SNpc, which is the statistic we are interested in.
  
  The main cell population of interest was dopaminergic neurons, but models were fitted to all genes for all relevant cell types to determine whether some genes that exhibit dopaminergic neuron-specific expression changes also exhibit expression dependent on other cell types.
  
  The selected cell types to use as regressors included astrocytes, oligodendrocytes, microglia, dopaminergic neurons, glutamatergic neurons, and GABAergic neurons. General neurons were not chosen due to their overlap with the more specific neuron types. Serotonergic and noradrenergic neurons were not chosen because they are not known to consist of a large proportion of the cell population in either the SNpc or the VTA.
  
```{r, echo=FALSE}
library(dplyr)

# Read on sample annotations (called "ontology")
ontology <- read.csv(paste0(ahba_download, "/normalized_microarray_donor9861/Ontology.csv")) # Same for each donor

# Sample IDs
# SNpc - 9074 (left) and 9077 (right)
# VTA - 9067 (left) and 9068 (right)

brainExpr_filt <- brainExpr
for (i in donorNames) {
  if ("9077" %in% colnames(brainExpr_filt[[i]])) {
    if (c("9077", "9067") %in% colnames(brainExpr_filt[[i]])) {
      if (c("9077", "9067", "9068") %in% colnames(brainExpr_filt[[i]])) {
        brainExpr_filt[[i]] <- (brainExpr_filt[[i]])[,c("9074", "9077", "9067", "9068")]
      } else {
        brainExpr_filt[[i]] <- (brainExpr_filt[[i]])[,c("9074", "9077", "9067")]
      }
    } else if (c("9077", "9068") %in% colnames(brainExpr_filt[[i]])) {
      brainExpr_filt[[i]] <- (brainExpr_filt[[i]])[,c("9074", "9077", "9068")]
    } else {
      brainExpr_filt[[i]] <- (brainExpr_filt[[i]])[,c("9074", "9077")]
    }
  } else if ("9067" %in% colnames(brainExpr_filt[[i]])) {
    brainExpr_filt[[i]] <- (brainExpr_filt[[i]])[,c("9074", "9067")]
  } else if ("9068" %in% colnames(brainExpr_filt[[i]])) {
    brainExpr_filt[[i]] <- (brainExpr_filt[[i]])[,c("9074", "9068")]
  }
    
}
AHBA_expr_SNpc_VTA <- Reduce(cbind, brainExpr_filt)
for (i in 1:length(colnames(AHBA_expr_SNpc_VTA))) {
  if (colnames(AHBA_expr_SNpc_VTA)[i] == "9074" | colnames(AHBA_expr_SNpc_VTA)[i] == "9077") {
    colnames(AHBA_expr_SNpc_VTA)[i] <- "SNpc"
  } else if (colnames(AHBA_expr_SNpc_VTA)[i] == "9067" | colnames(AHBA_expr_SNpc_VTA)[i] == "9068") {
    colnames(AHBA_expr_SNpc_VTA)[i] <- "VTA"
  }
}

probeInfo_rough <- read.csv(paste0(ahba_download, "/normalized_microarray_donor9861/Probes.csv")) # Same for each donor
AHBA_expr_SNpc_VTA$Gene <- probeInfo_rough$gene_symbol
AHBA_expr_SNpc_VTA$probe_id <- probeInfo_rough$probe_id
#AHBA_expr_SNpc_VTA <- AHBA_expr_SNpc_VTA %>% data.frame %>% filter(AHBA_expr_SNpc_VTA$probe_id %in% probeInfo$probe_id)
rownames(AHBA_expr_SNpc_VTA) <- AHBA_expr_SNpc_VTA$probe_id

AHBA_expr_SNpc_VTA <- AHBA_expr_SNpc_VTA %>% data.frame %>% filter(AHBA_expr_SNpc_VTA$probe_id %in% probeInfo$probe_id)
AHBA_expr_SNpc_VTA <- AHBA_expr_SNpc_VTA %>% distinct(Gene, .keep_all = TRUE)
rownames(AHBA_expr_SNpc_VTA) <- AHBA_expr_SNpc_VTA$Gene
```


  Defining groups (0 = VTA, SNpc = 1) and probesets, constructing reference signals, creating a function to perform PSEA analysis and performing PSEA analyses. The linear models constructed for each gene are filtered for only models that are at least 50% accurate (adjusted R2 > 0.5). The function created, PSEA_cell_type_specific_mDBR, returns a data frame containing all genes exhibiting a significant population-specific fold change in the SNpc relative to the VTA.
```{r}
library(PSEA)
library(dplyr)

groups_mDBR <- c(1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0)

VTA_samples <- AHBA_expr_SNpc_VTA[, c(3, 4, 7, 8, 10, 12, 14, 16)]

# Astrocyte
astro_probesets_mDBR <- list("GFAP", "AQP4", "GJA1")
astro_reference_mDBR <- marker(as.matrix(AHBA_expr_SNpc_VTA[, 1:16]), astro_probesets_mDBR)
astro_difference_mDBR <- groups_mDBR * astro_reference_mDBR

# Oligodendrocyte
oligo_probesets_mDBR <- list("MBP", "MOBP", "MOG", "MAG")
oligo_reference_mDBR <- marker(as.matrix(AHBA_expr_SNpc_VTA[, 1:16]), oligo_probesets_mDBR)
oligo_difference_mDBR <- groups_mDBR * oligo_reference_mDBR

# Microglia
micro_probesets_mDBR <- list("AIF1")
micro_reference_mDBR <- marker(as.matrix(AHBA_expr_SNpc_VTA[, 1:16]), micro_probesets_mDBR)
micro_difference_mDBR <- groups_mDBR * micro_reference_mDBR

# Dopaminergic neuron
DA_probesets_mDBR <- list("TH", "SLC6A3")
DA_reference_mDBR <- marker(as.matrix(AHBA_expr_SNpc_VTA[, 1:16]), DA_probesets_mDBR)
DA_difference_mDBR <- groups_mDBR * DA_reference_mDBR

# Glutamatergic neuron
Glut_probesets_mDBR <- list("SLC17A7", "SLC17A6")
Glut_reference_mDBR <- marker(as.matrix(AHBA_expr_SNpc_VTA[, 1:16]), Glut_probesets_mDBR)
Glut_difference_mDBR <- groups_mDBR * Glut_reference_mDBR

# GABAergic neuron
GABA_probesets_mDBR <- list("GAD1", "GAD2", "SLC32A1")
GABA_reference_mDBR <- marker(as.matrix(AHBA_expr_SNpc_VTA[, 1:16]), GABA_probesets_mDBR)
GABA_difference_mDBR <- groups_mDBR * GABA_reference_mDBR


PSEA_cell_type_specific_mDBR <- function(reference, difference) {
  
  AHBA_expr_SNpc_VTA_tranposed <- t(AHBA_expr_SNpc_VTA[, 1:16])
  
  models <- list()
  for (i in rownames(AHBA_expr_SNpc_VTA[, 1:16])) {
    models[[i]] <- lm(AHBA_expr_SNpc_VTA_tranposed[,i] ~ reference + difference)
  }
  
  reg <- c("(Intercept)", "reference", "difference")
  coef <- coefmat(models, reg)
  p <- pvalmat(models, reg)
  summaries <- lapply(models, summary)
  adj_R2 <- slt(summaries, 'adj.r.squared')

  # avg_expr <- apply(VTA_samples, 1, mean)
  filt <- adj_R2 > 0.5 # & (coef[,1] / avg_expr) < 0.75

  specific <- which(filt & p[, 2] < 0.05 & p[, 3] < 0.05 & coef[, 2] > 0)
  specific <- data.frame(specific)
  
  specific$Gene <- rownames(specific)
  
  coef <- data.frame(coef)
  coef$num <- as.numeric(rownames(coef))
  
  specific <- specific %>% inner_join(coef, by = c("specific" = "num"))
  specific <- specific[,-c(1,3)]
  specific$fold.change <- ((specific[,2] + specific[,3]) / specific[,2])
  
  return(specific)
}

AHBA_expr_SNpc_VTA_tranposed <- t(AHBA_expr_SNpc_VTA[, 1:16])

DA_models_mDBR <- list()
for (i in rownames(AHBA_expr_SNpc_VTA[, 1:16])) {
  DA_models_mDBR[[i]] <- lm(AHBA_expr_SNpc_VTA_tranposed[,i] ~ DA_reference_mDBR + DA_difference_mDBR)
}

astro_specific_mDBR <- PSEA_cell_type_specific_mDBR(astro_reference_mDBR, astro_difference_mDBR)
oligo_specific_mDBR <- PSEA_cell_type_specific_mDBR(oligo_reference_mDBR, oligo_difference_mDBR)
micro_specific_mDBR <- PSEA_cell_type_specific_mDBR(micro_reference_mDBR, micro_difference_mDBR)
DA_specific_mDBR <- PSEA_cell_type_specific_mDBR(DA_reference_mDBR, DA_difference_mDBR)
Glut_specific_mDBR <- PSEA_cell_type_specific_mDBR(Glut_reference_mDBR, Glut_difference_mDBR)
GABA_specific_mDBR <- PSEA_cell_type_specific_mDBR(GABA_reference_mDBR, GABA_difference_mDBR)
```

____________________________________________________________________________________________

  Finding genes that are differentially expressed in the SNpc in dopaminergic neurons and one other cell type.

```{r}
library(dplyr)

astro_DA_specific_mDBR <- intersect(DA_specific_mDBR$Gene, astro_specific_mDBR$Gene) %>% data.frame
oligo_DA_specific_mDBR <- intersect(DA_specific_mDBR$Gene, oligo_specific_mDBR$Gene) %>% data.frame
micro_DA_specific_mDBR <- intersect(DA_specific_mDBR$Gene, micro_specific_mDBR$Gene) %>% data.frame
Glut_DA_specific_mDBR <- intersect(DA_specific_mDBR$Gene, Glut_specific_mDBR$Gene) %>% data.frame
GABA_DA_specific_mDBR <- intersect(DA_specific_mDBR$Gene, GABA_specific_mDBR$Gene) %>% data.frame

```

____________________________________________________________________________________________
____________________________________________________________________________________________

VALIDATING RESULTS - UKBEC AGE CORRELATIONS IN OTHER BRAIN REGIONS
____________________________________________________________________________________________
____________________________________________________________________________________________

November 30, 2020
  
  In order to ensure that the enrichment for genes downregulated with age for the melanosome and pigment granule cellular components is specific to the melanized substantia nigra, age correlations were also performed for all other brain regions provided in the UKBEC database. This included the cerebellar cortex, the frontal cortex, the hippocampus, the medulla, the occipital cortex, the putamen, the temporal cortex, the thalamus, and white matter. The data frame containing sample annotations and expression data for all UKBEC samples (gene_filt) was used to filter for each brain region and perform correlation analyses. One function (age_correlations) was created to perform age correlations for each of the brain regions. It outputs data frames in a list, with the first being positively significantly correlated genes, and the second being negatively significantly correlated genes.
  
```{r}
# library(dplyr)
# library(WGCNA)
# 
# age_correlations <- function(brain_region) {
#   
#   # Filter for samples from that brain region
#   gene_filt_brain_region <- gene_filt %>% filter(V26494 == brain_region)
# 
#   # extract age as its own numeric vector
#   age_gene_brain_region <- extract_age_gene(gene_filt_brain_region)
# 
#   # Call convert_to_numeric function, with parameter being gene_filt_brain_region without the brain region and age columns
#   gene_filt_brain_region <- convert_to_numeric(gene_filt_brain_region[,-c(26494,26495)])
# 
#   gene_filt_brain_region <- gene_filt_brain_region %>% t %>% data.frame
#   gene_filt_brain_region$probe_set_ID <- gene_names[, "ID"]
#   gene_filt_brain_region$gene_assignment <- gene_names[, "gene_assignment"]
# 
#   # Remove probe sets not assigned to a gene first
#   gene_filt_brain_region <- gene_filt_brain_region %>% filter(gene_filt_brain_region$gene_assignment != "---")
# 
# 
#   # Align probe sets to genes
#   for (i in 1:dim(gene_filt_brain_region)[1]) {
#     gene_filt_brain_region[i,  "gene_assignment"] <- strsplit(gene_filt_brain_region[i, "gene_assignment"], " ")[[1]][3]
#   }
#   rownames(gene_filt_brain_region) <- gene_filt_brain_region$probe_set_ID
#   probe2gene_UKBEC_brain_region <- collapseRows(gene_filt_brain_region[, -(dim(gene_filt_brain_region)[2])],
#                                    rowGroup = gene_filt_brain_region$gene_assignment,
#                                    rowID = gene_filt_brain_region$probe_set_ID,
#                                    method = "maxRowVariance",
#                                    connectivityBasedCollapsing = TRUE)
#   gene_filt_brain_region <- probe2gene_UKBEC_brain_region[["datETcollapsed"]] %>% data.frame
# 
#   # Create a data frame that will store correlation coefficients and p values for each gene 
#   all_age_correlations_brain_region <- matrix(nrow = dim(gene_filt_brain_region)[1], ncol = 5) %>% data.frame
#   rownames(all_age_correlations_brain_region) <- gene_filt_brain_region$probe_set_ID
#   all_age_correlations_brain_region[,1] <- rownames(gene_filt_brain_region)
#   all_age_correlations_brain_region[,2] <- gene_filt_brain_region$probe_set_ID
#   colnames(all_age_correlations_brain_region) <- c("Gene", "ID", "Correlation Coefficient", "Raw p-value", "BH-corrected p-value")
# 
#   # Compute age correlations for all genes and store raw p-values and correlation coefficients
#   for (i in 1:dim(gene_filt_brain_region)[1]) {
#   
#     cor <- cor.test(as.numeric(gene_filt_brain_region[i, -(dim(gene_filt_brain_region)[2])]), age_gene_brain_region, method = "pearson")
#     all_age_correlations_brain_region[i, "Correlation Coefficient"] <- cor[["estimate"]][["cor"]]
#     all_age_correlations_brain_region[i, "Raw p-value"] <- cor[["p.value"]]
#   
#   }
# 
#   # Adjust p-values using Benjamini-Hochberg (BH) procedure
#   all_age_correlations_brain_region[, "BH-corrected p-value"] <- p.adjust(all_age_correlations_brain_region[, "Raw p-value"], method = "BH")
# 
#   # Filter for genes significantly correlated with age
#   sig_age_correlations_brain_region <- all_age_correlations_brain_region %>% 
#     filter(all_age_correlations_brain_region[, "BH-corrected p-value"] < 0.05)
# 
#   # Filter for genes with positive correlation coefficient
#   pos_sig_age_correlations_brain_region <- sig_age_correlations_brain_region %>% 
#     filter(sig_age_correlations_brain_region[, "Correlation Coefficient"] > 0)
# 
#   # Filter for genes with negative correlation coefficient
#   neg_sig_age_correlations_brain_region <- sig_age_correlations_brain_region %>% 
#     filter(sig_age_correlations_brain_region[, "Correlation Coefficient"] < 0)
# 
#   return(list(pos_sig_age_correlations_brain_region, neg_sig_age_correlations_brain_region))
#   
# }
# 
# # Cerebellar cortex
# cerebellar_cortex_cor <- age_correlations("brain region: cerebellar cortex")
# write.csv(x=cerebellar_cortex_cor[[1]], file="pos_cerebellar_cortex.csv")
# write.csv(x=cerebellar_cortex_cor[[2]], file="neg_cerebellar_cortex.csv")
# 
# # Frontal cortex
# frontal_cortex_cor <- age_correlations("brain region: frontal cortex")
# write.csv(x=frontal_cortex_cor[[1]], file="pos_frontal_cortex.csv")
# write.csv(x=frontal_cortex_cor[[2]], file="neg_frontal_cortex.csv")
# 
# # Hippocampus
# hippo_cor <- age_correlations("brain region: hippocampus")
# write.csv(x=hippo_cor[[1]], file="pos_hippocampus.csv")
# write.csv(x=hippo_cor[[2]], file="neg_hippocampus.csv")
# 
# # Medulla
# medulla_cor <- age_correlations("brain region: medulla")
# write.csv(x=medulla_cor[[1]], file="pos_medulla.csv")
# write.csv(x=medulla_cor[[2]], file="neg_medulla.csv")
# 
# # Occipital cortex
# occipital_cortex_cor <- age_correlations("brain region: occipital cortex")
# write.csv(x=occipital_cortex_cor[[1]], file="pos_occipital_cortex.csv")
# write.csv(x=occipital_cortex_cor[[2]], file="neg_occipital_cortex.csv")
# 
# # Putamen
# putamen_cor <- age_correlations("brain region: putamen")
# write.csv(x=putamen_cor[[1]], file="pos_putamen.csv")
# write.csv(x=putamen_cor[[2]], file="neg_putamen.csv")
# 
# # Temporal Cortex
# temporal_cortex_cor <- age_correlations("brain region: temporal cortex")
# write.csv(x=temporal_cortex_cor[[1]], file="pos_temporal_cortex.csv")
# write.csv(x=temporal_cortex_cor[[2]], file="neg_temporal_cortex.csv")
# 
# # Thalamus
# thalamus_cor <- age_correlations("brain region: thalamus")
# write.csv(x=thalamus_cor[[1]], file="pos_thalamus.csv")
# write.csv(x=thalamus_cor[[2]], file="neg_thalamus.csv")
# 
# # White Matter
# white_matter_cor <- age_correlations("brain region: white matter")
# write.csv(x=white_matter_cor[[1]], file="pos_white_matter.csv")
# write.csv(x=white_matter_cor[[2]], file="neg_white_matter.csv")

```

  GO-term enrichment analysis results (using PANTHER DB, search for biological process, molecular function, and cellular component) are listed below. The search was done to determine whether the melanosome/pigment granule enrichment, the astrocyte projection enrichment, and the enrichment for mitochondria-related terms (for example, mitochondrial respiratory chain complex I) was specific to the substantia nigra or if it was observed in any other brain regions as well.
  
Cerebellar cortex:
  Positively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms; enrichment for mitochondrial membrane only (cellular component), despite no enrichment for biological process GO terms in that area
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms, or mitochondria-related terms
  
Frontal cortex:
  Positively correlated: No statistically significant results altogether
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms, or mitochondria-related terms; only significant results altogether were for synaptic signaling (biological process), and synapse, dendrite, and neuronal cell body terms
  
Hippocampus:
  Positively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms, or mitochondria-related terms
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms; enrichment for mitochondrial envelope cellular component term, despite no enrichment for biological process GO terms in that area
  
Medulla:
  Positively correlated: No statistically significant results altogether
  Negatively correlated: No statistically significant results altogether
  
Occipital cortex:
  Positively correlated: No enrichment for melanosome/pigment granule-related terms or mitochondria-related terms; enrichment for astrocyte projection cellular component GO term
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms, or mitochondria-related terms
  
Putamen:
  Positively correlated: No statistically significant results altogether
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms; enriched for astrocyte end-foot, astrocyte projection, mitochondrial membrane, and mitochondrial matrix cellular component terms, despite no enrichment for biological process GO terms in those areas
  
Temporal cortex:
  Positively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms, or mitochondria-related terms
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms, or mitochondria-related terms
  
Thalamus:
  Positively correlated: No statistically significant results altogether
  Negatively correlated: No statistically significant results altogether
  
White matter:
  Positively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms; enriched for mitochondrion cellular component term only, despite no enrichment for biological process GO terms in that area
  Negatively correlated: No enrichment for melanosome/pigment granule-related terms, astrocyte-related terms; negatively enriched for mitochondrial matrix cellular component term
  
  
  Age correlations analysis of other brain regions revealed that the enrichment of genes negatively correlated with age in the substantia nigra for the melanosome, pigment granules, and for mitochondrial biological processes is indeed unique to the substantia nigra.

____________________________________________________________________________________________
____________________________________________________________________________________________

VALIDATING RESULTS - AGE-SPECIFIC EXPRESSION CHANGES IN PARKINSON'S DISEASE
____________________________________________________________________________________________
____________________________________________________________________________________________

December 3, 2020

Multiple linear regression analysis of Parkinson's disease gene expression in the substantia nigra

  The data sets GSE8397 and GSE20292 are to be re-analyzed using the multiple linear regression method, and a new data set, GSE49036, will be analyzed with this method as well. THe PSEA library can be used for this (although PSEA is not actually being performed), as a lot of the function are still applicable in this case.
  
```{r}
library(GEOquery)
library(dplyr)

GSE49036 <- getGEO("GSE49036", destdir = '/Users/rezaadibnia/Desktop/PD NM Project/', GSEMatrix = TRUE)
GSE49036_exp_gene <- GSE49036[["GSE49036_series_matrix.txt.gz"]]@assayData[["exprs"]]
GSE49036_sample_annotations <- GSE49036[["GSE49036_series_matrix.txt.gz"]]@phenoData@data
GSE49036_gene_IDs <- GSE49036[["GSE49036_series_matrix.txt.gz"]]@featureData@data

# Convert gene expression table to data frame
GSE49036_exp_gene <- data.frame(GSE49036_exp_gene)

# Add column containing gene symbols to gene expression data frame
for (i in 1:dim(GSE49036_exp_gene)[1]) {
  GSE49036_exp_gene[i, "Gene"] <- strsplit(GSE49036_gene_IDs[i,"Gene Symbol"], " ")[[1]][1]
}

# Remove rows (probes) without a gene assignment
print(paste(sum(is.na(GSE49036_exp_gene$Gene)), "probes without gene assignments in GSE49036 out of the total of", dim(GSE49036_exp_gene)[1], "probes"))
GSE49036_exp_gene <- GSE49036_exp_gene %>% filter(is.na(GSE49036_exp_gene$Gene) == FALSE)

# Align probes to genes
probe2gene_GSE49036 <- collapseRows(GSE49036_exp_gene[, -29], 
                           rowGroup = GSE49036_exp_gene$Gene, 
                           rowID = rownames(GSE49036_exp_gene),
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
GSE49036_exp_gene <- probe2gene_GSE49036[["datETcollapsed"]]
print(paste(dim(GSE49036_exp_gene)[1], "genes were aligned to 54675 probes in GSE49036"))

# Add columns that for type of sample and age to gene expression data frame
GSE49036_exp_gene <- data.frame(GSE49036_exp_gene)
GSE49036_exp_gene <- add_row(GSE49036_exp_gene, .before = 1)
GSE49036_exp_gene <- add_row(GSE49036_exp_gene, .before = 1)
GSE49036_exp_gene[1,] <- GSE49036_sample_annotations[,"disease state:ch1"]
GSE49036_exp_gene[2,] <- as.numeric(GSE49036_sample_annotations[,"rin:ch1"])
rownames(GSE49036_exp_gene)[1:2] <- c("sample_type", "age")

# Extract age as its own vector
GSE49036_exp_gene <- GSE49036_exp_gene %>% t %>% data.frame
GSE49036_age_gene <- as.numeric(GSE49036_exp_gene$age)

# Define sample groups
groups_GSE8397 <- GSE8397_exp_gene$sample_type
groups_GSE20292 <- GSE20292_exp_gene$sample_type
groups_GSE49036 <- GSE49036_exp_gene$sample_type

# Convert data frames to numeric data frame
GSE8397_exp_gene <- convert_to_numeric(GSE8397_exp_gene[,-c(1:2)])
GSE20292_exp_gene <- convert_to_numeric(GSE20292_exp_gene[,-c(1:2)])
GSE49036_exp_gene <- convert_to_numeric(GSE49036_exp_gene[,-c(1:2)])

```

  Preprocessing the data
```{r}
library(PSEA)

# Remove last three samples in GSE20292
GSE20292_exp_gene <- GSE20292_exp_gene[-c(27:29),]
GSE20292_age_gene <- GSE20292_age_gene[-c(27:29)]
groups_GSE20292 <- groups_GSE20292[-c(27:29)]

# Remove iLBD samples from GSE49036
GSE49036_exp_gene <- GSE49036_exp_gene[-c(9:13),]
GSE49036_age_gene <- GSE49036_age_gene[-c(9:13)]
groups_GSE49036 <- groups_GSE49036[-c(9:13)]

# Remove non-SN samples from GSE8397
GSE8397_exp_gene<- GSE8397_exp_gene[-c(1:8, 14, 24, 31, 47),]
GSE8397_age_gene <- GSE8397_age_gene[-c(1:8, 14, 24, 31, 47)]
groups_GSE8397 <- groups_GSE8397[-c(1:8, 14, 24, 31, 47)]

# Define groups as 0 (control) and 1 (PD)
groups_GSE8397 <- groups_GSE8397[-c(1:8, 14, 24, 31, 47)]
groups_GSE8397[c(1:5, 15:20)] <- 0
groups_GSE8397[c(6:14, 21:35)] <- 1
for (i in 1:length(groups_GSE20292)) {
  if (groups_GSE20292[i] == "Control") {
    groups_GSE20292[i] = 0
  } else if (groups_GSE20292[i] == "Parkinsons disease") {
    groups_GSE20292[i] = 1
  }
}
for (i in 1:length(groups_GSE49036)) {
  if (groups_GSE49036[i] == "control") {
    groups_GSE49036[i] = 0
  } else if (groups_GSE49036[i] == "Parkinson's disease") {
    groups_GSE49036[i] = 1
  }
}

```


  Defining a function that will construct the linear regression models for each gene in each data set
```{r}

compute_PD_age_correlations <- function(expr, age_vector, groups) {
  
  groups <- as.numeric(groups)
  age_difference <- groups * age_vector
  models <- list()
  for (i in colnames(expr)) {
    models[[i]] <- lm(expr[,i] ~ age_vector + age_difference)
  }
  return(models)
}

```


  Running function to create models and extracting relevant information to filter models
```{r}
library(PSEA)

age_models_GSE8397 <- compute_PD_age_correlations(GSE8397_exp_gene, GSE8397_age_gene, groups_GSE8397)
age_models_GSE20292 <- compute_PD_age_correlations(GSE20292_exp_gene, GSE20292_age_gene, groups_GSE20292)
age_models_GSE49036 <- compute_PD_age_correlations(GSE49036_exp_gene, GSE49036_age_gene, groups_GSE49036)

regressor_names_PD <- c("(Intercept)", "age_vector", "age_difference")
extract_PD_models_info <- function(models, expr) {
  
  coefs <- coefmat(models, regressor_names_PD)
  pval <- pvalmat(models, regressor_names_PD)
  models_summary <- lapply(models, summary)
  adjusted_R2 <- slt(models_summary, 'adj.r.squared')

  # avg_expression <- apply(t(expr), 1, mean)
  # filter <- (adjusted_R2 > 0.5) & ((coefs[,1] / avg_expression) < 0.75)

  age_specific <- which(adjusted_R2 > 0.5 & pval[, 2] < 0.05 & pval[, 3] < 0.05)
  # age_specific_not_PD <- which(filter & pval[, 2] < 0.05)
  
  age_specific <- data.frame(age_specific)
  genes <- rownames(age_specific)
  coefs <- coefs %>% data.frame
  coefs$num <- as.numeric(rownames(coefs))

  age_specific <- data.frame(age_specific) %>% inner_join(coefs, by = c("age_specific" = "num"))
  
  age_specific$fold.change <- ((age_specific$coef.age_vector + age_specific$coef.age_difference) / age_specific$coef.age_vector)

  age_specific$Gene <- genes
  
  age_specific <- age_specific[, -c(1,2)]
  
  return(age_specific)
}

age_specific_GSE8397 <- extract_PD_models_info(age_models_GSE8397, GSE8397_exp_gene)
age_specific_GSE20292 <- extract_PD_models_info(age_models_GSE20292, GSE20292_exp_gene)
age_specific_GSE49036 <- extract_PD_models_info(age_models_GSE49036, GSE49036_exp_gene)

# print(paste((dim(age_specific_GSE8397)[1] + dim(age_specific_GSE20292)[1] + dim(age_specific_GSE49036)[1]), "genes exhibiting age-dependent differential expression in Parkinson's disease"))

```

  Analysis was also done using the B-chip of GSE8397, which contains additional genes
```{r}
library(dplyr)

# B-chip of GSE8397 (contains more genes)

GSE8397_exp_gene_2 <- GSE8397[["GSE8397-GPL97_series_matrix.txt.gz"]]@assayData[["exprs"]]
GSE8397_sample_annotations_2 <- GSE8397[["GSE8397-GPL97_series_matrix.txt.gz"]]@phenoData@data
GSE8397_gene_IDs_2 <- GSE8397[["GSE8397-GPL97_series_matrix.txt.gz"]]@featureData@data

# Convert gene expression table to data frame
GSE8397_exp_gene_2 <- data.frame(GSE8397_exp_gene_2)

# Add column containing gene symbols to gene expression data frame
for (i in 1:dim(GSE8397_exp_gene_2)[1]) {
  GSE8397_exp_gene_2[i, "Gene"] <- strsplit(GSE8397_gene_IDs_2[i,"Gene Symbol"], " ")[[1]][1]
}

# Remove rows (probes) without a gene assignment
print(paste(sum(is.na(GSE8397_exp_gene_2$Gene)), "probes without gene assignments in GSE8397 Chip B out of the total of", dim(GSE8397_exp_gene_2)[1], "probes"))
GSE8397_exp_gene_2 <- GSE8397_exp_gene_2 %>% filter(is.na(GSE8397_exp_gene_2$Gene) == FALSE)

# Align probes to genes
probe2gene_GSE8397_2 <- collapseRows(GSE8397_exp_gene_2[, -48], 
                           rowGroup = GSE8397_exp_gene_2$Gene, 
                           rowID = rownames(GSE8397_exp_gene_2),
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
GSE8397_exp_gene_2 <- probe2gene_GSE8397_2[["datETcollapsed"]]
print(paste(dim(GSE8397_exp_gene_2)[1], "genes were aligned to 17023 probes in GSE8397 Chip B"))

# Add columns that for type of sample and age to gene expression data frame
GSE8397_exp_gene_2 <- data.frame(GSE8397_exp_gene_2)
GSE8397_exp_gene_2 <- add_row(GSE8397_exp_gene_2, .before = 1)
GSE8397_exp_gene_2 <- add_row(GSE8397_exp_gene_2, .before = 1)
GSE8397_exp_gene_2[1,] <- GSE8397_sample_annotations_2[,"title"]
GSE8397_exp_gene_2[2,] <- as.numeric(str_sub(GSE8397_sample_annotations_2[,"age:ch1"], 1, 2))
rownames(GSE8397_exp_gene_2)[1:2] <- c("sample_type", "age")

# Extract age as its own vector
GSE8397_exp_gene_2 <- GSE8397_exp_gene_2 %>% t %>% data.frame
GSE8397_age_gene_2 <- as.numeric(GSE8397_exp_gene_2$age)

# Remove non-SN samples from GSE8397_2
GSE8397_exp_gene_2 <- GSE8397_exp_gene_2[-c(1:8, 14, 24, 31, 47),]
GSE8397_age_gene_2 <- GSE8397_age_gene_2[-c(1:8, 14, 24, 31, 47)]
groups_GSE8397_2 <- GSE8397_exp_gene_2$sample_type

# Convert to numeric
GSE8397_exp_gene_2 <- convert_to_numeric(GSE8397_exp_gene_2[,-c(1:2)])

# Define groups as 0 (control) and 1 (PD)
groups_GSE8397_2[c(1:5, 15:20)] <- 0
groups_GSE8397_2[c(6:14, 21:35)] <- 1

# Perform analysis
age_models_GSE8397_2 <- compute_PD_age_correlations(GSE8397_exp_gene_2, GSE8397_age_gene_2, groups_GSE8397_2)
age_specific_GSE8397_2 <- extract_PD_models_info(age_models_GSE8397_2, GSE8397_exp_gene_2)

# print(paste((dim(age_specific_GSE8397)[1] + dim(age_specific_GSE8397_2)[1] + dim(age_specific_GSE20292)[1] + dim(age_specific_GSE49036)[1]), "genes exhibiting age-dependent differential expression in Parkinson's disease"))

# All age-specific differential expressed genes
age_specific_PD <- bind_rows(age_specific_GSE8397, age_specific_GSE8397_2, age_specific_GSE20292, age_specific_GSE49036)
age_specific_PD <- age_specific_PD[-c(12, 79),]

print(paste(dim(age_specific_PD)[1], "genes exhibiting age-dependent differential expression in Parkinson's disease"))

```


  CUX2
```{r}
summary(age_models_GSE8397[["CUX2"]])
crplot(age_models_GSE8397[["CUX2"]], "age_vector", g = "age_difference")
```
____________________________________________________________________________________________
____________________________________________________________________________________________

UKBEC AGE CORRELATIONS PSEA
____________________________________________________________________________________________
____________________________________________________________________________________________

December 25, 2020

UKBEC age correlations PSEA

  The difference between all previous analyses and PSEA is that the calculation of reference signals for PSEA is more accurate when multiple probes are used for the same gene. Therefore, since gene_filt_SN had, at this point, already had probe-to-gene alignment done, it had to be recreated without probe-to-gene alignment to construct reference signals.

```{r}

## Same process as before used to filter for substantia nigra expression data only, except this time not aligning probes to genes yet

# Filter for only substantia nigra (SN) gene expression data
gene_filt_SN <- gene_filt %>% 
  filter(V26494 == 'brain region: substantia nigra')

# Call convert_to_numeric function, with parameter being gene_filt_SN without the brain region and age columns
gene_filt_SN <- convert_to_numeric(gene_filt_SN[,-c(26494,26495)])

gene_filt_SN <- gene_filt_SN %>% t %>% data.frame
gene_filt_SN$probe_set_ID <- gene_names[, "ID"]
gene_filt_SN$gene_assignment <- gene_names[, "gene_assignment"]

# Remove probe sets not assigned to a gene
gene_filt_SN <- gene_filt_SN %>% filter(gene_filt_SN$gene_assignment != "---")

# Simplify gene assignments of each probe to just gene names
for (i in 1:dim(gene_filt_SN)[1]) {
  gene_filt_SN[i,  "gene_assignment"] <- strsplit(gene_filt_SN[i, "gene_assignment"], " ")[[1]][3]
}

# Make rownames probe set IDs to simplify probe-to-gene alignment later on
rownames(gene_filt_SN) <- gene_filt_SN$probe_set_ID


```


  To ensure that selection of marker genes was appropriate, it needed to be ensured that marker genes for the same cell types were highly correlated, and marker genes for different cell types were not correlated. This was done for astrocytes, oligodendrocytes, microglia, dopaminergic neurons, and GABAergic neurons. A correlation matrix was made to visualize this.
```{r}
library(Hmisc)
library(corrplot)
library(hash)

# Define probesets
astro_probesets <- list(3759410, 3733590, 3802396, 2923661)
oligo_probesets <- list(c(3814033, 3814063, 3814143), 2618407, 3918369, 2900940, 3830320)
micro_probesets <- list(2902444, 2316379, 3656990)
DA_probesets <- list(3359180, 2845921, 3900833, 3044518, 3304265)
GABA_probesets <- list(2514969, 3239667, 3884793)
all_probesets <- c(3759410, 3733590, 3802396, 2923661, 3814033, 3814063, 3814143, 2618407, 3918369, 2900940, 3830320, 2902444, 2316379, 3656990, 3359180, 2845921, 3900833, 3044518, 3304265, 2514969, 3239667, 3884793)

# Correlation matrix
cor_matrix_marker <- gene_filt_SN[as.character(all_probesets),1:101] %>% t %>% as.matrix %>% rcorr
cor_matrix_marker

# Visualization of correlation matrix, with insignificant correlations left blank
corrplot(cor_matrix_marker$r, type="upper", order="original", 
         p.mat = cor_matrix_marker$P, sig.level = 0.01, insig = "blank")

```
  
  From the correlation matrix, it is evident that probe 3814143, which is supposed to be a marker for oligodendrocytes, is highly correlated with probe 3759410, which is an astrocytes marker. It appears that both of those probes are also highly correlated with a number of other probes not within their cell type group, which is undesirable, therefore those probes were removed. Additionally, probe 2316379 is supposed to be a marker for microglia, but it is not significantly corerlated with either of the other two marker probes, therefore it was removed as well. Finally, probe 3044518, a marker probe for dopaminergic neurons, it actually negatively correlated with other dopaminergic neuron markers, therefore it was removed. The new correlation matrix looks like so:
  
```{r}
all_probesets <- c(3733590, 3802396, 2923661, 3814033, 3814063, 2618407, 3918369, 2900940, 3830320, 2902444, 3656990, 3359180, 2845921, 3900833, 3304265, 2514969, 3239667, 3884793)
cor_matrix_marker <- gene_filt_SN[as.character(all_probesets),1:101] %>% t %>% as.matrix %>% rcorr
corrplot(cor_matrix_marker$r, type="upper", order="original", 
         p.mat = cor_matrix_marker$P, sig.level = 0.01, insig = "blank")
```
  This correlation matrix is much cleaner and shows the expected "triangle" pattern of correlations of probes from the same marker gene, which minimal outside correlations, most of which have relatively small correlation coefficients.


  Redefined probe sets for each cell type and calculating reference signals. Neuron reference signal was removed because of overlap with reference signals of other neurons. Additionally, the serotonergic, noradrenergic, and glutamatergic reference signals are unnecessary as the substantia nigra is known to consist mostly of dopaminergic neurons (in the pars compacta) and GABAergic neurons (in the pars reticulata).
```{r}
library(PSEA)

# # Neuron
# neuro_probesets <- list(3772831, 3128271, 3403015, c(3887232, 3887241), 3378191, 4027056)
# neuro_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), neuro_probesets)

# Astrocyte
astro_probesets <- list(3733590, 3802396, 2923661)
astro_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), astro_probesets)

# Oligodendrocyte
oligo_probesets <- list(c(3814033, 3814063), 2618407, 3918369, 2900940, 3830320)
oligo_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), oligo_probesets)

# Microglia
micro_probesets <- list(2902444, 3656990)
micro_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), micro_probesets)

# Dopaminergic neuron
DA_probesets <- list(3359180, 2845921, 3900833, 3304265)
DA_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), DA_probesets)

# # Glutamatergic neuron
# Glut_probesets <- list(3867842, 3323824)
# Glut_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), Glut_probesets)

# GABAergic neuron
GABA_probesets <- list(2514969, 3239667, 3884793)
GABA_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), GABA_probesets)

# # Serotonergic
# sero_probesets <- list(2934521, 2599647, 3422386, 2789690)
# sero_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), sero_probesets)

# # Noradrenergic
# noradr_probesets <- list(3661766, 3769779, 3246182)
# noradr_reference <- marker(as.matrix(gene_filt_SN[, 1:101]), noradr_probesets)

```


  Now, after creating reference signals, we align probes to genes in gene_filt_SN
```{r}
library(WGCNA)
library(dplyr)

probe2gene_UKBEC <- collapseRows(gene_filt_SN[, -103],
                           rowGroup = gene_filt_SN$gene_assignment,
                           rowID = gene_filt_SN$probe_set_ID,
                           method = "maxRowVariance",
                           connectivityBasedCollapsing = TRUE)
gene_filt_SN <- probe2gene_UKBEC[["datETcollapsed"]] %>% data.frame

```


  Performing deconvolution of expression profiles:
  In this case, because genes were previously correlated with age, the models constructed do not just include cell type; they also include age of the sample (age_gene_SN is the sixth regressor).
```{r}
library(PSEA)
library(dplyr)
library(hash)

model_matrix_UKBEC <- cbind(intercept=1, astro_reference, oligo_reference, micro_reference, DA_reference, GABA_reference, age_gene_SN)

model_subset_UKBEC <- em_quantvg(c(2,3,4,5,6,7), tnv=6, ng=1)

# Models only for genes significantly correlated with age; including all genes exhausts memory and is unnecessary anyway
models_UKBEC <- lmfitst(t(gene_filt_SN[rownames(gene_filt_SN) %in% sig_age_correlations$Gene, 1:101]), model_matrix_UKBEC, model_subset_UKBEC)

models_UKBEC_all <- lmfitst(t(gene_filt_SN[, 1:101]), model_matrix_UKBEC, model_subset_UKBEC)

```


  The function, lmfitst, provides two lists; the second provides the best model for each gene as determined by AIC. Therefore, for example, the best model for the DDC gene can be called using the following code:
```{r}

summary(models_UKBEC[[2]][["OCA2"]]) # best model for OCA2

```


  We also want to select our models based on other criteria, including the adjusted coefficient of determination (adjusted R^2), the relative size of the intercept, and the sign of the coefficients [16].
```{r}

regressor_names_UKBEC <- as.character(1:7)
coefficients_UKBEC <- coefmat(models_UKBEC[[2]], regressor_names_UKBEC)
pvalues_UKBEC <- pvalmat(models_UKBEC[[2]], regressor_names_UKBEC)
models_summary_UKBEC <- lapply(models_UKBEC[[2]], summary)
adjusted_R2_UKBEC <- slt(models_summary_UKBEC, 'adj.r.squared')

average_expression_UKBEC <- apply(gene_filt_SN[rownames(gene_filt_SN) %in% sig_age_correlations$Gene, 1:101], 1, mean)
filter <- adjusted_R2_UKBEC > 0.5 & (coefficients_UKBEC[,1] / average_expression_UKBEC < 0.5)

DA_specific_UKBEC <- which(filter & pvalues_UKBEC[, 5] < 0.05 & coefficients_UKBEC[, 5] > 0)
DA_specific_UKBEC <- DA_specific_UKBEC %>% data.frame



# Selecting all genes with dopaminergic neuron-specific expression in UKBEC (not just ACGs)
regressor_names_UKBEC_all <- as.character(1:7)
coefficients_UKBEC_all <- coefmat(models_UKBEC_all[[2]], regressor_names_UKBEC_all)
pvalues_UKBEC_all <- pvalmat(models_UKBEC_all[[2]], regressor_names_UKBEC_all)
models_summary_UKBEC_all <- lapply(models_UKBEC_all[[2]], summary)
adjusted_R2_UKBEC_all <- slt(models_summary_UKBEC_all, 'adj.r.squared')

average_expression_UKBEC_all <- apply(gene_filt_SN[, 1:101], 1, mean)
filter <- adjusted_R2_UKBEC_all > 0.5 & (coefficients_UKBEC_all[,1] / average_expression_UKBEC_all < 0.5)

DA_specific_UKBEC_all <- which(filter & pvalues_UKBEC_all[, 5] < 0.05 & coefficients_UKBEC_all[, 5] > 0)
DA_specific_UKBEC_all <- DA_specific_UKBEC_all %>% data.frame
DA_specific_UKBEC_all[,1] <- rownames(DA_specific_UKBEC_all)

print(paste(dim(DA_specific_UKBEC_all)[1], "genes that exhibit dopaminergic neuron-specific expression in the substantia nigra"))

```
  
  In general, significant models did not include the age regressor, simply because its coefficient was so small compared to the other regressors. Additionally, most DA-specific genes did not exhibit oligodendrocyte-specific expression as well. This indicates that, among genes that are significantly correlated with age, oligodendrocytes and dopaminergic neurons do not have significant interactions with each other.

  Evidently, every gene whose expression could be modeled at least in part by dopaminergic neuron expression also had significant astrocyte-specific expression, and almost every DA neuron-specific gene was also a GABAergic neuron-specific gene. This emphasizes the importance of the astrocyte-dopaminergic neuron interaction, which is reflected in the dopamine oxidation pathway. The only exception to this is AHCYL1, the model for which is shown below.
```{r}
library(PSEA)

summary(models_UKBEC[[2]][["AHCYL1"]])

par(mfrow=c(1,2),cex=0.80)
crplot(models_UKBEC[[2]][["AHCYL1"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC[[2]][["AHCYL1"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression

```

  ALDH2 also appeared to have a significant dopaminergic neuron-specific expression, while an insignificant GABAergic neuron-specific expression.
```{r}
library(PSEA)

summary(models_UKBEC[[2]][["ALDH2"]])

par(mfrow=c(2,2),cex=0.7)
crplot(models_UKBEC[[2]][["ALDH2"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC[[2]][["ALDH2"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression
crplot(models_UKBEC[[2]][["ALDH2"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression
crplot(models_UKBEC[[2]][["ALDH2"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABAergic neuron-specific expression

```


  SDC2 had a very high coefficient for dopaminergic neuron-specific expression:
```{r}
library(PSEA)

summary(models_UKBEC[[2]][["SDC2"]])

par(mfrow=c(2,2),cex=0.7)
crplot(models_UKBEC[[2]][["SDC2"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC[[2]][["SDC2"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression
crplot(models_UKBEC[[2]][["SDC2"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression
crplot(models_UKBEC[[2]][["SDC2"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABAergic neuron-specific expression


```
  
  Of all the genes, however, PYGL had the highest coefficient for dopaminergic neuron-specific expression:
```{r}
library(PSEA)

summary(models_UKBEC[[2]][["PYGL"]])

par(mfrow=c(2,2),cex=0.7)
crplot(models_UKBEC[[2]][["PYGL"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC[[2]][["PYGL"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression
crplot(models_UKBEC[[2]][["PYGL"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression
crplot(models_UKBEC[[2]][["PYGL"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABAergic neuron-specific expression


```
  
  Looking at all of the genes, we can say that few of them were also significantly correlated with age. In fact, every gene whose expression was both dopaminergic-specific and age-dependent exhibited a negative correlation with age.
```{r}

# print(paste(dim(models_DA_filt_UKBEC_names)[1], "genes that exhibit both dopaminergic neuron-specific and age-dependent expression"))
# print(paste(dim((neg_sig_age_correlations$Gene %in% rownames(models_DA_filt_UKBEC_names)) %>% data.frame %>% filter((neg_sig_age_correlations$Gene %in% rownames(models_DA_filt_UKBEC_names)) == TRUE))[1], "genes that exhibit both dopaminergic neuron-specific and negatively-correlated age-dependent expression"))

print(paste(dim(DA_specific_UKBEC)[1], "genes that exhibit both dopaminergic neuron-specific and age-dependent expression"))
print(paste(dim((neg_sig_age_correlations$Gene %in% rownames(DA_specific_UKBEC)) %>% data.frame %>% filter((neg_sig_age_correlations$Gene %in% rownames(DA_specific_UKBEC)) == TRUE))[1], "genes that exhibit both dopaminergic neuron-specific and negatively-correlated age-dependent expression"))

```

  
  It is also important to note that the absence of a number of genes in this analysis is significant. The fact that almost every gene enriched for the melanosome/pigment granule terms (OCA2, CD63, HSP90AA1, GNA13, PRDX1, CCT4, RAB9A, DTNBP1, HSP90B1, SLC1A4, YWHAZ, NMB) did not have significant dopaminergic neuron-specific expression indicates that the enrichment for melanosome/pigment granule-related genes in genes downregulated with age is not specific to dopaminergic neurons. This means that these genes are not involved in age-dependent neuromelanin accumulation.
  
  However, it is important to recognize the two genes that are part of the melanosome/pigment granule GO term, are negatively correlated with age, and exhibit significant dopaminergic-neuron specific expression: CD63 and PRDX1. These two genes may play an inhibitory role in age-dependent neuromelanin accumulation. The two of them have very similar models, both being dependent on astrocytes and microglia as well, and they are highly correlated with each other as well (r = 0.7608129, p < 2.2e-16)
  
```{r}

# Correlation of CD63 and PRDX1
cor.test(as.numeric(gene_filt_SN["CD63", 1:101]), as.numeric(gene_filt_SN["PRDX1", 1:101]), method="pearson")

```
  
  
CD63
```{r}
library(PSEA)

summary(models_UKBEC[[2]][["CD63"]])
crplot(models_UKBEC[[2]][["CD63"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal", ) # DA neuron-specific expression

```
PRDX1
```{r}
library(PSEA)

summary(models_UKBEC[[2]][["PRDX1"]])
crplot(models_UKBEC[[2]][["PRDX1"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal", ) # DA neuron-specific expression

```
  
____________________________________________________________________________________________

December 26, 2020

  A couple of genes that were not correlated with age were selected for further PSEA analysis due to their prevalence in previous literature. This included some of the formerly target genes, including NQO1, GSTM2, SLC18A2, DDC, MAOA, MAOB, and COMT, as well as SNCA, the gene coding for the alpha synuclein protein. Component-to-residual plots of the best models for each of these genes are shown.
  
```{r}
library(PSEA)

# Constructing the models
models_UKBEC_other <- lmfitst(t(gene_filt_SN[c("NQO1", "GSTM2", "SLC18A2", "DDC", "MAOA", "MAOB", "COMT", "SNCA"), 1:101]), model_matrix_UKBEC, model_subset_UKBEC)
# models_UKBEC_confirm <- lmfitst(t(gene_filt_SN[rownames(gene_filt_SN) %in% age_specific_PD$Gene, 1:101]), model_matrix_UKBEC, model_subset_UKBEC)

```


NQO1  
```{r}
library(PSEA)

par(mfrow=c(1,2),cex=0.8)
summary(models_UKBEC_other[[2]][["NQO1"]])
crplot(models_UKBEC_other[[2]][["NQO1"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC_other[[2]][["NQO1"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression

```

GSTM2  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["GSTM2"]])
crplot(models_UKBEC_other[[2]][["GSTM2"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC_other[[2]][["GSTM2"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression
crplot(models_UKBEC_other[[2]][["GSTM2"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # microglia-specific expression
crplot(models_UKBEC_other[[2]][["GSTM2"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABA neuron-specific expression


```


SLC18A2  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["SLC18A2"]])
crplot(models_UKBEC_other[[2]][["SLC18A2"]], "3", newplot = FALSE, xlab = "Oligodendrocyte Reference Signal") # oligodendrocyte-specific expression
crplot(models_UKBEC_other[[2]][["SLC18A2"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression
crplot(models_UKBEC_other[[2]][["SLC18A2"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression
crplot(models_UKBEC_other[[2]][["SLC18A2"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABAergic neuron-specific expression


```


DDC  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["DDC"]])
crplot(models_UKBEC_other[[2]][["DDC"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC_other[[2]][["DDC"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression
crplot(models_UKBEC_other[[2]][["DDC"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABAergic neuron-specific expression
crplot(models_UKBEC_other[[2]][["DDC"]], "7", newplot = FALSE, xlab = "Age") # Age-specific expression

```


MAOA  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["MAOA"]])
crplot(models_UKBEC_other[[2]][["MAOA"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC_other[[2]][["MAOA"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal") # DA neuron-specific expression
crplot(models_UKBEC_other[[2]][["MAOA"]], "7", newplot = FALSE, xlab = "Age") # Age-specific expression

```


MAOB  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["MAOB"]])
crplot(models_UKBEC_other[[2]][["MAOA"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC_other[[2]][["MAOB"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal") # microglia-specific expression
crplot(models_UKBEC_other[[2]][["MAOB"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal") # GABAergic neuron-specific expression


```


COMT  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["COMT"]])
crplot(models_UKBEC_other[[2]][["COMT"]], "2", newplot = FALSE, xlab = "Astrocyte Reference Signal") # astrocyte-specific expression
crplot(models_UKBEC_other[[2]][["COMT"]], "3", newplot = FALSE, xlab = "Oligodendrocyte Reference Signal") # oligodendrocyte-specific expression


```


SNCA  
```{r}
library(PSEA)

par(mfrow=c(2,2),cex=0.7)
summary(models_UKBEC_other[[2]][["SNCA"]])
crplot(models_UKBEC_other[[2]][["SNCA"]], "4", newplot = FALSE, xlab = "Microglial Reference Signal")
crplot(models_UKBEC_other[[2]][["SNCA"]], "5", newplot = FALSE, xlab = "Dopaminergic Neuron Reference Signal")
crplot(models_UKBEC_other[[2]][["SNCA"]], "6", newplot = FALSE, xlab = "GABAergic Neuron Reference Signal")


```
  
  Out of these analyses, the most notable was the dopaminergic neuron-specific expression of SLC18A2 and DDC, both of which had very large slopes (as expected). Otherwise, the results were generally insignificant or not largely important.

________________________________________________________________________________
________________________________________________________________________________

EXTRACTING PD AGE CORRELATIONS FROM ORIGINAL PD DATASET ANALYSIS
________________________________________________________________________________
________________________________________________________________________________

December 12, 2020

  Correlations between age of the patient and gene expression had been previously calculated for only the PD patients in GSE8397 and GSE20292. The genes exhibiting significant correlations were extracted for comparison with ACGs, mDBR genes, genes exhibiting dopaminergic neuron-specific differential expression in the SNpc relative to the VTA, and genes exhibiting age-dependent and dopaminergic neuron-specific expression in the SN. The correlations were filtered based on raw p-values rather than BH-corrected p-values, as BH-correction led to no significant results. Although in theory, BH-correction should decrease type 1 error, with the small number of samples present in GSE8397 and GSE20292, it is already very unlikely that one will obtain very small p-values for any gene. Therefore, it was decided that type 1 error would be increased in order to be able to provide a more descriptive analysis, simply for the purpose of verifying ACGs (the p values of which were BH-corrected).

```{r}
library(dplyr)


GSE8397_sig_cor <- GSE8397_all_cor[,-4] %>% filter(GSE8397_all_cor$`Raw p-value` < 0.05)
GSE20292_sig_cor <- GSE20292_all_cor[,-4] %>% filter(GSE20292_all_cor$`Raw p-value` < 0.05)

# Overlap between age correlations in PD of two data sets
intersect(GSE20292_sig_cor$Gene, GSE8397_sig_cor$Gene)

PD_sig_age_cor <- bind_rows(GSE8397_sig_cor, GSE20292_sig_cor)

```

```{r}

# Overlap between all age correlations in PD and age-correlated genes in healthy individuals
intersect(PD_sig_age_cor$Gene, ACGs$Gene)

```

  Observing that there appear to be more genes correlated with age in PD than in non-neurological individuals (the significantly larger dataset), it now seems unreasonable not to perform BH-correction on the PD age correlations. The incredibly low amount of overlap between the two PD datasets supports this notion. It is probable that the vast majority of genes observed to be correlated with age in PD are type 1 error. Therefore, it was concluded that no meaningful verifications of age correlations could be drawn from the PD datasets.


________________________________________________________________________________
________________________________________________________________________________

VERIFYING ACGs and mDBR genes USING DIFFERENTIAL EXPRESSION ANALYSIS OF PD VS CONTROL
________________________________________________________________________________
________________________________________________________________________________

  Evidently, as age correlations in PD are meaningless due to the small range of patient ages in PD datasets, it may be simply more reasonable and more significant to verify ACGs, mDBR genes, genes exhibiting dopaminergic neuron-specific differential expression in the SNpc relative to the VTA, and genes exhibiting age-dependent and dopaminergic neuron-specific expression in the SN by performing differential expression analysis on the PD datasets. Genes differentially expressed between PD patients and controls would be aligned with ACGs and mDBR genes.
  The analysis on each datasets was completed using GEO2R. The corresponding R script is recorded below.
  
GSE8397 - A chip (GPL96)
```{r}

# Version info: R 3.2.3, Biobase 2.30.0, GEOquery 2.40.0, limma 3.26.8
################################################################
#   Differential expression analysis with limma
# library(GEOquery)
# library(limma)
# library(umap)
# library(dplyr)
# 
# # load series and platform data from GEO
# 
# gset <- getGEO("GSE8397", GSEMatrix =TRUE, AnnotGPL=TRUE)
# if (length(gset) > 1) idx <- grep("GPL96", attr(gset, "names")) else idx <- 1
# gset <- gset[[idx]]
# 
# # make proper column names to match toptable 
# fvarLabels(gset) <- make.names(fvarLabels(gset))
# 
# # group membership for all samples
# gsms <- "XXXXXXXX00000X111111111X000000X111111111111111X"
# sml <- strsplit(gsms, split="")[[1]]
# 
# # filter out excluded samples (marked as "X")
# sel <- which(sml != "X")
# sml <- sml[sel]
# gset <- gset[ ,sel]
# 
# # log2 transformation
# ex <- exprs(gset)
# qx <- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
# LogC <- (qx[5] > 100) ||
#           (qx[6]-qx[1] > 50 && qx[2] > 0)
# if (LogC) { ex[which(ex <= 0)] <- NaN
#   exprs(gset) <- log2(ex) }
# 
# # assign samples to groups and set up design matrix
# gs <- factor(sml)
# groups <- make.names(c("Control","PD"))
# levels(gs) <- groups
# gset$group <- gs
# design <- model.matrix(~group + 0, gset)
# colnames(design) <- levels(gs)
# 
# fit <- lmFit(gset, design)  # fit linear model
# 
# # set up contrasts of interest and recalculate model coefficients
# cts <- paste(groups[1], groups[2], sep="-")
# cont.matrix <- makeContrasts(contrasts=cts, levels=design)
# fit2 <- contrasts.fit(fit, cont.matrix)
# 
# # compute statistics and table of top significant genes
# fit2 <- eBayes(fit2, 0.01)
# tT <- topTable(fit2, adjust="fdr", sort.by="B", number=250)
# 
# tT <- subset(tT, select=c("ID","adj.P.Val","P.Value","t","B","logFC","Gene.symbol","Gene.title"))
# write.table(tT, file=stdout(), row.names=F, sep="\t")
# 
# # Visualize and quality control test results.
# # Build histogram of P-values for all genes. Normal test
# # assumption is that most genes are not differentially expressed.
# tT2 <- topTable(fit2, adjust="fdr", sort.by="B", number=Inf)
# hist(tT2$adj.P.Val, col = "grey", border = "white", xlab = "P-adj",
#   ylab = "Number of genes", main = "P-adj value distribution")
# 
# # summarize test results as "up", "down" or "not expressed"
# dT <- decideTests(fit2, adjust.method="fdr", p.value=0.05)
# 
# # Venn diagram of results
# # vennDiagram(dT, circle.col=palette())
# 
# # create Q-Q plot for t-statistic
# t.good <- which(!is.na(fit2$F)) # filter out bad probes
# qqt(fit2$t[t.good], fit2$df.total[t.good], main="Moderated t statistic")
# 
# # volcano plot (log P-value vs log fold change)
# colnames(fit2) # list contrast names
# ct <- 1        # choose contrast of interest
# # volcanoplot(fit2, coef=ct, main=colnames(fit2)[ct], pch=20,
# #   highlight=length(which(dT[,ct]!=0)), names=rep('+', nrow(fit2)))
# 
# # MD plot (log fold change vs mean log expression)
# # highlight statistically significant (p-adj < 0.05) probes
# plotMD(fit2, column=ct, status=dT[,ct], legend=F, pch=20, cex=1)
# abline(h=0)
# 
# ################################################################
# # General expression data analysis
# ex <- exprs(gset)
# 
# # box-and-whisker plot
# dev.new(width=3+ncol(gset)/6, height=5)
# ord <- order(gs)  # order samples by group
# palette(c("#1B9E77", "#7570B3", "#E7298A", "#E6AB02", "#D95F02",
#           "#66A61E", "#A6761D", "#B32424", "#B324B3", "#666666"))
# par(mar=c(7,4,2,1))
# title <- paste ("GSE8397", "/", annotation(gset), sep ="")
# boxplot(ex[,ord], boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=gs[ord])
# legend("topleft", groups, fill=palette(), bty="n")
# dev.off()
# 
# # expression value distribution
# par(mar=c(4,4,2,1))
# title <- paste ("GSE8397", "/", annotation(gset), " value distribution", sep ="")
# plotDensities(ex, group=gs, main=title, legend ="topright")
# 
# # UMAP plot (dimensionality reduction)
# ex <- na.omit(ex) # eliminate rows with NAs
# ex <- ex[!duplicated(ex), ]  # remove duplicates
# ump <- umap(t(ex), n_neighbors = 15, random_state = 123)
# par(mar=c(3,3,2,6), xpd=TRUE)
# plot(ump$layout, main="UMAP plot, nbrs=15", xlab="", ylab="", col=gs, pch=20, cex=1.5)
# legend("topright", inset=c(-0.15,0), legend=levels(gs), pch=20,
# col=1:nlevels(gs), title="Group", pt.cex=1.5)
# library("maptools")  # point labels without overlaps
# pointLabel(ump$layout, labels = rownames(ump$layout), method="SANN", cex=0.6)
# 
# # mean-variance trend, helps to see if precision weights are needed
# # plotSA(fit2, main="Mean variance trend, GSE8397")


################################################################ 
# Downloaded table
GSE8397_diff_exp <- read.delim("~/Desktop/PD NM Project/GSE8397.top.table.1.tsv")

# Filtering for significantly differentially expressed genes
GSE8397_diff_exp <- GSE8397_diff_exp %>% filter(GSE8397_diff_exp$adj.P.Val < 0.05)


```

GSE8397 - B chip (GPL97)
```{r}
# Version info: R 3.2.3, Biobase 2.30.0, GEOquery 2.40.0, limma 3.26.8
################################################################
#   Differential expression analysis with limma
# library(GEOquery)
# library(limma)
# library(umap)
# 
# # load series and platform data from GEO
# 
# gset <- getGEO("GSE8397", GSEMatrix =TRUE, AnnotGPL=TRUE)
# if (length(gset) > 1) idx <- grep("GPL97", attr(gset, "names")) else idx <- 1
# gset <- gset[[idx]]
# 
# # make proper column names to match toptable 
# fvarLabels(gset) <- make.names(fvarLabels(gset))
# 
# # group membership for all samples
# gsms <- "XXXXXXXX00000X111111111X000000X111111111111111X"
# sml <- strsplit(gsms, split="")[[1]]
# 
# # filter out excluded samples (marked as "X")
# sel <- which(sml != "X")
# sml <- sml[sel]
# gset <- gset[ ,sel]
# 
# # log2 transformation
# ex <- exprs(gset)
# qx <- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
# LogC <- (qx[5] > 100) ||
#           (qx[6]-qx[1] > 50 && qx[2] > 0)
# if (LogC) { ex[which(ex <= 0)] <- NaN
#   exprs(gset) <- log2(ex) }
# 
# # assign samples to groups and set up design matrix
# gs <- factor(sml)
# groups <- make.names(c("Control","PD"))
# levels(gs) <- groups
# gset$group <- gs
# design <- model.matrix(~group + 0, gset)
# colnames(design) <- levels(gs)
# 
# fit <- lmFit(gset, design)  # fit linear model
# 
# # set up contrasts of interest and recalculate model coefficients
# cts <- paste(groups[1], groups[2], sep="-")
# cont.matrix <- makeContrasts(contrasts=cts, levels=design)
# fit2 <- contrasts.fit(fit, cont.matrix)
# 
# # compute statistics and table of top significant genes
# fit2 <- eBayes(fit2, 0.01)
# tT <- topTable(fit2, adjust="fdr", sort.by="B", number=250)
# 
# tT <- subset(tT, select=c("ID","adj.P.Val","P.Value","t","B","logFC","Gene.symbol","Gene.title"))
# write.table(tT, file=stdout(), row.names=F, sep="\t")
# 
# # Visualize and quality control test results.
# # Build histogram of P-values for all genes. Normal test
# # assumption is that most genes are not differentially expressed.
# tT2 <- topTable(fit2, adjust="fdr", sort.by="B", number=Inf)
# hist(tT2$adj.P.Val, col = "grey", border = "white", xlab = "P-adj",
#   ylab = "Number of genes", main = "P-adj value distribution")
# 
# # summarize test results as "up", "down" or "not expressed"
# dT <- decideTests(fit2, adjust.method="fdr", p.value=0.05)
# 
# # Venn diagram of results
# vennDiagram(dT, circle.col=palette())
# 
# # create Q-Q plot for t-statistic
# t.good <- which(!is.na(fit2$F)) # filter out bad probes
# qqt(fit2$t[t.good], fit2$df.total[t.good], main="Moderated t statistic")
# 
# # volcano plot (log P-value vs log fold change)
# colnames(fit2) # list contrast names
# ct <- 1        # choose contrast of interest
# volcanoplot(fit2, coef=ct, main=colnames(fit2)[ct], pch=20,
#   highlight=length(which(dT[,ct]!=0)), names=rep('+', nrow(fit2)))
# 
# # MD plot (log fold change vs mean log expression)
# # highlight statistically significant (p-adj < 0.05) probes
# plotMD(fit2, column=ct, status=dT[,ct], legend=F, pch=20, cex=1)
# abline(h=0)
# 
# ################################################################
# # General expression data analysis
# ex <- exprs(gset)
# 
# # box-and-whisker plot
# dev.new(width=3+ncol(gset)/6, height=5)
# ord <- order(gs)  # order samples by group
# palette(c("#1B9E77", "#7570B3", "#E7298A", "#E6AB02", "#D95F02",
#           "#66A61E", "#A6761D", "#B32424", "#B324B3", "#666666"))
# par(mar=c(7,4,2,1))
# title <- paste ("GSE8397", "/", annotation(gset), sep ="")
# boxplot(ex[,ord], boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=gs[ord])
# legend("topleft", groups, fill=palette(), bty="n")
# dev.off()
# 
# # expression value distribution
# par(mar=c(4,4,2,1))
# title <- paste ("GSE8397", "/", annotation(gset), " value distribution", sep ="")
# plotDensities(ex, group=gs, main=title, legend ="topright")
# 
# # UMAP plot (dimensionality reduction)
# ex <- na.omit(ex) # eliminate rows with NAs
# ex <- ex[!duplicated(ex), ]  # remove duplicates
# ump <- umap(t(ex), n_neighbors = 15, random_state = 123)
# par(mar=c(3,3,2,6), xpd=TRUE)
# plot(ump$layout, main="UMAP plot, nbrs=15", xlab="", ylab="", col=gs, pch=20, cex=1.5)
# legend("topright", inset=c(-0.15,0), legend=levels(gs), pch=20,
# col=1:nlevels(gs), title="Group", pt.cex=1.5)
# library("maptools")  # point labels without overlaps
# pointLabel(ump$layout, labels = rownames(ump$layout), method="SANN", cex=0.6)
# 
# # mean-variance trend, helps to see if precision weights are needed
# plotSA(fit2, main="Mean variance trend, GSE8397")

################################################################ 
# Downloaded table
GSE8397_diff_exp_2 <- read.delim("~/Desktop/PD NM Project/GSE8397.top.table.2.tsv")

# Filtering for significantly differentially expressed genes
GSE8397_diff_exp_2 <- GSE8397_diff_exp_2 %>% filter(GSE8397_diff_exp_2$adj.P.Val < 0.05)


```

GSE20292
  GSM606624, GSM606625, and GSM606626 were removed, as they were significantly different from all the other samples (according to the boxplot).
```{r}
# Version info: R 3.2.3, Biobase 2.30.0, GEOquery 2.40.0, limma 3.26.8
################################################################
#   Differential expression analysis with limma
# library(GEOquery)
# library(limma)
# library(umap)
# 
# # load series and platform data from GEO
# 
# gset <- getGEO("GSE20292", GSEMatrix =TRUE, AnnotGPL=TRUE)
# if (length(gset) > 1) idx <- grep("GPL96", attr(gset, "names")) else idx <- 1
# gset <- gset[[idx]]
# 
# # make proper column names to match toptable 
# fvarLabels(gset) <- make.names(fvarLabels(gset))
# 
# # group membership for all samples
# gsms <- "01111111010000000100110000XXX"
# sml <- strsplit(gsms, split="")[[1]]
# 
# # filter out excluded samples (marked as "X")
# sel <- which(sml != "X")
# sml <- sml[sel]
# gset <- gset[ ,sel]
# 
# # log2 transformation
# ex <- exprs(gset)
# qx <- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
# LogC <- (qx[5] > 100) ||
#           (qx[6]-qx[1] > 50 && qx[2] > 0)
# if (LogC) { ex[which(ex <= 0)] <- NaN
#   exprs(gset) <- log2(ex) }
# 
# # assign samples to groups and set up design matrix
# gs <- factor(sml)
# groups <- make.names(c("Control","PD"))
# levels(gs) <- groups
# gset$group <- gs
# design <- model.matrix(~group + 0, gset)
# colnames(design) <- levels(gs)
# 
# fit <- lmFit(gset, design)  # fit linear model
# 
# # set up contrasts of interest and recalculate model coefficients
# cts <- paste(groups[1], groups[2], sep="-")
# cont.matrix <- makeContrasts(contrasts=cts, levels=design)
# fit2 <- contrasts.fit(fit, cont.matrix)
# 
# # compute statistics and table of top significant genes
# fit2 <- eBayes(fit2, 0.01)
# tT <- topTable(fit2, adjust="fdr", sort.by="B", number=250)
# 
# tT <- subset(tT, select=c("ID","adj.P.Val","P.Value","t","B","logFC","Gene.symbol","Gene.title"))
# write.table(tT, file=stdout(), row.names=F, sep="\t")
# 
# # Visualize and quality control test results.
# # Build histogram of P-values for all genes. Normal test
# # assumption is that most genes are not differentially expressed.
# tT2 <- topTable(fit2, adjust="fdr", sort.by="B", number=Inf)
# hist(tT2$adj.P.Val, col = "grey", border = "white", xlab = "P-adj",
#   ylab = "Number of genes", main = "P-adj value distribution")
# 
# # summarize test results as "up", "down" or "not expressed"
# dT <- decideTests(fit2, adjust.method="fdr", p.value=0.05)
# 
# # Venn diagram of results
# vennDiagram(dT, circle.col=palette())
# 
# # create Q-Q plot for t-statistic
# t.good <- which(!is.na(fit2$F)) # filter out bad probes
# qqt(fit2$t[t.good], fit2$df.total[t.good], main="Moderated t statistic")
# 
# # volcano plot (log P-value vs log fold change)
# colnames(fit2) # list contrast names
# ct <- 1        # choose contrast of interest
# volcanoplot(fit2, coef=ct, main=colnames(fit2)[ct], pch=20,
#   highlight=length(which(dT[,ct]!=0)), names=rep('+', nrow(fit2)))
# 
# # MD plot (log fold change vs mean log expression)
# # highlight statistically significant (p-adj < 0.05) probes
# plotMD(fit2, column=ct, status=dT[,ct], legend=F, pch=20, cex=1)
# abline(h=0)
# 
# ################################################################
# # General expression data analysis
# ex <- exprs(gset)
# 
# # box-and-whisker plot
# ord <- order(gs)  # order samples by group
# palette(c("#1B9E77", "#7570B3", "#E7298A", "#E6AB02", "#D95F02",
#           "#66A61E", "#A6761D", "#B32424", "#B324B3", "#666666"))
# par(mar=c(7,4,2,1))
# title <- paste ("GSE20292", "/", annotation(gset), sep ="")
# boxplot(ex[,ord], boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=gs[ord])
# legend("topleft", groups, fill=palette(), bty="n")
# 
# # expression value distribution
# par(mar=c(4,4,2,1))
# title <- paste ("GSE20292", "/", annotation(gset), " value distribution", sep ="")
# plotDensities(ex, group=gs, main=title, legend ="topright")
# 
# # UMAP plot (dimensionality reduction)
# ex <- na.omit(ex) # eliminate rows with NAs
# ex <- ex[!duplicated(ex), ]  # remove duplicates
# ump <- umap(t(ex), n_neighbors = 11, random_state = 123)
# par(mar=c(3,3,2,6), xpd=TRUE)
# plot(ump$layout, main="UMAP plot, nbrs=11", xlab="", ylab="", col=gs, pch=20, cex=1.5)
# legend("topright", inset=c(-0.15,0), legend=levels(gs), pch=20,
# col=1:nlevels(gs), title="Group", pt.cex=1.5)
# library("maptools")  # point labels without overlaps
# pointLabel(ump$layout, labels = rownames(ump$layout), method="SANN", cex=0.6)
# 
# # mean-variance trend, helps to see if precision weights are needed
# plotSA(fit2, main="Mean variance trend, GSE20292")


################################################################ 
# Downloaded table
GSE20292_diff_exp <- read.delim("~/Desktop/PD NM Project/GSE20292.top.table.tsv")

# Filtering for significantly differentially expressed genes
GSE20292_diff_exp <- GSE20292_diff_exp %>% filter(GSE20292_diff_exp$adj.P.Val < 0.05)


```

GSE49036
  Only PD vs control analysis was done. Analysis was not performed for the four iLBD samples.
```{r}
# Version info: R 3.2.3, Biobase 2.30.0, GEOquery 2.40.0, limma 3.26.8
################################################################
#   Differential expression analysis with limma
# library(GEOquery)
# library(limma)
# library(umap)
# 
# # load series and platform data from GEO
# 
# gset <- getGEO("GSE49036", GSEMatrix =TRUE, AnnotGPL=TRUE)
# if (length(gset) > 1) idx <- grep("GPL570", attr(gset, "names")) else idx <- 1
# gset <- gset[[idx]]
# 
# # make proper column names to match toptable 
# fvarLabels(gset) <- make.names(fvarLabels(gset))
# 
# # group membership for all samples
# gsms <- "00000000XXXXX111111111111111"
# sml <- strsplit(gsms, split="")[[1]]
# 
# # filter out excluded samples (marked as "X")
# sel <- which(sml != "X")
# sml <- sml[sel]
# gset <- gset[ ,sel]
# 
# # log2 transformation
# ex <- exprs(gset)
# qx <- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
# LogC <- (qx[5] > 100) ||
#           (qx[6]-qx[1] > 50 && qx[2] > 0)
# if (LogC) { ex[which(ex <= 0)] <- NaN
#   exprs(gset) <- log2(ex) }
# 
# # assign samples to groups and set up design matrix
# gs <- factor(sml)
# groups <- make.names(c("Control","PD"))
# levels(gs) <- groups
# gset$group <- gs
# design <- model.matrix(~group + 0, gset)
# colnames(design) <- levels(gs)
# 
# fit <- lmFit(gset, design)  # fit linear model
# 
# # set up contrasts of interest and recalculate model coefficients
# cts <- paste(groups[1], groups[2], sep="-")
# cont.matrix <- makeContrasts(contrasts=cts, levels=design)
# fit2 <- contrasts.fit(fit, cont.matrix)
# 
# # compute statistics and table of top significant genes
# fit2 <- eBayes(fit2, 0.01)
# tT <- topTable(fit2, adjust="fdr", sort.by="B", number=250)
# 
# tT <- subset(tT, select=c("ID","adj.P.Val","P.Value","t","B","logFC","Gene.symbol","Gene.title"))
# write.table(tT, file=stdout(), row.names=F, sep="\t")
# 
# # Visualize and quality control test results.
# # Build histogram of P-values for all genes. Normal test
# # assumption is that most genes are not differentially expressed.
# tT2 <- topTable(fit2, adjust="fdr", sort.by="B", number=Inf)
# hist(tT2$adj.P.Val, col = "grey", border = "white", xlab = "P-adj",
#   ylab = "Number of genes", main = "P-adj value distribution")
# 
# # summarize test results as "up", "down" or "not expressed"
# dT <- decideTests(fit2, adjust.method="fdr", p.value=0.05)
# 
# # Venn diagram of results
# vennDiagram(dT, circle.col=palette())
# 
# # create Q-Q plot for t-statistic
# t.good <- which(!is.na(fit2$F)) # filter out bad probes
# qqt(fit2$t[t.good], fit2$df.total[t.good], main="Moderated t statistic")
# 
# # volcano plot (log P-value vs log fold change)
# colnames(fit2) # list contrast names
# ct <- 1        # choose contrast of interest
# volcanoplot(fit2, coef=ct, main=colnames(fit2)[ct], pch=20,
#   highlight=length(which(dT[,ct]!=0)), names=rep('+', nrow(fit2)))
# 
# # MD plot (log fold change vs mean log expression)
# # highlight statistically significant (p-adj < 0.05) probes
# plotMD(fit2, column=ct, status=dT[,ct], legend=F, pch=20, cex=1)
# abline(h=0)
# 
# ################################################################
# # General expression data analysis
# ex <- exprs(gset)
# 
# # box-and-whisker plot
# ord <- order(gs)  # order samples by group
# palette(c("#1B9E77", "#7570B3", "#E7298A", "#E6AB02", "#D95F02",
#           "#66A61E", "#A6761D", "#B32424", "#B324B3", "#666666"))
# par(mar=c(7,4,2,1))
# title <- paste ("GSE49036", "/", annotation(gset), sep ="")
# boxplot(ex[,ord], boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=gs[ord])
# legend("topleft", groups, fill=palette(), bty="n")
# 
# # expression value distribution
# par(mar=c(4,4,2,1))
# title <- paste ("GSE49036", "/", annotation(gset), " value distribution", sep ="")
# plotDensities(ex, group=gs, main=title, legend ="topright")
# 
# # UMAP plot (dimensionality reduction)
# ex <- na.omit(ex) # eliminate rows with NAs
# ex <- ex[!duplicated(ex), ]  # remove duplicates
# ump <- umap(t(ex), n_neighbors = 10, random_state = 123)
# par(mar=c(3,3,2,6), xpd=TRUE)
# plot(ump$layout, main="UMAP plot, nbrs=10", xlab="", ylab="", col=gs, pch=20, cex=1.5)
# legend("topright", inset=c(-0.15,0), legend=levels(gs), pch=20,
# col=1:nlevels(gs), title="Group", pt.cex=1.5)
# library("maptools")  # point labels without overlaps
# pointLabel(ump$layout, labels = rownames(ump$layout), method="SANN", cex=0.6)
# 
# # mean-variance trend, helps to see if precision weights are needed
# plotSA(fit2, main="Mean variance trend, GSE49036")


################################################################ 
# Downloaded table
GSE49036_diff_exp <- read.delim("~/Desktop/PD NM Project/GSE49036.top.table.tsv")

# Filtering for significantly differentially expressed genes
GSE49036_diff_exp <- GSE49036_diff_exp %>% filter(GSE49036_diff_exp$adj.P.Val < 0.05)

```


  Concatenating all differential expression data into one data frame
```{r}

PD_diff_expr <- bind_rows(GSE8397_diff_exp, GSE8397_diff_exp_2, GSE20292_diff_exp, GSE49036_diff_exp)
PD_diff_expr <- PD_diff_expr %>% filter(Gene.symbol != "")
PD_diff_expr <- PD_diff_expr %>% distinct(Gene.symbol, .keep_all = TRUE)

```

________________________________________________________________________________
________________________________________________________________________________

VERIFYING ACGs and mDBR genes ARE ALSO DIFFERENTIALLY EXPRESSED IN PD
________________________________________________________________________________
________________________________________________________________________________

```{r}

ACGs_PD <- intersect(PD_diff_expr$Gene.symbol, ACGs$Gene)

print(paste(length(intersect(PD_diff_expr$Gene.symbol, ACGs$Gene)), "ACGs are also differentially expressed in PD"))

mDBR_genes_PD <- intersect(PD_diff_expr$Gene.symbol, mDBR_genes$gene.symbol)

print(paste(length(intersect(PD_diff_expr$Gene.symbol, mDBR_genes$gene.symbol)), "genes differentially expressed in the SNpc are also differentially expressed in PD"))

ACGs_mDBR_genes_PD <- intersect(intersect(PD_diff_expr$Gene.symbol, mDBR_genes$gene.symbol), ACGs$Gene)

print(paste(length(intersect(intersect(PD_diff_expr$Gene.symbol, mDBR_genes$gene.symbol), ACGs$Gene)), "genes that are both correlated with age and differentially expressed in the SNpc are also differentially expressed in PD"))

```


________________________________________________________________________________
________________________________________________________________________________

VISUALIZING INTERSECTIONS BETWEEN ACGs, mDBR genes, POPULATION-SPECIFIC GENES 
________________________________________________________________________________
________________________________________________________________________________


Scatterplot of target gene expression vs age (UKBEC age correlations)

```{r}
library(ggplot2)
library(gridExtra)
library(cowplot)

gene_filt_SN_with_age <- data.frame(t(gene_filt_SN[, 1:101]))
gene_filt_SN_with_age$age <- age_gene_SN

TH_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=TH)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.20424344",
       y="Expression", 
       x="", 
       title="TH") +
  theme

DDC_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=DDC)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.2340363",
       y="", 
       x="",
       title="DDC") +
  theme

SLC18A2_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=SLC18A2)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.4081625",
       y="", 
       x="",
       title="SLC18A2") +
  theme

SLC6A3_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=SLC6A3)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.2968265",
       y="Expression",
       x="", 
       title="SLC6A3") +
  theme

GSTM2_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=GSTM2)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.1977581",
       y="", 
       x="",
       title="GSTM2") +
  theme

NQO1_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=NQO1)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.06190312",
       y="", 
       x="",
       title="NQO1") +
  theme

MAOA_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=MAOA)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.9786372",
       y="Expression", 
       x="Age", 
       title="MAOA") +
  theme

MAOB_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=MAOB)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.3945046",
       y="", 
       x="Age",
       title="MAOB") +
  theme

COMT_age_plot <- ggplot(gene_filt_SN_with_age, aes(x=age, y=COMT)) + 
  geom_point() + 
  geom_smooth(method="lm", se = FALSE) + 
  labs(subtitle = "P = 0.2063955",
       y="", 
       x="Age",
       title="COMT") +
  theme

plot_grid(TH_age_plot, DDC_age_plot, SLC18A2_age_plot, SLC6A3_age_plot, NQO1_age_plot, GSTM2_age_plot, MAOA_age_plot, MAOB_age_plot, COMT_age_plot, ncol = 3, nrow = 3)

```

________________________________________________________________________________

Component-to-residual plots of target genes from mDBR PSEA (observing DA neuron-specific fold change in SNpc relative to VTA)

```{r}
library(PSEA)
library(cowplot)

summary(DA_models_mDBR[["TH"]])
crplot(DA_models_mDBR[["TH"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["DDC"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal", newplot = FALSE)
crplot(DA_models_mDBR[["SLC18A2"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["SLC6A3"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["GSTM2"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["NQO1"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["MAOA"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["MAOB"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")
crplot(DA_models_mDBR[["COMT"]], "DA_reference_mDBR", g="DA_difference_mDBR", xlab = "Dopaminergic Neuron Reference Signal")


```


________________________________________________________________________________

December 27, 2020

ACGs, mDBR genes, and genes differentially expressed in PD

```{r}
library(VennDiagram)
library(RColorBrewer)

# c1 <- ACGs$Gene
# c2 <- mDBR_genes$gene.symbol
# c3 <- PD_diff_expr$Gene.symbol

myCol <- brewer.pal(3, "Pastel2")

grid.newpage()
venn_diagram_all <- draw.triple.venn(area1 = dim(ACGs)[1],
                 area2 = dim(mDBR_genes)[1],
                 area3 = dim(PD_diff_expr)[1],
                 n12 = (length(intersect(ACGs$Gene, mDBR_genes$gene.symbol))),
                 n23 = (length(intersect(mDBR_genes$gene.symbol, PD_diff_expr$Gene.symbol))),
                 n13 = (length(intersect(ACGs$Gene, PD_diff_expr$Gene.symbol))),
                 n123 = length(intersect(intersect(ACGs$Gene, mDBR_genes$gene.symbol), PD_diff_expr$Gene.symbol)),
                 cex = 2.2,
                 col = "black",
                 fill = c("red","orange", "yellow"),
                 category = c("ACGs", "mDBR", "PD-implicated"),
                 cat.cex = 2.5,
                 cat.fontface = "bold",
                 cat.pos = c(-27, 27, 135),
                 cat.dist = c(0.055, 0.055, 0.085),
                 cat.fontfamily = "sans",
                 rotation = 1)

venn_diagram_all

# venn.diagram(
#    x = list(c1, c2, c3),
#    category.names = c("ACGs", "mDBR", "PD"),
#    filename = "ACG_mDBR_PD_diff.png",
#    output = TRUE,
# 
#    # Output features
#    imagetype="png" ,
#    height = 1200 ,
#    width = 1200 ,
#    resolution = 600,
#    compression = "lzw",
# 
#    # Circles
#    lwd = 1,
#    fill = myCol,
# 
#    # Numbers
#    cex = .6,
#    fontface = "bold",
#    fontfamily = "sans",
# 
#    # Set names
#    cat.cex = 0.6,
#    cat.fontface = "bold",
#    cat.default.pos = "outer",
#    cat.pos = c(-27, 27, 135),
#    cat.dist = c(0.055, 0.055, 0.085),
#    cat.fontfamily = "sans",
#    rotation = 1
# )

```

________________________________________________________________________________

Volcano plot of mDBR genes, highlighting genes also exhibiting DA neuron-specific expression

```{r}
library(ggplot2)
library(ggrepel)

# Add a column to mDBR_genes data frame indicating whether probe is upregulated, downregulated, or not significant
mDBR_expr_genes[, "Differential Expression"] <- "Not Significant"
# If log2Foldchange > horz_max and pvalue < 0.05, set as "Upregulated" 
mDBR_expr_genes[, "Differential Expression"][log2(mDBR_expr_genes$fold.change) > 0 & 
                                         mDBR_expr_genes$p < 0.05] <- "Upregulated"
# If mDBR_expr_genes < horz_min and pvalue < 0.05, set as "Downregulated"
mDBR_expr_genes[, "Differential Expression"][log2(mDBR_expr_genes$fold.change) < 0 & 
                                         mDBR_expr_genes$p < 0.05] <- "Downregulated"

mDBR_expr_genes[, "Differential Expression"][(mDBR_expr_genes$gene.symbol %in% DA_specific_mDBR$Gene)] <- "DA-specific"

# Add a label column to label gene names on volcano plot
mDBR_expr_genes$label <- NA

mDBR_expr_genes$label[mDBR_expr_genes$differentially_expressed == "DA-specific" & mDBR_expr_genes$p < 0.05  & (mDBR_expr_genes$fold.change > quantile(mDBR_expr_genes$fold.change, 0.99) | mDBR_expr_genes$fold.change < quantile(mDBR_expr_genes$fold.change, 0.01))] <- mDBR_expr_genes$gene_symbol[mDBR_expr_genes$differentially_expressed == "DA-specific" & mDBR_expr_genes$p < 0.05 & (mDBR_expr_genes$fold.change > quantile(mDBR_expr_genes$fold.change, 0.99) | mDBR_expr_genes$fold.change < quantile(mDBR_expr_genes$fold.change, 0.01))]

volcano_plot_mDBR <- ggplot(data=mDBR_expr_genes, 
                       aes(x=log2(fold.change), 
                           y=-log10(p), 
                           col=differentially_expressed,
                           label=label)) + 
                      geom_point(size = 1, alpha = 0.5) + 
                      geom_text_repel(size = 4.5) +
                      scale_colour_manual(values = c("Not Significant"="grey", 
                                                     "Downregulated"="red", 
                                                     "Upregulated"="skyblue",
                                                     "DA-specific"="black")) + 
                      geom_hline(yintercept=-log10(0.05), 
                                 col="black") +
                      labs(x = bquote(""*log[2]*"FC"), 
                           y = bquote("-"*log[10]*" "*italic(P)*"-value")) +
                      ggtitle("Genes Differentially Expressed in Substantia Nigra Pars Compacta (SNpc)") + 
                      theme

volcano_plot_mDBR

```


________________________________________________________________________________

Heatmap of dopaminergic neuron-specific mDBR genes

```{r}
library(dplyr)

heatmap(as.matrix(t(AHBA_expr_SNpc_VTA[(rownames(AHBA_expr_SNpc_VTA) %in% DA_specific_mDBR$Gene),c(1:2, 5:16)])))

```

________________________________________________________________________________

Hierarchical clustering of genes exhibiting DA neuron-specific expression changes in SNpc and age-dependent expression changes in normal individuals using UKBEC expression data

```{r}
library(ggplot2)
library(ggdendro)
theme_set(theme_bw())

hc <- hclust(dist(gene_filt_SN[(rownames(gene_filt_SN) %in% intersect(DA_specific_mDBR$Gene, mDBR_genes$gene.symbol)),]), "ave")  # hierarchical clustering

# plot
ggdendrogram(hc, rotate = TRUE, size = 2)

```


________________________________________________________________________________


```{r}
# A <- all_ACG_mDBR_joined[(all_ACG_mDBR_joined$Gene %in% DA_specific_UKBEC_all$.),]
# 
# plot(A[, "Correlation Coefficient"], log2(A[,"fold.change"]), xlab = "Correlation Coefficient", ylab = "log2(fold change)")
# 
# B <- all_ACG_mDBR_joined[(all_ACG_mDBR_joined$Gene %in% age_specific_PD$Gene),]
# 
# C <- ACGs %>% inner_join(age_specific_PD, by = c("Gene" = "Gene"))
# 
# D <- mDBR_genes %>% inner_join(age_specific_PD, by = c("gene.symbol" = "Gene"))
# 
# plot(log2(D$fold.change.x), log2(D$fold.change.y), xlab = "FC in SNpc", ylab = "Age-specific FC in PD")

```


